diff -Naur libmediainfo-0.7.91.orig/Source/MediaInfo/Audio/File_SmpteSt0302.cpp libmediainfo-0.7.91/Source/MediaInfo/Audio/File_SmpteSt0302.cpp
--- a/Source/MediaInfo/Audio/File_SmpteSt0302.cpp	2016-12-01 13:10:26.000000000 +0300
+++ b/Source/MediaInfo/Audio/File_SmpteSt0302.cpp	2016-12-07 02:05:37.543683093 +0300
@@ -56,7 +56,7 @@
     #if MEDIAINFO_EVENTS
         pid=(int16u)-1;
         stream_id=(int8u)-1;
-    #endif MEDIAINFO_EVENTS
+    #endif // MEDIAINFO_EVENTS
 }
 
 //---------------------------------------------------------------------------
diff -Naur libmediainfo-0.7.91.orig/Source/MediaInfo/Audio/File_SmpteSt0302.h libmediainfo-0.7.91/Source/MediaInfo/Audio/File_SmpteSt0302.h
--- a/Source/MediaInfo/Audio/File_SmpteSt0302.h	2016-12-01 13:10:26.000000000 +0300
+++ b/Source/MediaInfo/Audio/File_SmpteSt0302.h	2016-12-07 02:06:10.864516473 +0300
@@ -36,7 +36,7 @@
     #if MEDIAINFO_EVENTS
         int16u pid;
         int8u  stream_id;
-    #endif MEDIAINFO_EVENTS
+    #endif // MEDIAINFO_EVENTS
 
 private :
     //Streams management
diff -Naur libmediainfo-0.7.91.orig/Source/MediaInfo/Audio/File_SmpteSt0337.cpp libmediainfo-0.7.91/Source/MediaInfo/Audio/File_SmpteSt0337.cpp
--- a/Source/MediaInfo/Audio/File_SmpteSt0337.cpp	2016-12-01 13:10:26.000000000 +0300
+++ b/Source/MediaInfo/Audio/File_SmpteSt0337.cpp	2016-12-07 02:06:49.741488668 +0300
@@ -157,7 +157,7 @@
         pid=(int16u)-1;
         stream_id=(int8u)-1;
         IgnoreGuardBandTest=false;
-    #endif MEDIAINFO_EVENTS
+    #endif // MEDIAINFO_EVENTS
 
     // Parser
     Parser=NULL;
diff -Naur libmediainfo-0.7.91.orig/Source/MediaInfo/Audio/File_SmpteSt0337.h libmediainfo-0.7.91/Source/MediaInfo/Audio/File_SmpteSt0337.h
--- a/Source/MediaInfo/Audio/File_SmpteSt0337.h	2016-12-01 13:10:26.000000000 +0300
+++ b/Source/MediaInfo/Audio/File_SmpteSt0337.h	2016-12-07 02:07:06.441906249 +0300
@@ -38,7 +38,7 @@
         int16u pid;
         int8u  stream_id;
         bool   IgnoreGuardBandTest;
-    #endif MEDIAINFO_EVENTS
+    #endif // MEDIAINFO_EVENTS
 
     // Constructor/Destructor
     File_SmpteSt0337();
