--- ./scripts/3proxy.cfg.orig	2020-11-27 16:25:26.206817106 +0300
+++ ./scripts/3proxy.cfg	2020-11-27 16:30:32.391530675 +0300
@@ -2,17 +2,17 @@
 nserver 8.8.8.8
 nserver 8.8.4.4
 
-config /conf/3proxy.cfg
-monitor /conf/3proxy.cfg
+config /opt/etc/3proxy/3proxy.cfg
+monitor /opt/etc/3proxy/3proxy.cfg
 
-log /logs/3proxy-%y%m%d.log D
+log /opt/var/log/3proxy-%y%m%d.log D
 rotate 60
-counter /count/3proxy.3cf
+counter /opt/var/3proxy.3cf
 
-users $/conf/passwd
+users $/opt/etc/3proxy/passwd
 
-include /conf/counters
-include /conf/bandlimiters
+include /opt/etc/3proxy/counters
+include /opt/etc/3proxy/bandlimiters
 
 auth strong
 deny * * 127.0.0.1
