--- a/Makefile.am
+++ b/Makefile.am
@@ -38,11 +38,6 @@ include $(srcdir)/include/Makefile.am.in
 
 SUBDIRS = zbar
 
-if USE_NLS
-  SUBDIRS += po
-  dist_doc_DATA += ABOUT-NLS
-endif
-
 zbar/libzbar.la:
 	$(MAKE) -C @builddir@/zbar libzbar.la
 
--- a/configure.ac
+++ b/configure.ac
@@ -135,10 +135,6 @@ dnl libraries
 
 AC_SEARCH_LIBS([clock_gettime], [rt pthread])
 AM_ICONV()
-AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION(0.20)
-AM_GNU_GETTEXT_REQUIRE_VERSION(0.18)
-AM_CONDITIONAL([USE_NLS], [test "x${USE_NLS}" = "xyes"])
 
 dnl libraries linkage
 AC_ARG_ENABLE([static_qt],
@@ -851,7 +847,6 @@ AC_CONFIG_FILES([
 Makefile
 gtk/Makefile
 java/Makefile
-po/Makefile.in
 zbar/Makefile
 zbar.pc
 zbar-gtk.pc
