--- a/conf/config.php
+++ b/conf/config.php
@@ -28,7 +28,7 @@
 	$do_diagnostic = true;			// Diagnose ruTorrent. Recommended to keep enabled, unless otherwise required.
 	$al_diagnostic = true;			// Diagnose auto-loader. Set to "false" to make composer plugins work.
 
-	$log_file = $_ENV['RU_LOG_FILE'] ?? '/tmp/errors.log'; // path to log file (comment or leave blank to disable logging)
+	$log_file = $_ENV['RU_LOG_FILE'] ?? '/opt/var/log/rutorrent_errors.log'; // path to log file (comment or leave blank to disable logging)
 
 	$saveUploadedTorrents = true;		// Save uploaded torrents to profile/torrents directory or not
 	$overwriteUploadedTorrents = false;	// Overwrite existing uploaded torrents in profile/torrents directory or make unique name
@@ -36,8 +36,8 @@
 	$topDirectory = $_ENV['RU_TOP_DIR'] ?? '/';			// Upper available directory. Absolute path with trail slash.
 	$forbidUserSettings = false;
 
-	$scgi_port = $_ENV['RU_SCGI_PORT'] ?? 5000;
-	$scgi_host = $_ENV['RU_SCGI_HOST'] ?? "127.0.0.1";
+	$scgi_port = $_ENV['RU_SCGI_PORT'] ?? 0;
+	$scgi_host = $_ENV['RU_SCGI_HOST'] ?? "unix:///opt/var/rpc.socket";
 
 	// For web->rtorrent link through unix domain socket
 	// (scgi_local in rtorrent conf file), change variables
@@ -52,11 +52,11 @@
 	// Can't be greater then 327625*1024 due to limitation in libtorrent ResourceManager::set_max_upload_unchoked function.
 
 	$pathToExternals = array(
-		"php"	=> '',			// Something like /usr/bin/php. If empty, will be found in PATH.
-		"curl"	=> '',			// Something like /usr/bin/curl. If empty, will be found in PATH.
-		"gzip"	=> '',			// Something like /usr/bin/gzip. If empty, will be found in PATH.
-		"id"	=> '',			// Something like /usr/bin/id. If empty, will be found in PATH.
-		"stat"	=> '',			// Something like /usr/bin/stat. If empty, will be found in PATH.
+		"php"	=> '/opt/bin/php-cgi',	// Something like /usr/bin/php. If empty, will be found in PATH.
+		"curl"	=> '/opt/bin/curl',	// Something like /usr/bin/curl. If empty, will be found in PATH.
+		"gzip"	=> '/opt/bin/gzip',	// Something like /usr/bin/gzip. If empty, will be found in PATH.
+		"id"	=> '/opt/bin/id',	// Something like /usr/bin/id. If empty, will be found in PATH.
+		"stat"	=> '/opt/bin/stat',	// Something like /usr/bin/stat. If empty, will be found in PATH.
 	);
 
 	$localHostedMode = false;		// Set to true if rTorrent is hosted on the SAME machine as ruTorrent
