--- a/php/util.php	2019-10-08 14:30:59.000000000 +0300
+++ b/php/util.php	2020-01-15 17:25:42.280655112 +0300
@@ -55,13 +55,13 @@
 		ini_set('magic_quotes_runtime', 0);
 		ini_set('magic_quotes_sybase', 0);
 	}
-	if(get_magic_quotes_gpc())
-	{
-		stripSlashesFromArray($_POST);
-		stripSlashesFromArray($_GET);
-		stripSlashesFromArray($_COOKIE);
-		stripSlashesFromArray($_REQUEST);
-	}
+//	if(get_magic_quotes_gpc())
+//	{
+//		stripSlashesFromArray($_POST);
+//		stripSlashesFromArray($_GET);
+//		stripSlashesFromArray($_COOKIE);
+//		stripSlashesFromArray($_REQUEST);
+//	}
 }
 
 fix_magic_quotes_gpc();