# IFA_F_MANAGETEMPADDR since Linux 3.14
# IFA_FLAGS since Linux 3.14

--- a/qtbase/src/network/configure.json
+++ b/qtbase/src/network/configure.json
@@ -171,7 +171,7 @@
                     "(void)RTM_NEWLINK; (void)RTM_NEWADDR;",
                     "(void)IFLA_ADDRESS; (void)IFLA_IFNAME;",
                     "(void)IFA_ADDRESS; (void)IFA_LABEL; (void)IFA_CACHEINFO;",
-                    "(void)(IFA_F_SECONDARY | IFA_F_DEPRECATED | IFA_F_PERMANENT | IFA_F_MANAGETEMPADDR);"
+                    "(void)(IFA_F_SECONDARY | IFA_F_DEPRECATED | IFA_F_PERMANENT);"
                 ]
             }
         },
--- a/qtbase/src/network/kernel/qnetworkinterface_linux.cpp
+++ b/qtbase/src/network/kernel/qnetworkinterface_linux.cpp
@@ -60,7 +60,7 @@
 #define ARPHRD_PHONET       820     /* v2.6.29: PhoNet media type */
 #define ARPHRD_PHONET_PIPE  821     /* v2.6.29: PhoNet pipe header */
 #define ARPHRD_IEEE802154   804     /* v2.6.31 */
-#define ARPHRD_6LOWPAN      825     /* v3.14: IPv6 over LoWPAN */
+//#define ARPHRD_6LOWPAN      825     /* v3.14: IPv6 over LoWPAN */
 
 QT_BEGIN_NAMESPACE
 
@@ -107,8 +107,8 @@ static QNetworkInterface::InterfaceType
     case ARPHRD_PHONET_PIPE:
         return QNetworkInterface::Phonet;
 
-    case ARPHRD_6LOWPAN:
-        return QNetworkInterface::SixLoWPAN;
+//    case ARPHRD_6LOWPAN:
+//        return QNetworkInterface::SixLoWPAN;
 
     case ARPHRD_TUNNEL:
     case ARPHRD_TUNNEL6:
@@ -412,10 +412,10 @@ static void getAddresses(int sock, char
                 }
                 break;
 
-            case IFA_FLAGS:
-                Q_ASSERT(payloadLen == 4);
-                flags = qFromUnaligned<quint32>(payloadPtr);
-                break;
+//            case IFA_FLAGS:
+//                Q_ASSERT(payloadLen == 4);
+//                flags = qFromUnaligned<quint32>(payloadPtr);
+//                break;
             }
         }
 
