--- a/dropbox_uploader.sh
+++ b/dropbox_uploader.sh
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/opt/bin/bash
 #
 # Dropbox Uploader
 #
@@ -20,7 +20,7 @@
 #
 
 #Default configuration file
-CONFIG_FILE=~/.dropbox_uploader
+CONFIG_FILE=/opt/etc/.dropbox_uploader
 
 #Default chunk size in Mb for the upload process
 #It is recommended to increase this value only if you have enough free space on your /tmp partition
@@ -30,9 +30,10 @@
 #Curl location
 #If not set, curl will be searched into the $PATH
 #CURL_BIN="/usr/bin/curl"
+CURL_ACCEPT_CERTIFICATES="-k"
 
 #Default values
-TMP_DIR="/tmp"
+TMP_DIR="/opt/tmp"
 DEBUG=0
 QUIET=0
 SHOW_PROGRESSBAR=0
