From 988d37b7fcd6c0817639be6368f8a6367547f1c1 Mon Sep 17 00:00:00 2001
From: Agnibho Mondal <mail@agnibho.com>
Date: Wed, 9 Dec 2015 04:02:13 +0530
Subject: [PATCH 1/2] List shows human readable size with -h option

---
 dropbox_uploader.sh | 26 ++++++++++++++++++++++++--
 1 file changed, 24 insertions(+), 2 deletions(-)

diff --git a/dropbox_uploader.sh b/dropbox_uploader.sh
index 1e7b830..25bb2a0 100755
--- a/dropbox_uploader.sh
+++ b/dropbox_uploader.sh
@@ -75,7 +75,7 @@ shopt -s nullglob #Bash allows filename patterns which match no files to expand
 shopt -s dotglob  #Bash includes filenames beginning with a "." in the results of filename expansion
 
 #Look for optional config file parameter
-while getopts ":qpskdf:" opt; do
+while getopts ":qpskdhf:" opt; do
     case $opt in
 
     f)
@@ -102,6 +102,10 @@ while getopts ":qpskdf:" opt; do
       SKIP_EXISTING_FILES=1
     ;;
 
+    h)
+      HUMAN_READABLE_SIZE=1
+    ;;
+
     \?)
       echo "Invalid option: -$OPTARG" >&2
       exit 1
@@ -187,6 +191,24 @@ function remove_temp_files
     fi
 }
 
+#Converts bytes to human readable format
+function convert_bytes
+{
+    if [[ $HUMAN_READABLE_SIZE == 1 ]]; then
+	if (($1 > 1073741824));then
+	    echo $(($1/1073741824)).$(($1%1073741824/100000000))"G";
+	elif (($1 > 1048576));then
+	    echo $(($1/1048576)).$(($1%1048576/100000))"M";
+	elif (($1 > 1024));then
+	    echo $(($1/1024)).$(($1%1024/100))"K";
+	else
+	    echo $1;
+	fi
+    else
+	echo $1;
+    fi
+}
+
 #Returns the file size in bytes
 function file_size
 {
@@ -999,7 +1021,7 @@ function db_list
 
                 local FILE=$(echo "$line" | sed -n 's/.*"path": *"\([^"]*\)".*/\1/p')
                 local IS_DIR=$(echo "$line" | sed -n 's/.*"is_dir": *\([^,]*\).*/\1/p')
-                local SIZE=$(echo "$line" | sed -n 's/.*"bytes": *\([0-9]*\).*/\1/p')
+                local SIZE=$(convert_bytes $(echo "$line" | sed -n 's/.*"bytes": *\([0-9]*\).*/\1/p'))
 
                 echo -e "$FILE:$IS_DIR;$SIZE" >> "$RESPONSE_FILE"
 

From a00ccc35dd2e54f70575e10995eb9b76a1ef7aea Mon Sep 17 00:00:00 2001
From: Agnibho Mondal <mail@agnibho.com>
Date: Wed, 9 Dec 2015 04:13:59 +0530
Subject: [PATCH 2/2] Option -h with list shows file sizes in human readable
 format

---
 README.md           | 3 +++
 dropbox_uploader.sh | 1 +
 2 files changed, 4 insertions(+)

diff --git a/README.md b/README.md
index f9ac289..c446f76 100644
--- a/README.md
+++ b/README.md
@@ -117,6 +117,9 @@ Enable DEBUG mode
 * **-q**  
 Quiet mode. Don't show progress meter or messages
 
+* **-h**  
+Show file sizes in human readable format
+
 * **-p**  
 Show cURL progress meter
 
diff --git a/dropbox_uploader.sh b/dropbox_uploader.sh
index 25bb2a0..24d43f5 100755
--- a/dropbox_uploader.sh
+++ b/dropbox_uploader.sh
@@ -262,6 +262,7 @@ function usage
     echo -e "\t-s            Skip already existing files when download/upload. Default: Overwrite"
     echo -e "\t-d            Enable DEBUG mode"
     echo -e "\t-q            Quiet mode. Don't show messages"
+    echo -e "\t-h            Show file sizes in human readable format"
     echo -e "\t-p            Show cURL progress meter"
     echo -e "\t-k            Doesn't check for SSL certificates (insecure)"
 
