diff -u orig/configure mod/configure
--- orig/configure	2010-04-08 02:40:17.000000000 +0400
+++ mod/configure	2011-06-02 00:21:30.000000000 +0400
@@ -24093,7 +24093,7 @@
 else
 
                             LDFLAGS="-L$SEARCH_DIR_LIBS $LDFLAGS_SAVE $ADD_PTHREAD_CFLAGS"
-                            unset ac_cv_lib_js_JS_NewObject
+                            ac_cv_lib_js_JS_NewObject=yes
                             { $as_echo "$as_me:$LINENO: checking for JS_NewObject in -ljs" >&5
 $as_echo_n "checking for JS_NewObject in -ljs... " >&6; }
 if test "${ac_cv_lib_js_JS_NewObject+set}" = set; then
@@ -31425,7 +31425,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <$mt_avformat_search_headers/ffmpeg/avformat.h>
+#include <$mt_avformat_search_headers/libavformat/avformat.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (ac_try="$ac_compile"
@@ -31466,7 +31466,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <$mt_avformat_search_headers/ffmpeg/avformat.h>
+#include <$mt_avformat_search_headers/libavformat/avformat.h>
 _ACEOF
 if { (ac_try="$ac_cpp conftest.$ac_ext"
 case "(($ac_try" in
@@ -31583,7 +31583,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (ac_try="$ac_compile"
@@ -31624,7 +31624,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <ffmpeg/avformat.h>
+#include <libavformat/avformat.h>
 _ACEOF
 if { (ac_try="$ac_cpp conftest.$ac_ext"
 case "(($ac_try" in
@@ -31726,7 +31726,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <$MT_SEARCHPATH_HEADERS/ffmpeg/avformat.h>
+#include <$MT_SEARCHPATH_HEADERS/libavformat/avformat.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (ac_try="$ac_compile"
@@ -31767,7 +31767,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <$MT_SEARCHPATH_HEADERS/ffmpeg/avformat.h>
+#include <$MT_SEARCHPATH_HEADERS/libavformat/avformat.h>
 _ACEOF
 if { (ac_try="$ac_cpp conftest.$ac_ext"
 case "(($ac_try" in
@@ -33423,7 +33423,7 @@
     else
 
 cat >>confdefs.h <<_ACEOF
-#define AVFORMAT_INCLUDE <ffmpeg/avformat.h>
+#define AVFORMAT_INCLUDE <libavformat/avformat.h>
 _ACEOF
 
     fi
@@ -33541,7 +33541,7 @@
 fi
 
     else
-        unset ac_cv_lib_avformat_av_register_all
+        ac_cv_lib_avformat_av_register_all=yes
         { $as_echo "$as_me:$LINENO: checking for av_register_all in -lavformat" >&5
 $as_echo_n "checking for av_register_all in -lavformat... " >&6; }
 if test "${ac_cv_lib_avformat_av_register_all+set}" = set; then
@@ -33710,7 +33710,10 @@
     CPPFLAGS=$CPPFLAGS_SAVE
 
 
-    AVFORMAT_LIBS=${mt_avformat_libs}
+    AVFORMAT_LIBS="-lavformat -lavcodec -lavdevice"
+    if test "x$mt_mysql_package_status" = xyes; then
+		AVFORMAT_LIBS="$AVFORMAT_LIBS -lstdc++"
+	fi
     AVFORMAT_LDFLAGS=${mt_avformat_ldflags}
     AVFORMAT_STATUS=${mt_avformat_library_status}
