--- a/configure.ac
+++ b/configure.ac
@@ -76,16 +76,19 @@
 SEARCH_DIR_LIBS="$SEARCH_DIR/lib"
 SEARCH_DIR_PROGS="$SEARCH_DIR/bin"
 
-PKG_PROG_PKG_CONFIG
+AC_ARG_WITH(js-h,
+        AC_HELP_STRING([--with-js-h=DIR], [search for js (spidermonkey) headers in DIR]),
+        [
+            JS_SEARCH_HEADERS="$withval"
+        ]
+)
 
-AC_ARG_WITH(js-pkgconfig,
-    AC_HELP_STRING([--with-js-pkgconfig=DIR], [search for libjs.pc or mozjs185.c in DIR]),
-    [
-        if test "x$withval" != "x"; then
-            AC_MSG_NOTICE([Will also search for packages in $withval])
-            export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$withval
-        fi
-    ])
+AC_ARG_WITH(js-libs,
+        AC_HELP_STRING([--with-js-libs=DIR], [search for js (spidermonkey) libraries in DIR]),
+        [
+            JS_SEARCH_LIBS="$withval"
+        ]
+)
 
 AC_ARG_WITH(iconv-h,
         AC_HELP_STRING([--with-iconv-h=DIR], [search for iconv headers in DIR]),
@@ -881,33 +884,186 @@
 
 ######### javascript
 
-JS_STATUS="disabled"
-MT_OPTION([libjs], [disable],
-          [compile with spidermonkey support],[],[])
+AC_ARG_ENABLE(libjs,
+              AC_HELP_STRING([--enable-libjs], [compile with spidermonkey support (default: yes)]),
+              [
+                JS_EN=$enableval 
+                if test "x$enableval" = xno; then
+                    JS_OK=disabled
+                else
+                    JS_OK=yes
+                fi
+              ],
+              [
+                JS_OK=yes
+              ]
+)
 
-if test "x$LIBJS_OPTION_ENABLED" = "xyes"; then
-    PKG_CHECK_MODULES([LIBJS], [libjs >= 1.8.5],
+XP_ADD=""
+
+if test "x$JS_OK" = xyes; then
+dnl    if test $DARWIN_OS -eq 1 ; then
+dnl        XP_ADD="-DXP_MAC"
+dnl    else
+        XP_ADD="-DXP_UNIX"
+dnl    fi
+
+    CXXFLAGS="$CXXFLAGS $XP_ADD"
+    CPPFLAGS="$CPPFLAGS $XP_ADD"
+
+    if test -n "$JS_SEARCH_HEADERS" ; then
+        AC_MSG_NOTICE([Will search for js (spidermonkey) headers in $JS_SEARCH_HEADERS])
+        AC_CHECK_HEADER($JS_SEARCH_HEADERS/jsapi.h,
+                [
+                    JS_CXXFLAGS="-I$JS_SEARCH_HEADERS"
+                ],
+                [
+                    AC_MSG_ERROR([jsapi.h not found in requested location $JS_SEARCH_HEADERS])
+                ]
+        )
+    
+    else
+    AC_CHECK_HEADER(jsapi.h,
+        [],
         [
-            JS_STATUS="yes"
-            AC_DEFINE([HAVE_JS], [1], [libjs presence])
+            for sm_path in $SEARCH_DIR_HEADERS \
+                           $SEARCH_DIR_HEADERS/js \
+                           $SEARCH_DIR_HEADERS/smjs \
+                           $SEARCH_DIR_HEADERS/mozjs \
+                           $SEARCH_DIR_HEADERS/spidermonkey; do 
+                if test "x$ac_cv_header_jsapi_h" != xyes ; then
+                    AC_MSG_CHECKING([checking alternative libjs header location: $sm_path])
+                    CXXFLAGS="$XP_ADD -I$sm_path"
+                    CPPFLAGS="$XP_ADD -I$sm_path"
+                    unset ac_cv_header_jsapi_h
+                    AC_CHECK_HEADER(jsapi.h,
+                                    [
+                                        JS_CXXFLAGS="-I$sm_path"
+                                        AC_MSG_RESULT([jsapi.h found in $sm_path])
+                                    ],
+                                    [])
+               fi
+            done
+        ])
+   
+        if test "x$ac_cv_header_jsapi_h" != xyes ; then
+                JS_OK=missing
+                AC_MSG_RESULT([not found, giving up])
+        fi
+    fi
+fi
+
+MOZLIB=
+if test "x$JS_OK" = xyes; then
+    if test -n "$JS_SEARCH_LIBS" ; then
+        AC_MSG_NOTICE([Will search for js (spidermonkey) libraries in $JS_SEARCH_LIBS])
+        LDFLAGS="-L$JS_SEARCH_LIBS $LDFLAGS_SAVE -ljs $ADD_PTHREAD_CFLAGS"
+        AC_CHECK_LIB(js, JS_NewObject,
+                [
+                    JS_LIBS="-L$JS_SEARCH_LIBS -ljs"
+                    MOZLIB=js
+                ],
+                [
+                    LDFLAGS="-L$JS_SEARCH_LIBS $LDFLAGS_SAVE -lsmjs $ADD_PTHREAD_CFLAGS"
+                    AC_CHECK_LIB(smjs, JS_NewObject,
+                        [
+                            JS_LIBS="-L$JS_SEARCH_LIBS $LDFLAGS_SAVE -lsmjs $ADD_PTHREAD_CFLAGS"
+                            MOZLIB=smjs
+                        ],
+                        [
+                            LDFLAGS="-L$JS_SEARCH_LIBS $LDFLAGS_SAVE -lmozjs $ADD_PTHREAD_CFLAGS"
+                            AC_CHECK_LIB(mozjs, JS_NewObject,
+                                [
+                                    JS_LIBS="-L$JS_SEARCH_LIBS -lmozjs"
+                                    MOZLIB=mozjs
+                                ],
+                                [
+                                    AC_MSG_ERROR([Spidermonkey not found in requested location $JS_SEARCH_LIBS])
+                                ]
+                            )
+                        ]
+                    )
+                ]
+        )
+    else
+    LDFLAGS="$LDFLAGS_SAVE $ADD_PTHREAD_CFLAGS"
+    AC_CHECK_LIB(js, JS_NewObject,
+        [
+            MOZLIB=js
+            JS_LIBS="-ljs"
         ],
         [
-            AC_MSG_WARN([$LIBJS_PKG_ERRORS])
-            PKG_CHECK_MODULES([LIBJS], [mozjs185],
+            unset ac_cv_lib_js_JS_NewObject
+            AC_CHECK_LIB(smjs, JS_NewObject,
                 [
-                    JS_STATUS="yes"
-                    AC_DEFINE([HAVE_JS], [1], [libjs presence])
+                    JS_LIBS="-lsmjs"
+                    MOZLIB=smjs
                 ],
                 [
-                    if test "x$LIBJS_OPTION_REQUESTED" = "xyes"; then
-                        AC_MSG_ERROR([$LIBJS_PKG_ERRORS])
-                    else
-                        AC_MSG_WARN([$LIBJS_PKG_ERRORS])
-                    fi
-                    JS_STATUS="missing"
-                ])
-        ])
+                    unset ac_cv_lib_smjs_JS_NewObject
+                    AC_CHECK_LIB(mozjs, JS_NewObject,
+                        [
+                            MOZLIB=mozjs
+                            JS_LIBS="-lmozjs"
+                        ],
+                        [
+                            LDFLAGS="-L$SEARCH_DIR_LIBS $LDFLAGS_SAVE $ADD_PTHREAD_CFLAGS"
+                            unset ac_cv_lib_js_JS_NewObject
+                            AC_CHECK_LIB(js, JS_NewObject,
+                                [
+                                    JS_LIBS="-L$SEARCH_DIR_LIBS -ljs"
+                                    MOZLIB=js
+                                ],
+                                [
+                                    unset ac_cv_lib_smjs_JS_NewObject
+                                    AC_CHECK_LIB(smjs, JS_NewObject,
+                                        [
+                                            JS_LIBS="-L$SEARCH_DIR_LIBS -lsmjs"
+                                            MOZLIB=smjs
+                                        ],
+                                        [
+                                            LDFLAGS="-L$SEARCH_DIR_LIBS $LDFLAGS_SAVE -lmozjs"
+                                            unset ac_cv_lib_mozjs_JS_NewObject
+                                            AC_CHECK_LIB(mozjs, JS_NewObject,
+                                                [
+                                                    JS_LIBS="-L$SEARCH_DIR_LIBS -lmozjs"
+                                                    MOZLIB=mozjs
+                                                ],
+                                                [
+                                                    JS_OK=missing
+                                                    AC_MSG_RESULT(libjs/libsmjs link failed, giving up)
+                                                ]
+                                            )
+                                        ]
+                                    )
+                                ]
+                            )
+                        ]
+                    )
+                ]
+            )
+       ]
+    )
+    fi
+fi
+if test "x$JS_OK" = xyes; then
+    AC_DEFINE([HAVE_JS], [1], [libjs presence])
+    AC_SUBST(JS_CXXFLAGS)
+    AC_SUBST(JS_LIBS)
+
+    LDFLAGS="$LDFLAGS $JS_LIBS"
+    CFLAGS="$CFLAGS $JS_CXXFLAGS"
+    CXXFLAGS="$CXXFLAGS $JS_CXXFLAGS"
 
+    AC_CHECK_LIB($MOZLIB, JS_ClearContextThread,
+            [
+                AC_DEFINE([JS_THREADSAFE], [1],
+                          [this version of spidermonkey is threadsafe])
+            ])
+else
+    if test "x$JS_EN" = xyes; then
+        AC_MSG_ERROR(unable to configure libjs support)
+    fi
 fi
 
 CXXFLAGS="$CXXFLAGS_SAVE"
@@ -1699,7 +1855,7 @@
 echo
 echo "sqlite3               : $SQLITE3_STATUS"
 echo "mysql                 : $MYSQL_STATUS"
-echo "libjs                 : $JS_STATUS"
+echo "libjs                 : $JS_OK"
 echo "libmagic              : $LIBMAGIC_STATUS"
 echo "inotify               : $INOTIFY_STATUS"
 echo "libexif               : $LIBEXIF_STATUS"
