--- a/contrib/hdtbl/Makefile.sub
+++ b/contrib/hdtbl/Makefile.sub
@@ -62,18 +62,7 @@
   examples/rainbow.roff \
   examples/short_reference.roff
 
-PROCESSEDEXAMPLEFILES=\
-  examples/chess_board.ps \
-  examples/color_boxes.ps \
-  examples/color_nested_tables.ps \
-  examples/color_table_cells.ps \
-  examples/color_transitions.ps \
-  examples/col_rowspan_colors.ps \
-  examples/fonts_n.ps \
-  examples/fonts_x.ps \
-  examples/mixed_pickles.ps \
-  examples/rainbow.ps \
-  examples/short_reference.ps
+PROCESSEDEXAMPLEFILES=
 
 EXAMPLEFILES_=`echo $(EXAMPLEFILES) | sed 's|examples/||g'`
 GENFILES_=`echo $(GENFILES) | sed 's|examples/||g'`

--- a/doc/Makefile.sub
+++ b/doc/Makefile.sub
@@ -50,11 +50,7 @@
   meintro_fr.me \
   pic.ms
 
-PROCESSEDDOCFILES=\
-  meref.ps \
-  meintro.ps \
-  meintro_fr.ps \
-  pic.ps
+PROCESSEDDOCFILES=
 
 HTMLDOCFILES=\
   pic.html
@@ -71,9 +67,7 @@
   grnexmpl.g \
   grnexmpl.me
 
-PROCESSEDEXAMPLEFILES=\
-  webpage.ps \
-  grnexmpl.ps
+PROCESSEDEXAMPLEFILES=
 
 HTMLEXAMPLEFILES=\
   webpage.html

