--- a/core/hashRegManager.cpp
+++ b/core/hashRegManager.cpp
@@ -36,6 +36,22 @@
     #include "../gui.win/RegisteredUserDialog.h"
     #include "../gui.win/RegisteredUsersDialog.h"
 #endif
+
+uint32_t SwapWord(uint32_t a)
+{
+  a = ((a & 0x000000FF) << 24) |
+      ((a & 0x0000FF00) <<  8) |
+      ((a & 0x00FF0000) >>  8) |
+      ((a & 0xFF000000) >> 24);
+  return a;
+}
+
+uint32_t SwapShort(uint16_t a)
+{
+  a = ((a & 0x00FF) << 8) | ((a & 0xFF00) >> 8);
+  return a;
+}
+
 //---------------------------------------------------------------------------
 RegManager * RegManager::m_Ptr = NULL;
 //---------------------------------------------------------------------------
@@ -588,7 +604,7 @@ void RegManager::Load(void) {
     }
 
     {
-        uint32_t ui32FileVersion = ntohl(*((uint32_t *)(pxbRegs.m_pItemDatas[1])));
+        uint32_t ui32FileVersion = SwapWord(*((uint32_t *)(pxbRegs.m_pItemDatas[1])));
 
         if(ui32FileVersion < 1) {
             return;
@@ -610,7 +626,7 @@ void RegManager::Load(void) {
 
     while(bSuccess == true) {
 		if(pxbRegs.m_ui16ItemLengths[0] < 65 && pxbRegs.m_ui16ItemLengths[1] < 65 && pxbRegs.m_ui16ItemLengths[2] == 2) {
-            iProfile = (uint16_t)ntohs(*((uint16_t *)(pxbRegs.m_pItemDatas[2])));
+            iProfile = (uint16_t)SwapShort(*((uint16_t *)(pxbRegs.m_pItemDatas[2])));
 
             if(iProfile > iProfilesCount) {
                 iProfile = iProfilesCount;
