--- ./Makefile.orig	2015-12-03 09:10:27.000000000 +0300
+++ ./Makefile	2015-12-03 09:38:49.202782853 +0300
@@ -1,11 +1,11 @@
 # Makefile for dungeon
 
-CC = gcc
+#CC = gcc
 # Where to install the program
 BINDIR = /usr/games
 
 # Where to install the data file
-LIBDIR = /usr/games/lib
+LIBDIR = /opt/share/zork
 
 # The dungeon program provides a ``more'' facility which tries to
 # figure out how many rows the terminal has.  Several mechanisms are
@@ -17,8 +17,8 @@
 # variable may need to be set to -lcurses.  On some it may need to
 # be /usr/lib/termcap.o.  These options are commented out below.
 # LIBS = -ltermcap
-TERMFLAG =
-LIBS = -lcurses
+#TERMFLAG =
+#LIBS = -lcurses
 # LIBS = /usr/lib/termcap.o
 
 # more option 2: use the terminfo routines.  On some systems the LIBS
@@ -28,8 +28,8 @@
 # TERMFLAG = -DMORE_TERMINFO
 
 # more option 3: assume all terminals have 24 rows
-# LIBS =
-# TERMFLAG = -DMORE_24
+LIBS =
+TERMFLAG = -DMORE_24
 
 # more option 4: don't use the more facility at all
 # LIBS =
@@ -44,7 +44,7 @@
 
 # Compilation flags
 # CFLAGS = -g -DDEBUG
-CFLAGS = -O3
+#CFLAGS = -O3
 # On SCO Unix Development System 3.2.2a, the const type qualifier does
 # not work correctly when using cc.  The following line will cause it
 # to not be used and should be uncommented.
@@ -75,7 +75,7 @@
 	rm -f $(OBJS) dungeon dtextc.dat core dsave.dat *~
 
 dtextc.dat:
-	cat dtextc.uu1 dtextc.uu2 dtextc.uu3 dtextc.uu4 | uudecode
+	cat dtextc.uu1 dtextc.uu2 dtextc.uu3 dtextc.uu4 | ./uudecode
 
 dinit.o: dinit.c funcs.h vars.h
 	$(CC) $(CFLAGS) $(GDTFLAG) -DTEXTFILE=\"$(LIBDIR)/dtextc.dat\" -c dinit.c
