--- a/dverb2.c
+++ b/dverb2.c
@@ -22,7 +22,7 @@ void savegm_()
    /* 						!DISABLE GAME. */
    /* Note: save file format is different for PDP vs. non-PDP versions */
 
-   if ((e = fopen("dsave.dat", BINWRITE)) == NULL)
+   if ((e = fopen("/opt/var/dsave.dat", BINWRITE)) == NULL)
       goto L100;
 
    gttime_(&i);
@@ -113,7 +113,7 @@ void rstrgm_()
    /* 						!DISABLE GAME. */
    /* Note: save file format is different for PDP vs. non-PDP versions */
 
-   if ((e = fopen("dsave.dat", BINREAD)) == NULL)
+   if ((e = fopen("/opt/var/dsave.dat", BINREAD)) == NULL)
       goto L100;
 
 #define do_uio(i, zbuf, cbytes) \
