--- a/configure.ac
+++ b/configure.ac
@@ -12,10 +12,6 @@
 # Checks for programs.
 AC_PROG_CC
 
-# BSD locations for headers and libraries from packages, Linux locations for self-compiled stuff.
-CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-LDFLAGS="$LDFLAGS -L/usr/local/lib"
-
 # Checks for libraries.
 AC_SEARCH_LIBS([clock_gettime], [rt])
 AC_SEARCH_LIBS([backtrace], [execinfo])
@@ -109,7 +105,7 @@
 AC_MSG_CHECKING([for python])
 AC_ARG_ENABLE(python,[--enable-python/--disable-python],
   [
-    if test "x$enableval" = "xyes" ; then
+    if test "x$enableval" = "xno" ; then
       AC_MSG_RESULT([disabled])
     else
       AC_MSG_RESULT([enabled])
