--- a/cmake/FindPolarSSL.cmake
+++ b/cmake/FindPolarSSL.cmake
@@ -13,7 +13,7 @@
 
 include(FindPackageHandleStandardArgs)
 
-find_path(POLARSSL_INCLUDE_DIRS NAMES mbedtls/ssl.h HINTS /usr/local/include)
+find_path(POLARSSL_INCLUDE_DIRS NAMES polarssl/ssl.h)
 set(POLARSSL_REAL_NAME MBEDTLS)
 if( "${POLARSSL_INCLUDE_DIRS}" STREQUAL "POLARSSL_INCLUDE_DIRS-NOTFOUND")
   find_path(POLARSSL_INCLUDE_DIRS NAMES polarssl/ssl.h)
@@ -49,31 +49,6 @@
   return()
 endif()
 
-execute_process(
-  COMMAND echo "#include <${POLARSSL_INC_FOLDER}/version.h>\n#include <stdio.h>\nint main(){printf(${POLARSSL_REAL_NAME}_VERSION_STRING);return 0;}"
-  OUTPUT_FILE a.c
-)
-execute_process(
-  COMMAND ${CMAKE_C_COMPILER} a.c -I${POLARSSL_INCLUDE_DIRS} ${POLARSSL_LIBRARIES}
-)
-execute_process(
-  COMMAND ./a.out
-  OUTPUT_VARIABLE POLARSSL_VERSION_STRING
-)
-execute_process(
-  COMMAND ${CMAKE_COMMAND} -E remove a.c a.out
-)
-
-if( "${POLARSSL_VERSION_STRING}" STREQUAL "2.0.0" AND NOT "${POLARSSL_USED_LIBRARY}" STREQUAL "mbedcrypto" )
-  message("*** WARNING *** Your mbedTLS version is 2.0.0, it's possible the `make' command doesn't work.\nPlease refer to the INSTALL.md's \"mbedTLS 2.0.0\" section if you have any problem.\n")
-endif()
-
-string(REPLACE "." ";" POLARSSL_VERSION_LIST ${POLARSSL_VERSION_STRING})
-
-list(GET ${POLARSSL_VERSION_LIST} 0 POLARSSL_VERSION_MAJOR)
-list(GET ${POLARSSL_VERSION_LIST} 1 POLARSSL_VERSION_MINOR)
-list(GET ${POLARSSL_VERSION_LIST} 2 POLARSSL_VERSION_PATCH)
-
 set(POLARSSL_FOUND True)
 mark_as_advanced(
   POLARSSL_FOUND
--- a/include/dislocker/ssl_bindings.h.in
+++ b/include/dislocker/ssl_bindings.h.in
@@ -26,9 +26,9 @@
 /*
  * Here stand the bindings for polarssl SHA256/SHA2/SHA-2 function for dislocker
  */
-#include "@POLARSSL_INC_FOLDER@/config.h"
-#include "@POLARSSL_INC_FOLDER@/version.h"
-#include "@POLARSSL_INC_FOLDER@/aes.h"
+#include "polarssl/config.h"
+#include "polarssl/version.h"
+#include "polarssl/aes.h"
 
 // Function's name changed
 #if defined(MBEDTLS_SHA256_C)
