--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,15 +144,6 @@ endif()
 math(EXPR qpdf_SOVERSION "${PROJECT_VERSION_MAJOR} + 18")
 set(qpdf_LIBVERSION ${qpdf_SOVERSION}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH})
 
-if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
-  message(FATAL_ERROR "
-Please build with cmake in a subdirectory, e.g.
- mkdir build
- cmake ..
- cmake --build .
-Please remove CMakeCache.txt and the CMakeFiles directories.")
-endif()
-
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
