--- a/configure.ac
+++ b/configure.ac
@@ -21,18 +21,6 @@
     fi],
    [CHECK_AUTOFILES=1])
 
-# Check to see if automatically generated files are outdated and if we
-# can update them.
-AC_CHECK_PROG(AUTOCONF,autoconf,1,0)
-AC_CHECK_PROG(AUTOHEADER,autoheader,1,0)
-AC_CHECK_PROG(ACLOCAL,aclocal,1,0)
-AC_CHECK_PROG(SHA256SUM,sha256sum,1,0)
-if test "$CHECK_AUTOFILES$AUTOCONF$AUTOHEADER$ACLOCAL$SHA256SUM" = "11111"; then
-   if ! sha256sum -c autofiles.sums; then
-       AC_MSG_ERROR(autofiles are autodated; rerun autogen.sh)
-   fi
-fi
-
 AC_PROG_CC
 AC_PROG_CC_C99
 AC_PROG_CXX
