From: Sebastian Ramacher <sramacher@debian.org>
Date: Sun, 19 Sep 2021 11:34:49 +0200
Subject: Fix handling of local m4 macros

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index fc14fd1..d9482ca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.5)
 AC_INIT(cgdb, [m4_esyscmd_s([cat VERSION])])
-AC_CONFIG_AUX_DIR(config)
+AC_CONFIG_MACRO_DIRS([config])
 AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE
 
