--- a/libUseful-2.6/socket.c
+++ b/libUseful-2.6/socket.c
@@ -104,7 +104,7 @@
 #endif
 
 #ifdef SO_REUSEPORT
-if (SetFlags & SOCK_REUSEPORT) result=setsockopt(sock, SOL_SOCKET, SO_REUSEPORT, &result,sizeof(int));
+if (SetFlags & SOCK_REUSEPORT) result=setsockopt(sock, SOL_SOCKET, SO_REUSEADDR, &result,sizeof(int));
 #endif
 
 #ifdef IP_TRANSPARENT
@@ -132,7 +132,7 @@
 #endif
 
 #ifdef SO_REUSEPORT
-if (UnsetFlags & SOCK_REUSEPORT) result=setsockopt(sock, SOL_SOCKET, SO_REUSEPORT, &result,sizeof(int));
+if (UnsetFlags & SOCK_REUSEPORT) result=setsockopt(sock, SOL_SOCKET, SO_REUSEADDR, &result,sizeof(int));
 #endif
 
 #ifdef IP_TRANSPARENT
