--- a/vendor/cryptopp/vendor_cryptopp/config.h
+++ b/vendor/cryptopp/vendor_cryptopp/config.h
@@ -61,7 +61,7 @@
 // may use higher ISAs depending on compiler options, but the library will not
 // explictly use the ISAs. When disabling ASM, it is best to do it from
 // config.h to ensure the library and all programs share the setting.
-// #define CRYPTOPP_DISABLE_ASM 1
+#define CRYPTOPP_DISABLE_ASM 1
 
 // https://github.com/weidai11/cryptopp/issues/719
 #if defined(__native_client__)
