--- a/backend/beh.c
+++ b/backend/beh.c
@@ -161,7 +161,7 @@ main(int  argc,				// I - Number of comm
 
     tmpdir = getenv("TMPDIR");
     if (!tmpdir)
-      tmpdir = "/tmp";
+      tmpdir = "/opt/tmp";
     snprintf(tmpfilename, sizeof(tmpfilename), "%s/beh-XXXXXX", tmpdir);
     fd = mkstemp(tmpfilename);
     if (fd < 0)
--- a/filter/foomatic-rip/foomaticrip.c
+++ b/filter/foomatic-rip/foomaticrip.c
@@ -181,7 +181,8 @@ char gspath[PATH_MAX] = "gs";
 char echopath[PATH_MAX] = "echo";
 
 // CUPS raster drivers are searched here
-char cupsfilterpath[PATH_MAX] = "/usr/local/lib/cups/filter:"
+char cupsfilterpath[PATH_MAX] = "/opt/lib/cups/filter:"
+                                "/usr/local/lib/cups/filter:"
                                 "/usr/local/libexec/cups/filter:"
                                 "/opt/cups/filter:"
                                 "/usr/lib/cups/filter";
--- a/filter/foomatic-rip/foomaticrip.h
+++ b/filter/foomatic-rip/foomaticrip.h
@@ -31,3 +31,3 @@
 #ifndef LOG_FILE
-#define LOG_FILE "/tmp/foomatic-rip"
+#define LOG_FILE "/opt/var/log/foomatic-rip"
 #endif
--- a/filter/foomatic-rip/spooler.c
+++ b/filter/foomatic-rip/spooler.c
@@ -121 +121,4 @@ find_ppdfile(const char *user_default_pa
-
+  snprintf(job->ppdfile, 2048, "/opt/etc/cups/ppd/%s.ppd", job->printer);
+  if (access(job->ppdfile, R_OK) == 0)
+    return (1);
+
--- a/filter/foomatic-rip/util.c
+++ b/filter/foomatic-rip/util.c
@@ -30,7 +30,7 @@ temp_dir()
 
   if (!tmpdir)
   {
-    const char *dirs[] = { getenv("TMPDIR"), P_tmpdir, "/tmp" };
+    const char *dirs[] = { getenv("TMPDIR"), P_tmpdir, "/opt/tmp" };
     int i;
 
     for (i = 0; i < (sizeof(dirs) / sizeof(dirs[0])); i++)
