--- ./pyproject.toml.orig	2025-05-05 10:07:43.000000000 +0000
+++ ./pyproject.toml	2026-01-11 15:03:26.604931048 +0000
@@ -6,8 +6,6 @@
 build-backend = 'mesonpy'
 backend-path = ['.']
 requires = [
-  'meson >= 0.64.0; python_version < "3.12"',
-  'meson >= 1.2.3; python_version >= "3.12"',
   'packaging >= 23.2',
   'pyproject-metadata >= 0.9.0',
   'tomli >= 1.0.0; python_version < "3.11"',
@@ -35,8 +33,6 @@
 ]
 
 dependencies = [
-  'meson >= 0.64.0; python_version < "3.12"',
-  'meson >= 1.2.3; python_version >= "3.12"',
   'packaging >= 23.2',
   'pyproject-metadata >= 0.9.0',
   'tomli >= 1.0.0; python_version < "3.11"',
