--- a/init/Makefile.am
+++ b/init/Makefile.am
@@ -80,7 +80,7 @@ endif
 		-e 's#@''INITCONFIGDIR@#$(INITCONFIGDIR)#g' \
 		-e 's#@''INITDDIR@#$(INITDDIR)#g' \
 		-e 's#@''LOCALSTATEDIR@#$(localstatedir)#g' \
-		-e 's#@''BASHPATH@#${BASHPATH}#g' \
+		-e 's#@''BASHPATH@#/opt/bin/bash#g' \
 		-e 's#@''QNETDDIR@#$(qnetddir)#g' \
 	    > $@-t
 	mv $@-t $@
--- a/qdevices/Makefile.am
+++ b/qdevices/Makefile.am
@@ -88,7 +88,7 @@ corosync_qnetd_LDADD		= $(nss_LIBS)   $(
 
 corosync-qnetd-certutil: corosync-qnetd-certutil.sh
 	sed -e 's#@''DATADIR@#${datadir}#g' \
-	    -e 's#@''BASHPATH@#${BASHPATH}#g' \
+	    -e 's#@''BASHPATH@#/opt/bin/bash#g' \
 	    -e 's#@''COROSYSCONFDIR@#${COROSYSCONFDIR}#g' \
 	    $< > $@
 
@@ -161,7 +161,7 @@ corosync_qdevice_LDADD  = $(nss_LIBS) $(
 
 corosync-qdevice-net-certutil: corosync-qdevice-net-certutil.sh
 	sed -e 's#@''DATADIR@#${datadir}#g' \
-	    -e 's#@''BASHPATH@#${BASHPATH}#g' \
+	    -e 's#@''BASHPATH@#/opt/bin/bash#g' \
 	    -e 's#@''COROSYSCONFDIR@#${COROSYSCONFDIR}#g' \
 	    $< > $@
 
