diff -Naur cryfs-0.9.7.orig/src/blockstore/implementations/compressing/compressors/Gzip.cpp cryfs-0.9.7/src/blockstore/implementations/compressing/compressors/Gzip.cpp
--- cryfs-0.9.7.orig/src/blockstore/implementations/compressing/compressors/Gzip.cpp	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/blockstore/implementations/compressing/compressors/Gzip.cpp	2017-03-02 12:22:10.864280283 +0300
@@ -1,5 +1,5 @@
 #include "Gzip.h"
-#include <cryptopp/gzip.h>
+#include <crypto++/gzip.h>
 
 using cpputils::Data;
 
diff -Naur cryfs-0.9.7.orig/src/cpp-utils/CMakeLists.txt cryfs-0.9.7/src/cpp-utils/CMakeLists.txt
--- cryfs-0.9.7.orig/src/cpp-utils/CMakeLists.txt	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/cpp-utils/CMakeLists.txt	2017-03-02 12:20:35.436280283 +0300
@@ -43,7 +43,7 @@
 target_link_libraries(${PROJECT_NAME} PUBLIC pthread curl)
 
 # TODO From Crypto++ 5.7 on, it should support cmake with find_package() instead of find_library().
-find_library_with_path(CryptoPP cryptopp CRYPTOPP_LIB_PATH)
+find_library_with_path(CryptoPP crypto++ CRYPTOPP_LIB_PATH)
 target_link_libraries(${PROJECT_NAME} PUBLIC ${CryptoPP} scrypt spdlog)
 
 target_add_boost(${PROJECT_NAME} filesystem system thread)
diff -Naur cryfs-0.9.7.orig/src/cpp-utils/crypto/symmetric/CFB_Cipher.h cryfs-0.9.7/src/cpp-utils/crypto/symmetric/CFB_Cipher.h
--- cryfs-0.9.7.orig/src/cpp-utils/crypto/symmetric/CFB_Cipher.h	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/cpp-utils/crypto/symmetric/CFB_Cipher.h	2017-03-02 12:22:01.404280283 +0300
@@ -6,7 +6,7 @@
 #include "../../data/Data.h"
 #include "../../random/Random.h"
 #include <boost/optional.hpp>
-#include <cryptopp/modes.h>
+#include <crypto++/modes.h>
 #include "Cipher.h"
 
 namespace cpputils {
diff -Naur cryfs-0.9.7.orig/src/cpp-utils/crypto/symmetric/ciphers.h cryfs-0.9.7/src/cpp-utils/crypto/symmetric/ciphers.h
--- cryfs-0.9.7.orig/src/cpp-utils/crypto/symmetric/ciphers.h	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/cpp-utils/crypto/symmetric/ciphers.h	2017-03-02 12:21:50.704280283 +0300
@@ -2,11 +2,11 @@
 #ifndef MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 #define MESSMER_CPPUTILS_CRYPTO_SYMMETRIC_CIPHERS_H_
 
-#include <cryptopp/aes.h>
-#include <cryptopp/twofish.h>
-#include <cryptopp/serpent.h>
-#include <cryptopp/cast.h>
-#include <cryptopp/mars.h>
+#include <crypto++/aes.h>
+#include <crypto++/twofish.h>
+#include <crypto++/serpent.h>
+#include <crypto++/cast.h>
+#include <crypto++/mars.h>
 #include "GCM_Cipher.h"
 #include "CFB_Cipher.h"
 
diff -Naur cryfs-0.9.7.orig/src/cpp-utils/crypto/symmetric/GCM_Cipher.h cryfs-0.9.7/src/cpp-utils/crypto/symmetric/GCM_Cipher.h
--- cryfs-0.9.7.orig/src/cpp-utils/crypto/symmetric/GCM_Cipher.h	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/cpp-utils/crypto/symmetric/GCM_Cipher.h	2017-03-02 12:21:28.024280283 +0300
@@ -5,7 +5,7 @@
 #include "../../data/FixedSizeData.h"
 #include "../../data/Data.h"
 #include "../../random/Random.h"
-#include <cryptopp/gcm.h>
+#include <crypto++/gcm.h>
 #include "Cipher.h"
 
 namespace cpputils {
diff -Naur cryfs-0.9.7.orig/src/cpp-utils/data/FixedSizeData.h cryfs-0.9.7/src/cpp-utils/data/FixedSizeData.h
--- cryfs-0.9.7.orig/src/cpp-utils/data/FixedSizeData.h	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/cpp-utils/data/FixedSizeData.h	2017-03-02 12:20:45.960280283 +0300
@@ -2,7 +2,7 @@
 #ifndef MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 #define MESSMER_CPPUTILS_DATA_FIXEDSIZEDATA_H_
 
-#include <cryptopp/hex.h>
+#include <crypto++/hex.h>
 #include <string>
 #include <cstring>
 #include "../assert/assert.h"
diff -Naur cryfs-0.9.7.orig/src/cpp-utils/random/OSRandomGenerator.h cryfs-0.9.7/src/cpp-utils/random/OSRandomGenerator.h
--- cryfs-0.9.7.orig/src/cpp-utils/random/OSRandomGenerator.h	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/cpp-utils/random/OSRandomGenerator.h	2017-03-02 12:21:08.328280283 +0300
@@ -3,7 +3,7 @@
 #define MESSMER_CPPUTILS_RANDOM_OSRANDOMGENERATOR_H
 
 #include "RandomGenerator.h"
-#include <cryptopp/osrng.h>
+#include <crypto++/osrng.h>
 
 namespace cpputils {
     class OSRandomGenerator final : public RandomGenerator {
diff -Naur cryfs-0.9.7.orig/src/cpp-utils/random/RandomGeneratorThread.h cryfs-0.9.7/src/cpp-utils/random/RandomGeneratorThread.h
--- cryfs-0.9.7.orig/src/cpp-utils/random/RandomGeneratorThread.h	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/src/cpp-utils/random/RandomGeneratorThread.h	2017-03-02 12:20:57.880280283 +0300
@@ -4,7 +4,7 @@
 
 #include "../thread/LoopThread.h"
 #include "ThreadsafeRandomDataBuffer.h"
-#include <cryptopp/osrng.h>
+#include <crypto++/osrng.h>
 
 namespace cpputils {
     //TODO Test
diff -Naur cryfs-0.9.7.orig/test/cryfs/config/CompatibilityTest.cpp cryfs-0.9.7/test/cryfs/config/CompatibilityTest.cpp
--- cryfs-0.9.7.orig/test/cryfs/config/CompatibilityTest.cpp	2017-02-11 15:34:50.000000000 +0300
+++ cryfs-0.9.7/test/cryfs/config/CompatibilityTest.cpp	2017-03-02 12:21:18.424280283 +0300
@@ -2,7 +2,7 @@
 #include <vector>
 #include <boost/filesystem.hpp>
 #include <cpp-utils/data/Data.h>
-#include <cryptopp/hex.h>
+#include <crypto++/hex.h>
 #include <cpp-utils/crypto/symmetric/ciphers.h>
 #include <cpp-utils/tempfile/TempFile.h>
 #include <cryfs/config/CryConfigFile.h>
--- a/src/cpp-utils/crypto/cryptopp_byte.h
+++ b/src/cpp-utils/crypto/cryptopp_byte.h
@@ -2,7 +2,7 @@
 #ifndef _CPPUTILS_CRYPTO_CRYPTOPP_BYTE_H
 #define _CPPUTILS_CRYPTO_CRYPTOPP_BYTE_H
 
-#include <cryptopp/cryptlib.h>
+#include <crypto++/cryptlib.h>
 
 // If we're running an older CryptoPP version, CryptoPP::byte isn't defined yet.
 // Define it. Refer to "byte" type in the global namespace (placed by CryptoPP).
