--- a/vendor/scrypt/CMakeLists.txt
+++ b/vendor/scrypt/CMakeLists.txt
@@ -14,7 +14,7 @@
 
 ExternalProject_Add(scrypt_original
         SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/scrypt-1.2.0/scrypt
-        CONFIGURE_COMMAND /usr/bin/env CC=${CMAKE_C_COMPILER} CFLAGS=${SCRYPT_CFLAGS} ${CMAKE_CURRENT_SOURCE_DIR}/scrypt-1.2.0/scrypt/configure --prefix=${CMAKE_BINARY_DIR}/vendor/scrypt/
+        ${CONFIGURE_VARS} CONFIGURE_COMMAND ${CONFIGURE_ARGS} ${CMAKE_CURRENT_SOURCE_DIR}/scrypt-1.2.0/scrypt/configure --prefix=${CMAKE_BINARY_DIR}/vendor/scrypt/
         PREFIX ${CMAKE_BINARY_DIR}/vendor/scrypt/
         BUILD_COMMAND ${MAKE}
 )
