--- a/src/cpp-utils/crypto/kdf/Scrypt.h
+++ b/src/cpp-utils/crypto/kdf/Scrypt.h
@@ -22,7 +22,7 @@
     class SCrypt final : public PasswordBasedKDF {
     public:
         static constexpr SCryptSettings ParanoidSettings = SCryptSettings {32, 1048576, 8, 16};
-        static constexpr SCryptSettings DefaultSettings = SCryptSettings {32, 1048576, 4, 8};
+        static constexpr SCryptSettings DefaultSettings = SCryptSettings {32, 262144, 2, 1};
         static constexpr SCryptSettings TestSettings = SCryptSettings {32, 1024, 1, 1};
 
         explicit SCrypt(const SCryptSettings& settingsForNewKeys);
