--- a/src/cpp-utils/crypto/kdf/Scrypt.h
+++ b/src/cpp-utils/crypto/kdf/Scrypt.h
@@ -25,7 +25,7 @@
     class SCrypt final : public PasswordBasedKDF {
     public:
         static constexpr SCryptSettings ParanoidSettings = SCryptSettings {32, 1048576, 8, 16};
-        static constexpr SCryptSettings DefaultSettings = SCryptSettings {32, 1048576, 4, 1};
+        static constexpr SCryptSettings DefaultSettings = SCryptSettings {32, 262144, 2, 1};
         static constexpr SCryptSettings TestSettings = SCryptSettings {32, 1024, 1, 1};
 
         static unique_ref<SCrypt> forNewKey(const SCryptSettings &settings);
