--- a/vendor/cryptopp/vendor_cryptopp/config.h
+++ b/vendor/cryptopp/vendor_cryptopp/config.h
@@ -62,6 +62,13 @@
 // explictly use the ISAs. When disabling ASM, it is best to do it from
 // config.h to ensure the library and all programs share the setting.
 // #define CRYPTOPP_DISABLE_ASM 1
+
+// Configuration option doesn't saved to config.h after compilation, so we have to add
+// following definition manually:
+#ifndef CRYPTOPP_DISABLE_ASM
+#define CRYPTOPP_DISABLE_ASM
+#endif
+
 
 // https://github.com/weidai11/cryptopp/issues/719
 #if defined(__native_client__)
