--- a/Makefile
+++ b/Makefile
@@ -2,14 +2,12 @@
 VERSION=2.6.0
 COMMIT=$(shell git describe --no-match --always --abbrev=7 --dirty)
 VKBD=uinput
-PREFIX?=/usr/local
+PREFIX?=/opt
 
-CONFIG_DIR?=/etc/keyd
-SOCKET_PATH=/var/run/keyd.socket
+CONFIG_DIR?=/opt/etc/keyd
+SOCKET_PATH=/opt/var/run/keyd.socket
 
-CFLAGS:=-DVERSION=\"v$(VERSION)\ \($(COMMIT)\)\" \
-	-I/usr/local/include \
-	-L/usr/local/lib \
+CFLAGS:=-DVERSION=\"v$(VERSION)\" \
 	-Wall \
 	-Wextra \
 	-Wstrict-prototypes \
@@ -21,7 +19,8 @@ CFLAGS:=-DVERSION=\"v$(VERSION)\ \($(COM
 	-D_FORTIFY_SOURCE=2 \
 	-D_DEFAULT_SOURCE \
 	-Werror=format-security \
-	$(CFLAGS)
+	$(CFLAGS) \
+	$(CPPFLAGS)
 
 platform=$(shell uname -s)
 
@@ -49,22 +48,6 @@ man:
 	done
 install:
 
-	@if [ -e /run/systemd/system -o "$(FORCE_SYSTEMD)" ]; then \
-		sed -e 's#@PREFIX@#$(PREFIX)#' keyd.service.in > keyd.service; \
-		mkdir -p $(DESTDIR)$(PREFIX)/lib/systemd/system/; \
-		install -Dm644 keyd.service $(DESTDIR)$(PREFIX)/lib/systemd/system/keyd.service; \
-		mkdir -p $(DESTDIR)$(PREFIX)/lib/sysusers.d/; \
-		install -m644 data/sysusers.d $(DESTDIR)$(PREFIX)/lib/sysusers.d/keyd.conf; \
-	else \
-		echo "NOTE: systemd not found, you will need to manually add keyd to your system's init process."; \
-	fi
-
-	@if [ "$(VKBD)" = "usb-gadget" ]; then \
-		sed -e 's#@PREFIX@#$(PREFIX)#' src/vkbd/usb-gadget.service.in > src/vkbd/usb-gadget.service; \
-		install -Dm644 src/vkbd/usb-gadget.service $(DESTDIR)$(PREFIX)/lib/systemd/system/keyd-usb-gadget.service; \
-		install -Dm755 src/vkbd/usb-gadget.sh $(DESTDIR)$(PREFIX)/bin/keyd-usb-gadget.sh; \
-	fi
-
 	mkdir -p $(DESTDIR)$(CONFIG_DIR)
 	mkdir -p $(DESTDIR)$(PREFIX)/bin/
 	mkdir -p $(DESTDIR)$(PREFIX)/share/keyd/
@@ -73,7 +56,7 @@ install:
 	mkdir -p $(DESTDIR)$(PREFIX)/share/doc/keyd/
 	mkdir -p $(DESTDIR)$(PREFIX)/share/doc/keyd/examples/
 
-	-groupadd keyd
+#	-groupadd keyd
 	install -m755 bin/* $(DESTDIR)$(PREFIX)/bin/
 	install -m644 docs/*.md $(DESTDIR)$(PREFIX)/share/doc/keyd/
 	install -m644 examples/* $(DESTDIR)$(PREFIX)/share/doc/keyd/examples/
