--- a/libgrive/src/drive2/Syncer2.cc
+++ b/libgrive/src/drive2/Syncer2.cc
@@ -17,6 +17,7 @@
 	Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
 
+#include <stdint.h>
 #include "base/Resource.hh"
 #include "CommonUri.hh"
 #include "Entry2.hh"
--- a/libgrive/src/util/File.cc
+++ b/libgrive/src/util/File.cc
@@ -19,6 +19,7 @@
 
 #include "File.hh"
 
+#include <stdint.h>
 #include <cassert>
 
 // boost headers
