--- a/configure.ac
+++ b/configure.ac
@@ -137,9 +137,9 @@ AC_ARG_WITH([build-number],
 #
 # Build environment
 #
-build_person="`id -nu | sed -e 's/\\\\/\\\\\\\\/g'`"
-build_group="`id -ng | sed -e 's/\\\\/\\\\\\\\/g'`"
-build_machine="`uname -n | sed -e 's/\\\\/\\\\\\\\/g'`"
+build_person="entware"
+build_group="entware-team"
+build_machine="entware-generic"
 AC_SUBST([build_machine])
 AC_SUBST([build_person])
 AC_SUBST([build_group])
