--- a/Makefile
+++ b/Makefile
@@ -37,23 +37,23 @@ INCFILES=$(TBLFILES:.tbl=.inc)
 HFILES_INT=$(sort $(wildcard src/*.h)) $(HFILES)
 
 VERSION_CURRENT=0
-VERSION_REVISION=0
-VERSION_AGE=0
+VERSION_REVISION=3
+VERSION_AGE=3
 
 VERSION=0.3.3
 
-PREFIX=/usr/local
+PREFIX=/opt
 BINDIR=$(PREFIX)/bin
 LIBDIR=$(PREFIX)/lib
 INCDIR=$(PREFIX)/include
 MANDIR=$(PREFIX)/share/man
 MAN3DIR=$(MANDIR)/man3
 
-all: $(LIBRARY) $(BINFILES)
+all: $(LIBRARY)
 
 $(LIBRARY): $(OBJECTS)
 	@echo LINK $@
-	@$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath $(LIBDIR) -version-info $(VERSION_CURRENT):$(VERSION_REVISION):$(VERSION_AGE) -o $@ $^ $(LDFLAGS)
+	@$(LIBTOOL) --mode=link --tag=CC $(CC) $(CFLAGS) --version-info $(VERSION_CURRENT):$(VERSION_REVISION):$(VERSION_AGE) -o $@ $^ $(LDFLAGS)
 
 src/%.lo: src/%.c $(HFILES_INT)
 	@echo CC $<
--- a/vterm.pc.in
+++ b/vterm.pc.in
@@ -1,5 +1,6 @@
-libdir=@LIBDIR@
-includedir=@INCDIR@
+prefix=/opt
+includedir=${prefix}/include
+libdir=${prefix}/lib
 
 Name: vterm
 Description: Abstract VT220/Xterm/ECMA-48 emulation library
