--- a/_Projects_/ps3netsrv/Makefile.linux
+++ b/_Projects_/ps3netsrv/Makefile.linux
@@ -1,5 +1,5 @@
 OS =  linux
-BUILD_TYPE = release_static
+BUILD_TYPE = release
 
 OUTPUT := ps3netsrv
 OBJS = src/main.o src/padlock.o src/aes.o src/compat.o src/mem.o src/File.o src/VIsoFile.o
@@ -14,7 +14,7 @@ CPPFLAGS += -Doff64_t=off_t
 #CPPFLAGS +=-DNOSSL
 #OBJS = src/main.o src/compat.o src/mem.o src/File.o src/VIsoFile.o
 
-LDFLAGS = -L.
+LDFLAGS += -L.
 LIBS = -lstdc++
 
 
