--- a/ps3netsrv/Makefile.linux
+++ b/ps3netsrv/Makefile.linux
@@ -1,17 +1,17 @@
-OS =  linux
-BUILD_TYPE = release_static
+OS = linux
+BUILD_TYPE = release
 
 OUTPUT := ps3netsrv
 OBJS = src/main.o src/compat.o src/File.o src/VIsoFile.o
 
-CFLAGS = -Wall -I./include -I./polarssl-1.3.2/include -std=gnu99 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -DPOLARSSL
-CPPFLAGS += -Wall -I./include -I./polarssl-1.3.2/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -DPOLARSSL
+CFLAGS += -Wall -I./include -std=gnu99 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+CPPFLAGS += -Wall -I./include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 
 #CFLAGS += -DNOSSL
 #CPPFLAGS +=-DNOSSL
 
-LDFLAGS = -L. -L./polarssl-1.3.2/library
-LIBS = -lstdc++ -lpolarssl
+LDFLAGS += -L. 
+LIBS = -lstdc++
 
 
 ifeq ($(OS), linux)
