--- a/source/libutil.cc
+++ b/source/libutil.cc
@@ -1056,7 +1056,7 @@ void cscroll(int n, GotoRegion region)
 
 mouse_mode mouse_control::ms_current_mode = MOUSE_MODE_NORMAL;
 
-size_t strlcpy(char *dst, const char *src, size_t n)
+size_t strlcpy(char *dst, const char *src, size_t n) throw ()
 {
     if (!n)
         return strlen(src);
--- a/source/libutil.h
+++ b/source/libutil.h
@@ -72,7 +72,7 @@ static inline ucs_t toalower(ucs_t c)
 
 int numcmp(const char *a, const char *b, int limit = 0);
 bool numcmpstr(string a, string b);
-size_t strlcpy(char *dst, const char *src, size_t n);
+size_t strlcpy(char *dst, const char *src, size_t n) throw();
 
 int strwidth(const char *s);
 int strwidth(const string &s);
