--- ./crawl-ref/source/util/gen_ver.pl.orig	2023-05-25 08:48:48.856457850 +0000
+++ ./crawl-ref/source/util/gen_ver.pl	2023-05-25 10:20:28.213044677 +0000
@@ -20,8 +20,7 @@
 #
 # Source tarbells distributed as part of a release include this file already
 # generated with the release version.
-$_ = `git describe $mergebase 2> /dev/null`
-    || (open(IN, "<", "$scriptpath/release_ver") ? <IN>
+$_ = (open(IN, "<", "$scriptpath/release_ver") ? <IN>
         : die "Can't get version information: `git describe` failed (no git, no repository, or shallow clone), and $scriptpath/release_ver doesn't exist.\n")
     or die "couldn't get the version information\n";
 
