--- a/source/util/gen_ver.pl
+++ b/source/util/gen_ver.pl
@@ -13,19 +13,9 @@
 
 mkdir dirname($outfile);
 
-$_ = `git describe --tags $mergebase 2> /dev/null`
-    || (open(IN, "<", "$scriptpath/release_ver") ? <IN>
-        : die "No Git, and $scriptpath/release_ver doesn't exist.\n")
-    or die "couldn't get the version information\n";
+my ($major, $tag, $pretyp) = ("0.12","0.12.3", "1");
 
-chomp;
-
-/v?(([0-9]+\.[0-9]+)(?:\.[0-9]+)?(?:-([a-zA-Z]+[0-9]+))?)(?:-[0-9]+-g[a-fA-F0-9]+)?/
-    or die "Version string '$_' is malformed.\n";
-
-my ($major, $tag, $pretyp) = ($2, $1, $3);
-
-my $rel = defined($pretyp) ? $pretyp le "b" ? "ALPHA" : "BETA" : "FINAL";
+my $rel = "FINAL";
 
 my $prefix = "CRAWL";
 
@@ -34,6 +24,6 @@
 #define ${prefix}_VERSION_MAJOR "$major"
 #define ${prefix}_VERSION_RELEASE VER_$rel
 #define ${prefix}_VERSION_SHORT "$tag"
-#define ${prefix}_VERSION_LONG "$_"
+#define ${prefix}_VERSION_LONG "$tag"
 __eof__
 close OUT or die $!;
