--- a/src/gnuchess.ini
+++ b/src/gnuchess.ini
@@ -6,16 +6,16 @@ LogFile = adapter.log
 Resign = true
 ResignMoves = 1 # 1 is safe for Fruit
 
-Book = false
+Book = true
 BookFile = smallbook.bin
 
 [Engine]
 
 # standard UCI options
 
-Hash = 256 # in MB
+Hash = 16 # in MB
 
-NalimovCache = 16 # in MB
+NalimovCache = 2 # in MB
 
 OwnBook =  false # false for no book (e.g. Nunn-like matches)
 BookFile = smallbook.bin
