--- a/src/mpu_util.cpp
+++ b/src/mpu_util.cpp
@@ -22,6 +22,8 @@
 #include <cstdlib>
 #include <string>
 
+#include <libxml/parser.h> /* xmlReadMemory */
+
 #include "s3fs_logger.h"
 #include "mpu_util.h"
 #include "curl.h"
--- a/src/s3fs.cpp
+++ b/src/s3fs.cpp
@@ -34,6 +34,7 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <getopt.h>
+#include <libxml/parser.h> /* xmlReadMemory|xmlInitParser|xmlCleanupParser */
 
 #include "common.h"
 #include "s3fs.h"
--- a/src/s3fs_xml.cpp
+++ b/src/s3fs_xml.cpp
@@ -24,6 +24,7 @@
 #include <mutex>
 #include <string>
 
+#include <libxml/parser.h> /* xmlReadMemory */
 #include "common.h"
 #include "s3fs_logger.h"
 #include "s3fs_xml.h"
--- a/src/string_util.cpp
+++ b/src/string_util.cpp
@@ -27,6 +27,8 @@
 #include <string>
 #include <utility>
 
+#include <libxml/parser.h> /* xmlReadMemory */
+
 #include "s3fs_logger.h"
 #include "string_util.h"
 
