--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,8 @@
 
 CC ?= gcc
 CFLAGS ?= -O2 -Wall -Wwrite-strings -pedantic
-CFLAGS += -std=gnu99 -I/usr/local/include
-LFLAGS += -L/usr/local/lib -lc
+#CFLAGS += -std=gnu99 -I/usr/local/include
+#LFLAGS += -L/usr/local/lib -lc
 FEATURES ?= auth cmd lpd nss dns #natpmp upnp debug web
 
 OBJS = build/main.o build/results.o build/kad.o build/log.o \
@@ -87,7 +87,7 @@
 	$(CC) $(CFLAGS) -fPIC -shared -Wl,-soname,libnss_kadnode.so.2 -o build/libnss_kadnode.so.2 build/ext-libnss.o
 
 kadnode-ctl:
-	$(CC) $(CFLAGS) src/kadnode-ctl.c -o build/kadnode-ctl
+	$(CC) $(CFLAGS) src/kadnode-ctl.c -o build/kadnode-ctl $(LFLAGS)
 
 kadnode: $(OBJS) $(EXTRA)
 	$(CC) $(OBJS) -o build/kadnode $(LFLAGS)
