--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,6 @@
 
 CC ?= gcc
 CFLAGS ?= -Os -Wall -Wwrite-strings -pedantic
-CFLAGS += -std=gnu99 -I/usr/local/include
-LFLAGS += -L/usr/local/lib -lc
 FEATURES ?= dns lpd tls bob cmd debug #nss natpmp upnp
 
 OBJS = build/searches.o build/kad.o build/log.o \
@@ -84,7 +82,7 @@
 	$(CC) $(CFLAGS) -c -o $@ $<
 
 kadnode-ctl:
-	$(CC) $(CFLAGS) src/kadnode-ctl.c -o build/kadnode-ctl
+	$(CC) $(CFLAGS) src/kadnode-ctl.c -o build/kadnode-ctl $(LFLAGS)
 
 libnss-kadnode.so.2:
 	$(CC) $(CFLAGS) -fPIC -c -o build/ext-libnss.o src/ext-libnss.c
