--- a/src/stream_sys.c
+++ b/src/stream_sys.c
@@ -951,7 +951,7 @@
 	/* Create buf */
 	thrpt_gettimev(str_hub->thrpt, 0, &tv_now);
 	md5_get_digest_strA((char*)&tv_now, sizeof(tv_now), (char*)hash);
-	snprintf(filename, (sizeof(filename) - 1), "/tmp/msd-%zu-%s.tmp",
+	snprintf(filename, (sizeof(filename) - 1), "/opt/tmp/msd-%zu-%s.tmp",
 	    (size_t)getpid(), hash);
 	str_hub->r_buf_fd = open(filename, (O_CREAT | O_EXCL | O_RDWR), 0600);
 	if ((uintptr_t)-1 == str_hub->r_buf_fd) {
