--- a/meson.build
+++ b/meson.build
@@ -5,9 +5,13 @@
 ### if preferred.
 ### The script tries to compile the GUI using Qt6, with Qt5 5.15 as fallback.
 
-project('highlight-all', 'cpp',
-  meson_version: '>=0.63',
-  default_options : ['cpp_std=c++17', 'warning_level=3', 'optimization=2'])
+project(
+  'highlight-all',
+  'cpp',
+  meson_version : '>=0.63',
+  default_options : ['cpp_std=c++17', 'warning_level=3', 'optimization=2'],
+  version : '4.16'
+)
 
 cmake = import('cmake')
 pkg = import('pkgconfig')
@@ -22,7 +26,7 @@ highlight_conf_dir = conf_dir / 'highlig
 
 highlight_data_dir = data_dir / 'highlight'
 highlight_doc_dir = data_dir / 'doc' / 'highlight'
-gui_files_dir = highlight_data_dir / 'gui_files'
+#gui_files_dir = highlight_data_dir / 'gui_files'
 
 
 # Compiler options
@@ -53,7 +57,7 @@ endif
 # Subdirectories
 core_dir = 'src/core'
 cli_dir = 'src/cli'
-gui_qt_dir = 'src/gui-qt'
+#gui_qt_dir = 'src/gui-qt'
 
 # Source files
 core_objects = files(
@@ -118,44 +122,44 @@ highlight = executable('highlight', cli_
 
 # GUI components
 
-qt_version = 'qt6'
-qt = dependency('qt6', modules : ['Core', 'Gui', 'Widgets'], required: false)
-if not qt.found()
-  qt = dependency('qt5', version : '>=5.15', modules : ['Core', 'Gui', 'Widgets'])
-  qt_version = 'qt5'
-endif
-
-gui_objects = files(
-  'src/gui-qt/main.cpp',
-  'src/gui-qt/mainwindow.cpp',
-  'src/gui-qt/io_report.cpp',
-  'src/gui-qt/showtextfile.cpp',
-  'src/gui-qt/syntax_chooser.cpp'
-)
-
-qt5_mod = import(qt_version)
-processed = qt5_mod.preprocess(
-  moc_headers : ['src/gui-qt/mainwindow.h', 'src/gui-qt/io_report.h', 'src/gui-qt/showtextfile.h', 'src/gui-qt/syntax_chooser.h'],   # Only headers that need moc should be put here
-  # moc_sources : 'helperFile.cpp', # must have #include"moc_helperFile.cpp"
-  ui_files    : ['src/gui-qt/mainwindow.ui', 'src/gui-qt/io_report.ui', 'src/gui-qt/showtextfile.ui', 'src/gui-qt/syntax_chooser.ui'],
-  qresources  : 'src/gui-qt/highlight-gui.qrc',
-)
-
-gui_exe = executable('highlight-gui',
-  sources: [gui_objects, processed],
-  include_directories: inc,
-  dependencies: [boost_dep, lua_dep, qt],
-  link_with: highlight_lib,
-  install: true
-)
+#qt_version = 'qt6'
+#qt = dependency('qt6', modules : ['Core', 'Gui', 'Widgets'], required: false)
+#if not qt.found()
+#  qt = dependency('qt5', version : '>=5.15', modules : ['Core', 'Gui', 'Widgets'])
+#  qt_version = 'qt5'
+#endif
+
+#gui_objects = files(
+#  'src/gui-qt/main.cpp',
+#  'src/gui-qt/mainwindow.cpp',
+#  'src/gui-qt/io_report.cpp',
+#  'src/gui-qt/showtextfile.cpp',
+#  'src/gui-qt/syntax_chooser.cpp'
+#)
+
+#qt5_mod = import(qt_version)
+#processed = qt5_mod.preprocess(
+#  moc_headers : ['src/gui-qt/mainwindow.h', 'src/gui-qt/io_report.h', 'src/gui-qt/showtextfile.h', 'src/gui-qt/syntax_chooser.h'],   # Only headers that need moc should be put here
+#  # moc_sources : 'helperFile.cpp', # must have #include"moc_helperFile.cpp"
+#  ui_files    : ['src/gui-qt/mainwindow.ui', 'src/gui-qt/io_report.ui', 'src/gui-qt/showtextfile.ui', 'src/gui-qt/syntax_chooser.ui'],
+#  qresources  : 'src/gui-qt/highlight-gui.qrc',
+#)
+
+#gui_exe = executable('highlight-gui',
+#  sources: [gui_objects, processed],
+#  include_directories: inc,
+#  dependencies: [boost_dep, lua_dep, qt],
+#  link_with: highlight_lib,
+#  install: true
+#)
 
 
 bash_comp_dir = data_dir / 'bash-completion' / 'completions'
 fish_comp_dir = data_dir / 'fish' / 'vendor_completions.d'
 zsh_comp_dir = data_dir / 'zsh' / 'site-functions'
 
-desktop_apps_dir = data_dir / 'applications'
-desktop_icons_dir = data_dir / 'icons' / 'hicolor' / '256x256' / 'apps'
+#desktop_apps_dir = data_dir / 'applications'
+#desktop_icons_dir = data_dir / 'icons' / 'hicolor' / '256x256' / 'apps'
 
 # Install files and directories
 install_data('filetypes.conf', install_dir: highlight_conf_dir)
@@ -177,9 +181,9 @@ install_data(
   rename: '_highlight',
   install_dir: zsh_comp_dir,
 )
-install_data('highlight.desktop', install_dir: desktop_apps_dir)
-install_data('src/gui-qt/highlight.png', install_dir: desktop_icons_dir)
-install_data('gui_files/ext/fileopenfilter.conf', install_dir: gui_files_dir / 'ext')
+#install_data('highlight.desktop', install_dir: desktop_apps_dir)
+#install_data('src/gui-qt/highlight.png', install_dir: desktop_icons_dir)
+#install_data('gui_files/ext/fileopenfilter.conf', install_dir: gui_files_dir / 'ext')
 #install_subdir('gui_files/l10n', install_dir: join_paths(gui_files_dir, 'l10n'), files: '*.qm')
 
 # Platform-specific configurations
