--- a/minissdpd/minissdpd.c
+++ b/minissdpd/minissdpd.c
@@ -1246,9 +1246,9 @@
 	int max_fd;
 	struct lan_addr_s * lan_addr;
 	int i;
-	const char * sockpath = "/var/run/minissdpd.sock";
+	const char * sockpath = "/opt/var/run/minissdpd.sock";
 #ifndef NO_BACKGROUND_NO_PIDFILE
-	const char * pidfilename = "/var/run/minissdpd.pid";
+	const char * pidfilename = "/opt/var/run/minissdpd.pid";
 #endif
 	int debug_flag = 0;
 #ifdef ENABLE_IPV6
--- a/minissdpd/showminissdpdnotif.c
+++ b/minissdpd/showminissdpdnotif.c
@@ -31,7 +31,7 @@
 	int i;
 	int s;
 	struct sockaddr_un addr;
-	const char * sockpath = "/var/run/minissdpd.sock";
+	const char * sockpath = "/opt/var/run/minissdpd.sock";
 	unsigned char buffer[4096];
 	ssize_t n;
 	const char command5[] = { 0x05, 0x00 };
--- a/minissdpd/testminissdpd.c
+++ b/minissdpd/testminissdpd.c
@@ -74,7 +74,7 @@
 	unsigned char buf[4096];
 	ssize_t n;
 	int total = 0;
-	const char * sockpath = "/var/run/minissdpd.sock";
+	const char * sockpath = "/opt/var/run/minissdpd.sock";
 
 	for(i=0; i<argc-1; i++) {
 		if(0==strcmp(argv[i], "-s"))
