--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 #
 #
-COMPILE = i686
+COMPILE = mipsel
 
 VERSION = 0.4
 
@@ -48,8 +48,19 @@
 LDFLAGS = -lusb -lhid -lsqlite3 -lpthread
 endif
 
+ifeq ($(COMPILE),mipsel)
+##
+## for MIPSEL Linux
+##
+DDD = compile for MIPSEL Linux
+#TOOLCHAINPATH=/usr
+#CC      = gcc
+#LIBS	= -L/usr/local/lib -L-L/usr/lib
+#CFLAGS  = -I/usr/local/include -I/usr/include
+LDFLAGS += -lusb -lhid -lsqlite3 -lpthread -lm
+endif
+
 
-include Makefile.inc
 
 OBJS		=	src/wmr.o src/wmr_conf.o src/wmr_syslog.o src/wmr_util.o src/wmr_sensor.o
 OBJSLIBRRD	=	src/wmr_rrdtool.o src/rrdupdate/rrd_update.o src/rrdupdate/rrd_misc.o src/rrdupdate/rrd_generic.o
@@ -66,7 +77,7 @@
 # If you add a new OSTYPE here please email it to me so that I can add
 # it to the distribution in the next release
 # -----------------------------------------------------------------------
-SYSTYPE := $(shell uname -s)
+SYSTYPE := Linux
 
 ifeq ($(SYSTYPE), Linux)
   CFLAGS += -DLINUX
@@ -117,17 +128,13 @@
 
 prog:		$(OBJS)
 		@echo '#define DATA_VERSION "'Den68 ASCII/SQL/RRD/SNMP - building: `date` '"' >src/wmr_build.h
-		$(CC) $(OBJS) -o wmrd $(LIBS) ${LDFLAGS} -Wl,-rpath,/usr/lib/wmr -L. -lwmr-alarm -lwmr-rrd -lwmr-sql -lwmr-file -lwmr-upd
+		$(CC) $(OBJS) -o wmrd $(LIBS) ${LDFLAGS} -Wl,-rpath,/opt/lib/wmr -L. -lwmr-alarm -lwmr-rrd -lwmr-sql -lwmr-file -lwmr-upd
 		@rm -f $(OBJS)
-		@${TOOLCHAINPATH}/bin/strip wmrd
-		@./goftp.sh
 
 single:         $(OBJS) $(OBJSLIBUPD) $(OBJSLIBSNMP) $(OBJSLIBFILE) $(OBJSLIBSQL) $(OBJSLIBRRD) $(OBJSLIBALARM)
 		@echo '#define DATA_VERSION "'Den68 ASCII/SQL/RRD/SNMP - building: `date` '"' >src/wmr_build.h
 		$(CC) $(OBJS) $(OBJSLIBUPD) $(OBJSLIBSNMP) $(OBJSLIBFILE) $(OBJSLIBSQL) $(OBJSLIBRRD) $(OBJSLIBALARM) -o wmrd $(LIBS) ${LDFLAGS}
 		@rm -f $(OBJS) $(OBJSLIBUPD) $(OBJSLIBSNMP) $(OBJSLIBFILE) $(OBJSLIBSQL) $(OBJSLIBRRD) $(OBJSLIBALARM)
-		@${TOOLCHAINPATH}/bin/strip wmrd
-		@./goftp.sh
 
 shm:		$(OBJSSHM)
 		$(CC) $(OBJSSHM) -o shm
