--- a/Makefile.linux
+++ b/Makefile.linux
@@ -29,18 +29,18 @@
 
 ifeq ($(USE_STATIC),yes)
   LIBDIR := /usr/lib
-  LDLIBS  = $(LIBDIR)/libboost_system.a
-  LDLIBS += $(LIBDIR)/libboost_date_time.a
-  LDLIBS += $(LIBDIR)/libboost_filesystem.a
-  LDLIBS += $(LIBDIR)/libboost_regex.a
-  LDLIBS += $(LIBDIR)/libboost_program_options.a
+  LDLIBS  = $(LIBDIR)/libboost_system-mt.a
+  LDLIBS += $(LIBDIR)/libboost_date_time-mt.a
+  LDLIBS += $(LIBDIR)/libboost_filesystem-mt.a
+  LDLIBS += $(LIBDIR)/libboost_regex-mt.a
+  LDLIBS += $(LIBDIR)/libboost_program_options-mt.a
   LDLIBS += $(LIBDIR)/libcrypto.a
   LDLIBS += $(LIBDIR)/libssl.a
   LDLIBS += $(LIBDIR)/libz.a
   LDLIBS += -lpthread -static-libstdc++ -static-libgcc
   USE_AESNI := no
 else
-  LDLIBS = -lcrypto -lssl -lz -lboost_system -lboost_date_time -lboost_filesystem -lboost_regex -lboost_program_options -lpthread
+  LDLIBS = -lcrypto -lssl -lz -lboost_system-mt -lboost_date_time-mt -lboost_filesystem-mt -lboost_regex-mt -lboost_program_options-mt -lpthread
 endif
 
 # UPNP Support (miniupnpc 1.5 or 1.6)
