--- a/FS.cpp
+++ b/FS.cpp
@@ -65,11 +65,11 @@
 #endif	  
     char *home = getenv("HOME");
     if (isService) {
-      dataDir = "/var/lib/" + appName;
+      dataDir = "/opt/var/lib/" + appName;
     } else if (home != NULL && strlen(home) > 0) {
       dataDir = std::string(home) + "/." + appName;
     } else {
-      dataDir = "/tmp/" + appName;
+      dataDir = "/opt/tmp/" + appName;
     }
     return;
 #endif
