--- a/lib/unix-lpr.sh
+++ b/lib/unix-lpr.sh
@@ -19,16 +19,16 @@
 # executable name set in the makefile
 GS_EXECUTABLE=gs
 
-PBMPLUSPATH=/usr/local/bin
-PSFILTERPATH=/usr/local/lib/ghostscript
-LOCALPATH=/usr/local/bin
-X11HOME=/usr/X11R6
+PBMPLUSPATH=/opt/bin
+PSFILTERPATH=/opt/lib/ghostscript
+LOCALPATH=/opt/bin
+X11HOME=/opt/usr/X11R6
 
 PATH=/bin:/usr/bin:/usr/ucb:/usr/etc
 PATH=${PATH}\:${LOCALPATH}\:${PBMPLUSPATH}\:${PSFILTERPATH}
-LD_LIBRARY_PATH=${X11HOME}/lib
+#LD_LIBRARY_PATH=${X11HOME}/lib
 
-export PATH LD_LIBRARY_PATH acctfile host user
+#export PATH LD_LIBRARY_PATH acctfile host user
 
 user= host= acctfile=/dev/null
 
--- a/lib/lprsetup.sh
+++ b/lib/lprsetup.sh
@@ -14,7 +14,7 @@
             "usage: $0 [-p <gs install prefix>]\n"\
             "Where <gs install prefix> is the ""prefix""\n"\
             "for the Ghostscript install - the default is\n"\
-            """/usr/local"""
+            """/opt"""
             exit 1;;
 	-*) echo >&2 \
 	    "usage: $0 [-p <gs install prefix>]"
@@ -33,14 +33,10 @@
 # The kind of printer (accepted values: 'parallel' and 'serial')
 PRINTERTYPE=parallel
 
-if [ "x$GSPATH"="x" ] ; then
-  GSDIR=`which gs | awk -F / 'sub(FS $NF,x)' | awk -F / 'sub(FS $NF,x)'`/lib/ghostscript
-else
-  GSDIR=$GSPATH
-fi
+GSDIR=/opt/lib/ghostscript
 
 GSFILTERDIR=$GSDIR/filt
-SPOOLDIR=/var/spool
+SPOOLDIR=/opt/var/spool
 GSIF=unix-lpr.sh
 PCAP=printcap.insert
 
@@ -55,9 +51,6 @@
   echo
 fi
 
-PATH=/bin:/usr/bin:/usr/ucb
-export PATH
-
 if [ ! -w $GSDIR ]; then
   echo "$GSDIR must be writable to create filter directory"
   exit 1
