--- a/configure.ac
+++ b/configure.ac
@@ -314,23 +314,23 @@
 AC_MSG_CHECKING(directory to use for temporary files)
 if test -n "$enable_tmpdir"; then
 	tmpdir="$enable_tmpdir"
-elif test -n "$TMPDIR"; then
-	tmpdir="$TMPDIR"
-elif test -n "$TMP"; then
-	tmpdir="$TMP"
-elif test -n "$TEMP"; then
-	tmpdir="$TEMP"
-elif test -d "c:/"; then
-	tmpdir="c:/"
-else
-	tmpdir="/tmp"
+#elif test -n "$TMPDIR"; then
+#	tmpdir="$TMPDIR"
+#elif test -n "$TMP"; then
+#	tmpdir="$TMP"
+#elif test -n "$TEMP"; then
+#	tmpdir="$TEMP"
+#elif test -d "c:/"; then
+#	tmpdir="c:/"
+#else
+#	tmpdir="/tmp"
 fi
-if test -d $tmpdir ; then
+#if test -d $tmpdir ; then
 	AC_MSG_RESULT($tmpdir)
 	AC_DEFINE_UNQUOTED(TMPDIR, "$tmpdir")
-else
-	AC_MSG_ERROR($tmpdir does not exist)
-fi
+#else
+#	AC_MSG_ERROR($tmpdir does not exist)
+#fi
 
 # Test for case-insensitive filenames
 AC_MSG_CHECKING(for case-insensitive filenames)
@@ -388,38 +388,6 @@
 
 AC_CHECK_HEADERS([stdbool.h])
 
-AC_MSG_CHECKING(if "%zu" in printf works)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <stdio.h>
-int main(void) {
-	static char buffer[10];
-	size_t i = 999;
-	sprintf(buffer, "%zu", i);
-	return !(buffer[0] == '9' && buffer[1] == '9' && buffer[2] == '9');
-}
-]])],[have_zu_formatter=yes],[have_zu_formatter=no])
-AC_MSG_RESULT($have_zu_formatter)
-
-if test x$have_zu_formatter != xyes; then
-	AC_CHECK_SIZEOF(size_t)
-	AC_CHECK_SIZEOF(unsigned int)
-	AC_CHECK_SIZEOF(unsigned long)
-	AC_CHECK_SIZEOF(unsigned long long)
-	AC_MSG_CHECKING(alternative for "%zu")
-	if test x"$ac_cv_sizeof_size_t" = x"$ac_cv_sizeof_unsigned_int"; then
-		use_no_formatter_for_size_t=yes
-		AC_MSG_RESULT("%u")
-	elif test x"$ac_cv_sizeof_size_t" = x"$ac_cv_sizeof_unsigned_long"; then
-		use_ulong_formatter_for_size_t=yes
-		AC_MSG_RESULT("%lu")
-	elif test x"$ac_cv_sizeof_size_t" = x"$ac_cv_sizeof_unsigned_long_long"; then
-		use_ulonglong_formatter_for_size_t=yes
-		AC_MSG_RESULT("%llu")
-	else
-		AC_MSG_ERROR(no alternative for "%zu" formatter)
-	fi
-fi
-
 # Checks for compiler characteristics
 # -----------------------------------
 
--- a/source.mak
+++ b/source.mak
@@ -364,11 +364,4 @@
 	$(NULL)
 READTAGS_HEADS = read/readtags.h
 READTAGS_OBJS  = $(READTAGS_SRCS:.c=.$(OBJEXT))
-
-PACKCC_SRCS = \
-	misc/packcc/packcc.c \
-	\
-	$(NULL)
-
-PACKCC_OBJS = $(PACKCC_SRCS:.c=.$(OBJEXT))
 # vim: ts=8
