--- a/configure.ac
+++ b/configure.ac
@@ -419,23 +419,23 @@ AC_CHECK_PROGS([PYTHON],
 AC_MSG_CHECKING(directory to use for temporary files)
 if test -n "$enable_tmpdir"; then
 	tmpdir="$enable_tmpdir"
-elif test -n "$TMPDIR"; then
-	tmpdir="$TMPDIR"
-elif test -n "$TMP"; then
-	tmpdir="$TMP"
-elif test -n "$TEMP"; then
-	tmpdir="$TEMP"
-elif test -d "c:/"; then
-	tmpdir="c:/"
-else
-	tmpdir="/tmp"
+#elif test -n "$TMPDIR"; then
+#	tmpdir="$TMPDIR"
+#elif test -n "$TMP"; then
+#	tmpdir="$TMP"
+#elif test -n "$TEMP"; then
+#	tmpdir="$TEMP"
+#elif test -d "c:/"; then
+#	tmpdir="c:/"
+#else
+#	tmpdir="/tmp"
 fi
-if test -d $tmpdir ; then
+#if test -d $tmpdir ; then
 	AC_MSG_RESULT($tmpdir)
 	AC_DEFINE_UNQUOTED(TMPDIR, "$tmpdir")
-else
-	AC_MSG_ERROR($tmpdir does not exist)
-fi
+#else
+#	AC_MSG_ERROR($tmpdir does not exist)
+#fi
 
 # Test for case-insensitive filenames
 AC_MSG_CHECKING(for case-insensitive filenames)
