--- a/C/CpuArch.h
+++ b/C/CpuArch.h
@@ -82,7 +82,7 @@ MY_CPU_64BIT means that processor can wo
   || defined(__ARMEB__) \
   || defined(__THUMBEL__) \
   || defined(__THUMBEB__)
-  #define MY_CPU_ARM
+  #define MY_CPU_ARM1
 
   #if defined(__thumb__) || defined(__THUMBEL__) || defined(_M_ARMT)
     #define MY_CPU_ARMT
@@ -217,7 +217,7 @@ MY_CPU_64BIT means that processor can wo
 #endif
 
 
-#if defined(MY_CPU_X86) || defined(MY_CPU_AMD64)
+#if defined(MY_CPU_X861) || defined(MY_CPU_AMD64)
 #define MY_CPU_X86_OR_AMD64
 #endif
 
