--- a/src/int/file_magic.c
+++ b/src/int/file_magic.c
@@ -290,6 +290,7 @@ get_handlers(const char mime_type[])
 	}
 
 #if !defined(_WIN32) && defined(ENABLE_DESKTOP_FILES)
+	parse_app_dir("/opt/share/applications", mime_type, &handlers);
 	parse_app_dir("/usr/share/applications", mime_type, &handlers);
 	parse_app_dir("/usr/local/share/applications", mime_type, &handlers);
 
--- a/src/lua/lua/luaconf.h
+++ b/src/lua/lua/luaconf.h
@@ -223,7 +223,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"/opt/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
--- a/src/ui/color_scheme.c
+++ b/src/ui/color_scheme.c
@@ -741,7 +741,7 @@ cs_write(void)
 	fprintf(fp, "\" Vifm supports 256 colors you can use color numbers 0-255\n");
 	fprintf(fp, "\" (requires properly set up terminal: set your TERM environment variable\n");
 	fprintf(fp, "\" (directly or using resources) to some color terminal name (e.g.\n");
-	fprintf(fp, "\" xterm-256color) from /usr/lib/terminfo/; you can check current number\n");
+	fprintf(fp, "\" xterm-256color) from /opt/lib/terminfo/; you can check current number\n");
 	fprintf(fp, "\" of colors in your terminal with tput colors command)\n\n");
 
 	fprintf(fp, "\" highlight group cterm=attrs ctermfg=foreground_color ctermbg=background_color\n\n");
