--- a/include/ddns.h
+++ b/include/ddns.h
@@ -39,8 +39,8 @@
 #define SUPPORT_ADDR	PACKAGE_BUGREPORT
 
 /* Test values */
-#define DEFAULT_CONFIG_FILE	"/etc/inadyn.conf"
-#define RUNTIME_DATA_DIR        _PATH_VARRUN     "inadyn/"
+#define DEFAULT_CONFIG_FILE	"/opt/etc/inadyn.conf"
+#define RUNTIME_DATA_DIR        "/opt/var/run/inadyn/"
 #define DEFAULT_PIDFILE         RUNTIME_DATA_DIR "inadyn.pid"
 
 #define DYNDNS_MY_IP_SERVER	"checkip.dyndns.org"
