--- a/include/ddns.h
+++ b/include/ddns.h
@@ -40,8 +40,8 @@
 
 /* Test values */
 /* Some default configurations */
-#define DEFAULT_CONFIG_FILE	"/etc/inadyn.conf"
-#define DEFAULT_CACHE_DIR       "/var/cache/inadyn/"
+#define DEFAULT_CONFIG_FILE	"/opt/etc/inadyn.conf"
+#define DEFAULT_CACHE_DIR       "/opt/var/cache/inadyn/"
 
 #define DYNDNS_MY_IP_SERVER	"checkip.dyndns.org"
 #define DYNDNS_MY_CHECKIP_URL	"/"
