Apparently 60ms is too much for some cards.

--- a/src/libespeak-ng/speech.c
+++ b/src/libespeak-ng/speech.c
@@ -278,7 +278,7 @@ ESPEAK_NG_API espeak_ng_STATUS espeak_ng
 
 	// buffer_length is in mS, allocate 2 bytes per sample
 	if (buffer_length == 0)
-		buffer_length = 60;
+		buffer_length = 49;
 
 	outbuf_size = (buffer_length * samplerate)/500;
 	out_start = (unsigned char *)realloc(outbuf, outbuf_size);
