--- a/configure.ac
+++ b/configure.ac
@@ -96,7 +96,7 @@ AC_SUBST(GRADLE)
 dnl ================================================================
 dnl C99 compiler checks.
 dnl ================================================================
-
+AC_SUBST(HOST_BUILD_DIR)
 AC_LANG_PUSH(C)
 AC_MSG_CHECKING([if $CC supports C99 without any flags])
 AC_COMPILE_IFELSE(
--- a/Makefile.am
+++ b/Makefile.am
@@ -397,8 +397,8 @@ EXTRA_DIST += $(filter-out phsource/phon
 phsource/phonemes.stamp: \
 	$(PHSOURCE) \
 	src/espeak-ng
-	ESPEAK_DATA_PATH=$(CURDIR) src/espeak-ng --compile-intonations && \
-		ESPEAK_DATA_PATH=$(CURDIR) src/espeak-ng --compile-phonemes && \
+	ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile-intonations && \
+		ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile-phonemes && \
 		touch $@
 
 ##### android targets:
@@ -438,10 +438,10 @@ dictsource/%_extra:
 # (at the cost of a spurious error message.)
 #
 # NOTE: keep in sync with 'espeak-ng-data/yue_dict' more specific rule.
-espeak-ng-data/%_dict: src/espeak-ng phsource/phonemes.stamp
+espeak-ng-data/%_dict: $(HOST_BUILD_DIR)/src/espeak-ng phsource/phonemes.stamp
 	@echo "  DICT      $@"
 	rm -f $@
-	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) ../src/espeak-ng --compile="$(*F)"
+	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile="$(*F)"
 
 dictionaries: \
 	espeak-ng-data/af_dict \
@@ -952,13 +952,13 @@ dictsource/yue_listx:
 
 # NOTE: keep in sync with espeak-ng-data/%_dict more generic rule.
 if HAVE_YUE_EXTENDED_DICTIONARY
-espeak-ng-data/yue_dict: src/espeak-ng phsource/phonemes.stamp dictsource/yue_list dictsource/yue_rules dictsource/yue_extra dictsource/yue_listx dictsource/yue_emoji
+espeak-ng-data/yue_dict: $(HOST_BUILD_DIR)/src/espeak-ng phsource/phonemes.stamp dictsource/yue_list dictsource/yue_rules dictsource/yue_extra dictsource/yue_listx dictsource/yue_emoji
 else
-espeak-ng-data/yue_dict: src/espeak-ng phsource/phonemes.stamp dictsource/yue_list dictsource/yue_rules dictsource/yue_extra dictsource/yue_emoji
+espeak-ng-data/yue_dict: $(HOST_BUILD_DIR)/src/espeak-ng phsource/phonemes.stamp dictsource/yue_list dictsource/yue_rules dictsource/yue_extra dictsource/yue_emoji
 endif
 	@echo "  DICT      $@"
 	rm -f $@
-	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) LD_LIBRARY_PATH=../src:${LD_LIBRARY_PATH} ../src/espeak-ng --compile=yue && cd ..
+	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile=yue && cd ..
 
 dictsource/yue_emoji:
 	@echo "  EMOJI     $@"
@@ -978,9 +978,9 @@ PHSOURCE_MBROLA = \
 mbrola: \
 	$(patsubst phsource/mbrola/%,espeak-ng-data/mbrola_ph/%_phtrans,$(PHSOURCE_MBROLA))
 
-espeak-ng-data/mbrola_ph/%_phtrans: phsource/mbrola/% src/espeak-ng
+espeak-ng-data/mbrola_ph/%_phtrans: phsource/mbrola/% $(HOST_BUILD_DIR)/src/espeak-ng
 	mkdir -p espeak-ng-data/mbrola_ph
-	ESPEAK_DATA_PATH=$(CURDIR) src/espeak-ng --compile-mbrola=phsource/mbrola/$*
+	ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile-mbrola=phsource/mbrola/$*
 
 EXTRA_DIST += $(PHSOURCE_MBROLA)
 
