--- a/build/rules.mk
+++ b/build/rules.mk
@@ -1,14 +1,14 @@
-AR      = ar
-CC      = gcc
-CPP     = g++
-STRIP   = strip
-RM      = rm
+AR      += ar
+CC      += gcc
+CPP     = $(CXX)
+STRIP   += strip
+RM      += rm
 OBJS    = main.o common.o ssdp.o http.o soap.o soap_int.o db_sqlite.o scan.o mime.o charset.o scripting.o live.o md5.o luajson.o \
  compat.o plugin_hls_common.o plugin_hls.o plugin_hls_new.o plugin_tsbuf.o plugin_lua.o plugin_udprtp.o plugin_tsfilter.o
 SLIBS   = sqlite3/libsqlite3.a lua-5.1.5/liblua.a
 
 all: version $(SLIBS) $(OBJS)
-	$(CPP) $(CFLAGS) -o xupnpd $(OBJS) $(SLIBS) $(LIBS)
+	$(CPP) $(CFLAGS) -o xupnpd $(OBJS) $(LDFLAGS) $(SLIBS) $(LIBS)
 	$(STRIP) xupnpd
 
 sqlite3/libsqlite3.a:
