--- a/goo/Makefile.in
+++ b/goo/Makefile.in
@@ -54,7 +54,7 @@
 
 $(LIBPREFIX)Goo.a: $(GOO_OBJS)
 	rm -f $(LIBPREFIX)Goo.a
-	$(AR) $(LIBPREFIX)Goo.a $(GOO_OBJS)
+	$(AR) q $(LIBPREFIX)Goo.a $(GOO_OBJS)
 	$(RANLIB) $(LIBPREFIX)Goo.a
 
 #------------------------------------------------------------------------
--- a/fofi/Makefile.in
+++ b/fofi/Makefile.in
@@ -55,7 +55,7 @@
 
 $(LIBPREFIX)fofi.a: $(FOFI_OBJS)
 	rm -f $(LIBPREFIX)fofi.a
-	$(AR) $(LIBPREFIX)fofi.a $(FOFI_OBJS)
+	$(AR) q $(LIBPREFIX)fofi.a $(FOFI_OBJS)
 	$(RANLIB) $(LIBPREFIX)fofi.a
 
 #------------------------------------------------------------------------
--- a/splash/Makefile.in
+++ b/splash/Makefile.in
@@ -77,7 +77,7 @@
 
 $(LIBPREFIX)splash.a: $(SPLASH_OBJS)
 	rm -f $(LIBPREFIX)splash.a
-	$(AR) $(LIBPREFIX)splash.a $(SPLASH_OBJS)
+	$(AR) q $(LIBPREFIX)splash.a $(SPLASH_OBJS)
 	$(RANLIB) $(LIBPREFIX)splash.a
 
 #------------------------------------------------------------------------
