--- a/splash/Splash.cc
+++ b/splash/Splash.cc
@@ -2887,7 +2887,7 @@
 	 splashAbs(path->pts[1].y - path->pts[2].y) < 0.001)))) {
     wx = state->matrix[0] + state->matrix[2];
     wy = state->matrix[1] + state->matrix[3];
-    w = sqrt(wx*wx + wy*wy);
+    w = sqrt((double)(wx*wx + wy*wy));
     if (w < 0.001) {
       w = 0;
     } else {
@@ -2905,7 +2905,7 @@
     }
     dx = xx1 - xx0;
     dy = yy1 - yy0;
-    d = sqrt(dx * dx + dy * dy);
+    d = sqrt((double)(dx * dx + dy * dy));
     if (d < 0.001) {
       d = 0;
     } else {
