--- a/libUseful/Makefile.in
+++ b/libUseful/Makefile.in
@@ -2,7 +2,7 @@ CC = @CC@
 VERSION = 4.11
 CFLAGS = @CFLAGS@ @SIMD_CFLAGS@
 LDFLAGS=@LDFLAGS@
-LIBS = @LIBS@
+LIBS = -lcrypto -lssl -lc
 FLAGS=$(LDFLAGS) $(CPPFLAGS) $(CFLAGS) -fPIC @DEFS@ -DHAVE_MADVISE -DHAVE_MADVISE_NOFORK -DHAVE_MADVISE_DONTDUMP -DHAVE_MLOCK -DVERSION=\"$(VERSION)\"
 prefix=@prefix@
 OBJ=String.o List.o Socket.o UnixSocket.o Stream.o Errors.o Unicode.o Terminal.o FileSystem.o GeneralFunctions.o DataProcessing.o Pty.o Log.o Http.o Smtp.o inet.o Expect.o base64.o  crc32.o md5c.o sha1.o sha2.o whirlpool.o jh_ref.o Hash.o Ssh.o Compression.o OAuth.o LibSettings.o Vars.o Time.o Markup.o SpawnPrograms.o Tokenizer.o PatternMatch.o URL.o DataParser.o ConnectionChain.o OpenSSL.o Process.o Encodings.o RawData.o SecureMem.o CommandLineParser.o SysInfo.o
--- a/Makefile.in
+++ b/Makefile.in
@@ -8,14 +8,14 @@ exec_prefix=@exec_prefix@
 bindir=@sbindir@
 sysconfdir=@sysconfdir@
 FLAGS=$(CFLAGS) $(CPPFLAGS) @DEFS@ 
-OBJ=auth_access_token.o Authenticate.o auth_client_certificate.o auth_alaya_native.o auth_unix.o auth_pam.o MimeType.o DavProps.o Settings.o common.o server.o cgi.o FileProperties.o tar.o directory_listing.o FileDetailsPage.o VPath.o ChrootHelper.o UserAdminScreen.o Events.o ID3.o upload.o proxy.o websocket.o icecast.o xssi.o @STATIC_LIBS@
+OBJ=auth_access_token.o Authenticate.o auth_client_certificate.o auth_alaya_native.o auth_unix.o auth_pam.o MimeType.o DavProps.o Settings.o common.o server.o cgi.o FileProperties.o tar.o directory_listing.o FileDetailsPage.o VPath.o ChrootHelper.o UserAdminScreen.o Events.o ID3.o upload.o proxy.o websocket.o icecast.o xssi.o
 EXE=alaya
 
-all: $(OBJ)
+all: libUseful/libUseful.a $(OBJ)
 	$(CC) $(FLAGS) -o $(EXE) $(OBJ) main.c $(LIBS) 
 
 libUseful/libUseful.a: 
-	@cd libUseful; $(MAKE)
+	cd libUseful; $(MAKE)
 
 auth_access_token.o: auth_access_token.c auth_access_token.h
 	$(CC) $(FLAGS) -c auth_access_token.c
