--- a/cryptoplugin/src/config.h
+++ b/cryptoplugin/src/config.h
@@ -580,7 +580,7 @@
 #if !defined(CRYPTOPP_ARM_NEON_AVAILABLE) && !defined(CRYPTOPP_DISABLE_ASM)
 # if defined(__ARM_NEON) || defined(__ARM_NEON_FP) || defined(__ARM_FEATURE_NEON) || \
 	(__ARM_ARCH >= 7) || (CRYPTOPP_MSC_VERSION >= 1700)
-#  define CRYPTOPP_ARM_NEON_AVAILABLE 1
+#  define CRYPTOPP_ARM_NEON_AVAILABLE 0
 # endif
 #endif
 
