--- a/configure.ac
+++ b/configure.ac
@@ -15,7 +15,7 @@ AC_CONFIG_AUX_DIR([autotools])
 AC_CONFIG_MACRO_DIR([autotools])
 
 # Init automake
-AM_INIT_AUTOMAKE([1.12.0 foreign subdir-objects serial-tests])
+AM_INIT_AUTOMAKE([foreign subdir-objects])
 AM_MAINTAINER_MODE
 AB_INIT
 
@@ -95,8 +95,6 @@ AC_CHECK_HEADERS([inttypes.h limits.h me
 
 # Check OpenSSL
 AX_CHECK_OPENSSL([], AC_MSG_FAILURE([OpenSSL not found]))
-# Define if X509_TEA_set_state exists
-AX_CHECK_X509(AC_DEFINE([HAVE_X509_TEA_SET_STATE],[1],[Define if X509_set_state exists]), [])
 # Default PAM install dir
 case "$host" in
      *darwin*)         PAM_DIR="/usr/lib/pam" ;;
