--- a/Makefile.am
+++ b/Makefile.am
@@ -80,13 +80,13 @@ install-exec-local:
 	$(INSTALL) -d $(DESTDIR)/${COROSYSCONFDIR}/service.d
 	$(INSTALL) -d $(DESTDIR)/${COROSYSCONFDIR}/uidgid.d
 	$(INSTALL) -d $(DESTDIR)/${localstatedir}/lib/corosync
-	$(INSTALL) -d $(DESTDIR)/${localstatedir}/log/cluster
+	$(INSTALL) -d $(DESTDIR)/${localstatedir}/log/corosync
 
 uninstall-local:
 	rmdir $(DESTDIR)/${COROSYSCONFDIR}/service.d || :;
 	rmdir $(DESTDIR)/${COROSYSCONFDIR}/uidgid.d || :;
 	rmdir $(DESTDIR)/${localstatedir}/lib/corosync || :;
-	rmdir $(DESTDIR)/${localstatedir}/log/cluster || :;
+	rmdir $(DESTDIR)/${localstatedir}/log/corosync || :;
 
 if AUGTOOL
 check_SCRIPTS = test_lense.sh
