--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,7 @@ endif()
 find_package(Jansson 2.2 REQUIRED)
 include_directories(${JANSSON_INCLUDE_DIRS})
 
-add_library(mfapi STATIC
+add_library(mfapi SHARED
 	mfapi/mfconn.c
 	mfapi/file.c
 	mfapi/account.c
@@ -71,7 +71,7 @@ add_library(mfapi STATIC
 	mfapi/apicalls/upload_poll_upload.c
 	)
 
-add_library(mfutils STATIC
+add_library(mfutils SHARED
 	utils/http.c
 	utils/strings.c
 	utils/stringv.c
@@ -155,6 +155,7 @@ add_test(valgrind_fuse ${CMAKE_SOURCE_DI
 add_test(valgrind_shell ${CMAKE_SOURCE_DIR}/tests/valgrind_shell.sh ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR})
 
 install (TARGETS mediafire-fuse mediafire-shell DESTINATION bin)
+install (TARGETS mfapi mfutils DESTINATION lib)
 
 SET(CPACK_GENERATOR "DEB")
 
