--- a/utils/config.c
+++ b/utils/config.c
@@ -41,35 +41,10 @@
 
 void config_file_init(char **configfile)
 {
-    const char     *homedir;
     const char     *configdir;
     int             fd;
 
-    homedir = getenv("HOME");
-    if (homedir == NULL) {
-        homedir = getpwuid(getuid())->pw_dir;
-    }
-
-    configdir = getenv("XDG_CONFIG_HOME");
-    if (configdir == NULL) {
-        configdir = strdup_printf("%s/.config", homedir);
-        /* EEXIST is okay, so only fail if it is something else */
-        if (mkdir(configdir, 0755) != 0 && errno != EEXIST) {
-            perror("mkdir");
-            fprintf(stderr, "cannot create %s\n", configdir);
-            exit(1);
-        }
-        free((void *)configdir);
-        configdir = strdup_printf("%s/.config/mediafire-tools", homedir);
-    } else {
-        // $XDG_CONFIG_HOME/mediafire-tools
-        if (mkdir(configdir, 0755) != 0 && errno != EEXIST) {
-            perror("mkdir");
-            fprintf(stderr, "cannot create %s\n", configdir);
-            exit(1);
-        }
-        configdir = strdup_printf("%s/mediafire-tools", configdir);
-    }
+    configdir = strdup_printf("%s/mediafire-tools", "/opt/etc");
     /* EEXIST is okay, so only fail if it is something else */
     if (mkdir(configdir, 0755) != 0 && errno != EEXIST) {
         perror("mkdir");
