--- a/configure	2012-05-08 20:06:41.094926851 +0400
+++ b/configure	2012-05-08 20:08:23.302511280 +0400
@@ -208,36 +208,7 @@
 	emulate sh
 fi
 
-rm -f systest systest.exe
-echo $n "compiling test program... $c"
-make -fsystest.mk >/dev/null
-if [ $? -ne 0 ]
-then make -f systest.mk >/dev/null
-fi
-if [ $? -eq 0 ]; then
-	if [ -f systest.exe ]
-	then SYSTEST_OUT=`./systest.exe`
-	else SYSTEST_OUT=`./systest`
-	fi
-	if [ $? -ne 0 ]; then SYSTEST_OUT= ; fi
-else
-	SYSTEST_OUT=
-fi
-if [ -n "$SYSTEST_OUT" ]; then
-	echo ok
-else
-	echo FAILED!
-	# don't give the annoying message if these are set :-)
-	if [ -n "$CC$CFLAGS$CPPFLAGS" ]
-	then exit 1
-	fi
-	echo 'Are you sure you have "make" and a C compiler installed?'
-	echo "Or are the standard header files missing?  (If you're using a Debian-based"
-	echo 'distribution, there should be a package called "build-essential").'
-	echo 'Try to run "make -fsystest.mk".'
-	echo When that works, run configure again.
-	exit 1
-fi
+SYSTEST_OUT=linux
 
 OS=${SYSTEST_OUT%% _*}
 echo "operating system: $OS"
