--- a/src/image.c
+++ b/src/image.c
@@ -75,6 +75,10 @@
 #include "jp2tools.h"
 #endif
 
+#ifndef GIFLIB_MAJOR
+#define GIFLIB_MAJOR 0
+#endif
+
 #define SRC_IMAGE_C
 #include "image.h"
 #include "cfgfile.h"
@@ -904,10 +908,13 @@
 	desc.buf=inbuf;
 	desc.size=insize;
 	desc.x.pos=0;
-
+#if GIFLIB_MAJOR < 5
 	if ((GifFile = DGifOpen((void*)&desc, &gif_mem_input)) == NULL) 
     		return( IMG_RET_ERR_UNKNOWN + IMG_RET_FLG_WHILE_DECOMP);//more possible reasons
-
+#else
+	if ((GifFile = DGifOpen((void*)&desc, &gif_mem_input, NULL)) == NULL) 
+    		return( IMG_RET_ERR_UNKNOWN + IMG_RET_FLG_WHILE_DECOMP);//more possible reasons
+#endif
 	bmp = new_raw_bitmap();
 	*out = bmp;
 	bmp->width=GifFile->SWidth;
@@ -1083,7 +1090,11 @@
 	}
 	bmp->pal_bpp = bmp->bpp;
 	
+#if GIFLIB_MAJOR < 5
 	DGifCloseFile(GifFile);
+#else
+	DGifCloseFile(GifFile, NULL);
+#endif
 	return IMG_RET_OK;
 }
 
