--- a/src/main.c
+++ b/src/main.c
@@ -157,7 +157,7 @@
 bool
 get_package_file(const char *name, char **outname)
 {
-    *outname = xasprintf("%s/.%s/%s", getenv_default("HOME", ""), PACKAGE, name);
+    *outname = xasprintf("%s/%s/%s", "/opt/share", PACKAGE, name);
     return true;
 }
 
--- a/src/common/tempdir.c
+++ b/src/common/tempdir.c
@@ -53,7 +53,7 @@
     if (tmpdir != NULL && direxists(tmpdir))
         return tmpdir;
 
-    tmpdir = "/tmp";
+    tmpdir = "/opt/tmp";
     if (direxists(tmpdir))
         return tmpdir;
 
