From: Pawel Sulowicz <Pawel.Sulowicz@grupaonet.pl>
Date: Thu, 2 Feb 2017 09:39:37 +0100
Subject: [PATCH] Fix compatibility with nginx-1.11.6+
Origin: other, https://github.com/FRiCKLE/ngx_cache_purge/pull/51

---
 ngx_cache_purge_module.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ngx_cache_purge_module.c b/ngx_cache_purge_module.c
index 62d3818..fd1ebde 100644
--- a/ngx_cache_purge_module.c
+++ b/ngx_cache_purge_module.c
@@ -492,7 +492,11 @@ typedef struct {
     ngx_str_t                      body_source;
 #  endif /* nginx_version < 1007008 */
 
+#  if (nginx_version >= 1011006)
+    ngx_http_complex_value_t      *method;
+#  else
     ngx_str_t                      method;
+#  endif /* nginx_version >= 1011006 */
     ngx_str_t                      location;
     ngx_str_t                      url;
 
