--- a/GNUmakefile
+++ b/GNUmakefile
@@ -2,7 +2,7 @@
 BINDIR?= $(PREFIX)/bin
 MANDIR?= $(PREFIX)/share/man
 
-CFLAGS+= -O2 -Wall -ggdb -D_GNU_SOURCE -D_BSD_SOURCE -I.
+CFLAGS+= -Wall -ggdb -D_GNU_SOURCE -D_BSD_SOURCE -I.
 LDFLAGS+=
 
 LDADD+= -lldns -levent
@@ -10,7 +10,7 @@
 CC?= cc
 
 adsuck: adsuck.o log.o linux/strlcpy.o resolv.o
-	$(CC) $(LDFLAGS) -o $@ $+ $(LDADD)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $+ $(LDADD)
 
 clean:
 	rm -f adsuck *.o linux/*.o
