--- a/daemon/remote.c
+++ b/daemon/remote.c
@@ -1659,6 +1659,7 @@ processResponse( const char * rpcurl, co
     else
     {
         int64_t      tag = -1;
+	int          itag;
         const char * str;
 
         if(tr_bencDictFindStr(&top, "result", &str))
@@ -1672,7 +1673,8 @@ processResponse( const char * rpcurl, co
             {
         tr_bencDictFindInt( &top, "tag", &tag );
 
-        switch( tag )
+	itag = tag;
+	switch( itag )
         {
             case TAG_SESSION:
                 printSession( &top ); break;
