--- a/libtransmission/fdlimit.c
+++ b/libtransmission/fdlimit.c
@@ -25,11 +25,6 @@
  #include <fcntl.h>
 #endif
 
-#ifdef HAVE_FALLOCATE64
-  /* FIXME can't find the right #include voodoo to pick up the declaration.. */
-  extern int fallocate64( int fd, int mode, uint64_t offset, uint64_t len );
-#endif
-
 #ifdef HAVE_XFS_XFS_H
  #include <xfs/xfs.h>
 #endif
@@ -76,11 +76,6 @@
     if( !length )
         success = true;
 
-#ifdef HAVE_FALLOCATE64
-    if( !success ) /* fallocate64 is always preferred, so try it first */
-        success = !fallocate64( fd, 0, 0, length );
-#endif
-
     if( !success ) /* fallback: the old-style seek-and-write */
         success = ( lseek( fd, length-1, SEEK_SET ) != -1 )
                && ( write( fd, &zero, 1 ) != -1 )
@@ -111,12 +106,6 @@
     int fd = open( filename, flags, 0666 );
     if( fd >= 0 )
     {
-# ifdef HAVE_FALLOCATE64
-       if( !success )
-       {
-           success = !fallocate64( fd, 0, 0, length );
-       }
-# endif
 # ifdef HAVE_XFS_XFS_H
         if( !success && platform_test_xfs_fd( fd ) )
         {
