--- a/configure
+++ b/configure
@@ -5020,18 +5020,18 @@
   if test x$ncurses_exec_prefix != x ; then
      ncurses_args="$ncurses_args --exec-prefix=$ncurses_exec_prefix"
      if test x${NCURSES_CONFIG+set} != xset ; then
-        NCURSES_CONFIG=$ncurses_exec_prefix/bin/ncursesw5-config
+        NCURSES_CONFIG=$ncurses_exec_prefix/bin/ncursesw6-config
      fi
   fi
   if test x$ncurses_prefix != x ; then
      ncurses_args="$ncurses_args --prefix=$ncurses_prefix"
      if test x${NCURSES_CONFIG+set} != xset ; then
-        NCURSES_CONFIG=$ncurses_prefix/bin/ncursesw5-config
+        NCURSES_CONFIG=$ncurses_prefix/bin/ncursesw6-config
      fi
   fi
 
-  # Extract the first word of "ncursesw5-config", so it can be a program name with args.
-set dummy ncursesw5-config; ac_word=$2
+  # Extract the first word of "ncursesw6-config", so it can be a program name with args.
+set dummy ncursesw6-config; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
 if ${ac_cv_path_NCURSES_CONFIG+:} false; then :
@@ -5124,10 +5124,10 @@
      { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
      if test "$NCURSES_CONFIG" = "no" ; then
-       echo "*** The ncursesw5-config script installed by ncursesw could not be found"
+       echo "*** The ncursesw6-config script installed by ncursesw could not be found"
        echo "*** If ncursesw was installed in PREFIX, make sure PREFIX/bin is in"
        echo "*** your path, or set the NCURSES_CONFIG environment variable to the"
-       echo "*** full path to ncursesw5-config."
+       echo "*** full path to ncursesw6-config."
 	 else
 	   if test -f conf.ncursestest ; then
         :
@@ -5162,7 +5162,7 @@
    echo "*** The test program failed to compile or link. See the file config.log for the"
           echo "*** exact error that occured. This usually means ncursesw was incorrectly"
           echo "*** installed or that you have moved ncursesw since it was installed. In the"
-          echo "*** latter case, you may want to edit the ncursesw5-config script:"
+          echo "*** latter case, you may want to edit the ncursesw6-config script:"
           echo "*** $NCURSES_CONFIG"
 fi
 rm -f core conftest.err conftest.$ac_objext \
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -324,7 +324,7 @@
 dnl
 AC_DEFUN([AM_PATH_NCURSESW],
 [dnl 
-dnl Get the cflags and libraries from the ncursesw5-config script
+dnl Get the cflags and libraries from the ncursesw6-config script
 dnl
 AC_ARG_WITH(ncurses-prefix,[  --with-ncurses-prefix=PFX   Prefix where ncurses is installed (optional)],
             ncurses_prefix="$withval", ncurses_prefix="")
@@ -336,17 +336,17 @@
   if test x$ncurses_exec_prefix != x ; then
      ncurses_args="$ncurses_args --exec-prefix=$ncurses_exec_prefix"
      if test x${NCURSES_CONFIG+set} != xset ; then
-        NCURSES_CONFIG=$ncurses_exec_prefix/bin/ncursesw5-config
+        NCURSES_CONFIG=$ncurses_exec_prefix/bin/ncursesw6-config
      fi
   fi
   if test x$ncurses_prefix != x ; then
      ncurses_args="$ncurses_args --prefix=$ncurses_prefix"
      if test x${NCURSES_CONFIG+set} != xset ; then
-        NCURSES_CONFIG=$ncurses_prefix/bin/ncursesw5-config
+        NCURSES_CONFIG=$ncurses_prefix/bin/ncursesw6-config
      fi
   fi
 
-  AC_PATH_PROG(NCURSES_CONFIG, ncursesw5-config, no)
+  AC_PATH_PROG(NCURSES_CONFIG, ncursesw6-config, no)
   AC_MSG_CHECKING(for ncurses - wide char support)
   no_ncurses=""
   if test "$NCURSES_CONFIG" = "no" ; then
@@ -361,7 +361,7 @@
     LIBS="$LIBS $NCURSES_LIBS"
 dnl
 dnl Now check if the installed ncurses is installed OK. (Also sanity
-dnl checks the results of ncursesw5-config to some extent)
+dnl checks the results of ncursesw6-config to some extent)
 dnl
     rm -f conf.ncursestest
     AC_TRY_RUN([
@@ -386,10 +386,10 @@
   else
      AC_MSG_RESULT(no)
      if test "$NCURSES_CONFIG" = "no" ; then
-       echo "*** The ncursesw5-config script installed by ncursesw could not be found"
+       echo "*** The ncursesw6-config script installed by ncursesw could not be found"
        echo "*** If ncursesw was installed in PREFIX, make sure PREFIX/bin is in"
        echo "*** your path, or set the NCURSES_CONFIG environment variable to the"
-       echo "*** full path to ncursesw5-config."
+       echo "*** full path to ncursesw6-config."
 	 else
 	   if test -f conf.ncursestest ; then
         :
@@ -412,7 +412,7 @@
         [ echo "*** The test program failed to compile or link. See the file config.log for the"
           echo "*** exact error that occured. This usually means ncursesw was incorrectly"
           echo "*** installed or that you have moved ncursesw since it was installed. In the"
-          echo "*** latter case, you may want to edit the ncursesw5-config script:"
+          echo "*** latter case, you may want to edit the ncursesw6-config script:"
           echo "*** $NCURSES_CONFIG" ])
           CFLAGS="$ac_save_CFLAGS"
           LIBS="$ac_save_LIBS"
