--- a/config.h
+++ b/config.h
@@ -412,6 +412,18 @@ NAMESPACE_END
 #define CRYPTOPP_UNCAUGHT_EXCEPTION_AVAILABLE
 #endif
 
+// Configuration option doesn't saved to config.h after compilation, so we have to add
+// following definition manually:
+#ifndef CRYPTOPP_DISABLE_ASM
+#define CRYPTOPP_DISABLE_ASM
+#endif
+#ifndef CRYPTOPP_DISABLE_SSSE3
+#define CRYPTOPP_DISABLE_SSSE3
+#endif
+#ifndef CRYPTOPP_DISABLE_AESNI
+#define CRYPTOPP_DISABLE_AESNI
+#endif
+
 #ifdef CRYPTOPP_DISABLE_X86ASM		// for backwards compatibility: this macro had both meanings
 #define CRYPTOPP_DISABLE_ASM
 #define CRYPTOPP_DISABLE_SSE2
