--- a/oauth.py
+++ b/oauth.py
@@ -4 +4 @@
-import httplib
+import httplib_TLSv1
@@ -94 +94 @@
-        conn = httplib.HTTPSConnection(url)
+        conn = httplib_TLSv1.HTTPSConnection(url)
--- a/lib/thrift/transport/THttpClient.py
+++ b/lib/thrift/transport/THttpClient.py
@@ -24 +24 @@
-import httplib
+import httplib_TLSv1
@@ -52 +52 @@
-        self.port = parsed.port or httplib.HTTP_PORT
+        self.port = parsed.port or httplib_TLSv1.HTTP_PORT
@@ -54 +54 @@
-        self.port = parsed.port or httplib.HTTPS_PORT
+        self.port = parsed.port or httplib_TLSv1.HTTPS_PORT
@@ -65 +65 @@
-      self.__http = httplib.HTTP(self.host, self.port)
+      self.__http = httplib_TLSv1.HTTP(self.host, self.port)
@@ -67 +67 @@
-      self.__http = httplib.HTTPS(self.host, self.port)
+      self.__http = httplib_TLSv1.HTTPS(self.host, self.port)
