--- a/src/runtime/os_linux32.go
+++ b/src/runtime/os_linux32.go
@@ -14,21 +14,11 @@ import (
 //go:noescape
 func futex_time32(addr unsafe.Pointer, op int32, val uint32, ts *timespec32, addr2 unsafe.Pointer, val3 uint32) int32
 
-//go:noescape
-func futex_time64(addr unsafe.Pointer, op int32, val uint32, ts *timespec, addr2 unsafe.Pointer, val3 uint32) int32
 
 var isFutexTime32bitOnly atomic.Bool
 
 //go:nosplit
 func futex(addr unsafe.Pointer, op int32, val uint32, ts *timespec, addr2 unsafe.Pointer, val3 uint32) int32 {
-	if !isFutexTime32bitOnly.Load() {
-		ret := futex_time64(addr, op, val, ts, addr2, val3)
-		// futex_time64 is only supported on Linux 5.0+
-		if ret != -_ENOSYS {
-			return ret
-		}
-		isFutexTime32bitOnly.Store(true)
-	}
 	// Downgrade ts.
 	var ts32 timespec32
 	var pts32 *timespec32
@@ -42,21 +32,10 @@ func futex(addr unsafe.Pointer, op int32
 //go:noescape
 func timer_settime32(timerid int32, flags int32, new, old *itimerspec32) int32
 
-//go:noescape
-func timer_settime64(timerid int32, flags int32, new, old *itimerspec) int32
-
 var isSetTime32bitOnly atomic.Bool
 
 //go:nosplit
 func timer_settime(timerid int32, flags int32, new, old *itimerspec) int32 {
-	if !isSetTime32bitOnly.Load() {
-		ret := timer_settime64(timerid, flags, new, old)
-		// timer_settime64 is only supported on Linux 5.0+
-		if ret != -_ENOSYS {
-			return ret
-		}
-		isSetTime32bitOnly.Store(true)
-	}
 
 	var newts, oldts itimerspec32
 	var new32, old32 *itimerspec32
--- a/src/runtime/sys_linux_386.s
+++ b/src/runtime/sys_linux_386.s
@@ -48,13 +48,11 @@
 #define SYS_madvise		219
 #define SYS_gettid		224
 #define SYS_futex		240
-#define SYS_futex_time64	422
 #define SYS_sched_getaffinity	242
 #define SYS_set_thread_area	243
 #define SYS_exit_group		252
 #define SYS_timer_create	259
 #define SYS_timer_settime	260
-#define SYS_timer_settime64	409
 #define SYS_timer_delete	263
 #define SYS_clock_gettime	265
 #define SYS_tgkill		270
@@ -210,7 +208,6 @@ TEXT runtime·timer_create(SB),NOSPLIT,$
 	MOVL	AX, ret+12(FP)
 	RET
 
-// Linux: kernel/time/posix-timer.c, requiring COMPAT_32BIT_TIME
 TEXT runtime·timer_settime32(SB),NOSPLIT,$0-20
 	MOVL	$SYS_timer_settime, AX
 	MOVL	timerid+0(FP), BX
@@ -221,16 +218,6 @@ TEXT runtime·timer_settime32(SB),NOSPLI
 	MOVL	AX, ret+16(FP)
 	RET
 
-TEXT runtime·timer_settime64(SB),NOSPLIT,$0-20
-	MOVL	$SYS_timer_settime64, AX
-	MOVL	timerid+0(FP), BX
-	MOVL	flags+4(FP), CX
-	MOVL	new+8(FP), DX
-	MOVL	old+12(FP), SI
-	INVOKE_SYSCALL
-	MOVL	AX, ret+16(FP)
-	RET
-
 TEXT runtime·timer_delete(SB),NOSPLIT,$0-8
 	MOVL	$SYS_timer_delete, AX
 	MOVL	timerid+0(FP), BX
@@ -545,7 +532,6 @@ TEXT runtime·madvise(SB),NOSPLIT,$0
 	MOVL	AX, ret+12(FP)
 	RET
 
-// Linux: kernel/futex/syscalls.c, requiring COMPAT_32BIT_TIME
 // int32 futex(int32 *uaddr, int32 op, int32 val,
 //	struct old_timespec32 *timeout, int32 *uaddr2, int32 val2);
 TEXT runtime·futex_time32(SB),NOSPLIT,$0
@@ -553,21 +539,6 @@ TEXT runtime·futex_time32(SB),NOSPLIT,$
 	MOVL	addr+0(FP), BX
 	MOVL	op+4(FP), CX
 	MOVL	val+8(FP), DX
-	MOVL	ts+12(FP), SI
-	MOVL	addr2+16(FP), DI
-	MOVL	val3+20(FP), BP
-	INVOKE_SYSCALL
-	MOVL	AX, ret+24(FP)
-	RET
-
-// Linux: kernel/futex/syscalls.c
-// int32 futex(int32 *uaddr, int32 op, int32 val,
-//	struct timespec *timeout, int32 *uaddr2, int32 val2);
-TEXT runtime·futex_time64(SB),NOSPLIT,$0
-	MOVL	$SYS_futex_time64, AX
-	MOVL	addr+0(FP), BX
-	MOVL	op+4(FP), CX
-	MOVL	val+8(FP), DX
 	MOVL	ts+12(FP), SI
 	MOVL	addr2+16(FP), DI
 	MOVL	val3+20(FP), BP
--- a/src/runtime/sys_linux_arm.s
+++ b/src/runtime/sys_linux_arm.s
@@ -30,7 +30,6 @@
 #define SYS_sigaltstack (SYS_BASE + 186)
 #define SYS_mmap2 (SYS_BASE + 192)
 #define SYS_futex (SYS_BASE + 240)
-#define SYS_futex_time64 (SYS_BASE + 422)
 #define SYS_exit_group (SYS_BASE + 248)
 #define SYS_munmap (SYS_BASE + 91)
 #define SYS_madvise (SYS_BASE + 220)
@@ -44,7 +43,6 @@
 #define SYS_clock_gettime (SYS_BASE + 263)
 #define SYS_timer_create (SYS_BASE + 257)
 #define SYS_timer_settime (SYS_BASE + 258)
-#define SYS_timer_settime64 (SYS_BASE + 409)
 #define SYS_timer_delete (SYS_BASE + 261)
 #define SYS_pipe2 (SYS_BASE + 359)
 #define SYS_access (SYS_BASE + 33)
@@ -232,7 +230,7 @@ TEXT runtime·timer_create(SB),NOSPLIT,$
 	SWI	$0
 	MOVW	R0, ret+12(FP)
 	RET
-// Linux: kernel/time/posix-timer.c, requiring COMPAT_32BIT_TIME.
+
 TEXT runtime·timer_settime32(SB),NOSPLIT,$0-20
 	MOVW	timerid+0(FP), R0
 	MOVW	flags+4(FP), R1
@@ -243,16 +241,6 @@ TEXT runtime·timer_settime32(SB),NOSPLI
 	MOVW	R0, ret+16(FP)
 	RET
 
-TEXT runtime·timer_settime64(SB),NOSPLIT,$0-20
-	MOVW	timerid+0(FP), R0
-	MOVW	flags+4(FP), R1
-	MOVW	new+8(FP), R2
-	MOVW	old+12(FP), R3
-	MOVW	$SYS_timer_settime64, R7
-	SWI	$0
-	MOVW	R0, ret+16(FP)
-	RET
-
 TEXT runtime·timer_delete(SB),NOSPLIT,$0-8
 	MOVW	timerid+0(FP), R0
 	MOVW	$SYS_timer_delete, R7
@@ -415,7 +403,6 @@ finish:
 
 	RET
 
-// Linux: kernel/futex/syscalls.c, requiring COMPAT_32BIT_TIME
 // int32 futex(int32 *uaddr, int32 op, int32 val,
 //	struct old_timespec32 *timeout, int32 *uaddr2, int32 val2);
 TEXT runtime·futex_time32(SB),NOSPLIT,$0
@@ -429,21 +416,6 @@ TEXT runtime·futex_time32(SB),NOSPLIT,$
 	SWI	$0
 	MOVW	R0, ret+24(FP)
 	RET
-
-// Linux: kernel/futex/syscalls.c
-// int32 futex(int32 *uaddr, int32 op, int32 val,
-//	struct timespec *timeout, int32 *uaddr2, int32 val2);
-TEXT runtime·futex_time64(SB),NOSPLIT,$0
-	MOVW    addr+0(FP), R0
-	MOVW    op+4(FP), R1
-	MOVW    val+8(FP), R2
-	MOVW    ts+12(FP), R3
-	MOVW    addr2+16(FP), R4
-	MOVW    val3+20(FP), R5
-	MOVW	$SYS_futex_time64, R7
-	SWI	$0
-	MOVW	R0, ret+24(FP)
-	RET
 
 // int32 clone(int32 flags, void *stack, M *mp, G *gp, void (*fn)(void));
 TEXT runtime·clone(SB),NOSPLIT,$0
--- a/src/runtime/sys_linux_mipsx.s
+++ b/src/runtime/sys_linux_mipsx.s
@@ -34,12 +34,10 @@
 #define SYS_mincore		4217
 #define SYS_gettid		4222
 #define SYS_futex		4238
-#define SYS_futex_time64	4422
 #define SYS_sched_getaffinity	4240
 #define SYS_exit_group		4246
 #define SYS_timer_create	4257
 #define SYS_timer_settime	4258
-#define SYS_timer_settime64	4409
 #define SYS_timer_delete	4261
 #define SYS_clock_gettime	4263
 #define SYS_tgkill		4266
@@ -198,7 +196,6 @@ TEXT runtime·timer_create(SB),NOSPLIT,$
 	MOVW	R2, ret+12(FP)
 	RET
 
-// Linux: kernel/time/posix-timer.c, requiring COMPAT_32BIT_TIME
 TEXT runtime·timer_settime32(SB),NOSPLIT,$0-20
 	MOVW	timerid+0(FP), R4
 	MOVW	flags+4(FP), R5
@@ -209,16 +206,6 @@ TEXT runtime·timer_settime32(SB),NOSPLI
 	MOVW	R2, ret+16(FP)
 	RET
 
-TEXT runtime·timer_settime64(SB),NOSPLIT,$0-20
-	MOVW	timerid+0(FP), R4
-	MOVW	flags+4(FP), R5
-	MOVW	new+8(FP), R6
-	MOVW	old+12(FP), R7
-	MOVW	$SYS_timer_settime64, R2
-	SYSCALL
-	MOVW	R2, ret+16(FP)
-	RET
-
 TEXT runtime·timer_delete(SB),NOSPLIT,$0-8
 	MOVW	timerid+0(FP), R4
 	MOVW	$SYS_timer_delete, R2
@@ -375,7 +362,6 @@ TEXT runtime·madvise(SB),NOSPLIT,$0-16
 	MOVW	R2, ret+12(FP)
 	RET
 
-// Linux: kernel/futex/syscalls.c, requiring COMPAT_32BIT_TIME
 // int32 futex(int32 *uaddr, int32 op, int32 val,
 //	struct old_timespec32 *timeout, int32 *uaddr2, int32 val2);
 TEXT runtime·futex_time32(SB),NOSPLIT,$20-28
@@ -394,28 +380,6 @@ TEXT runtime·futex_time32(SB),NOSPLIT,$
 	SYSCALL
 	BEQ	R7, 2(PC)
 	SUBU	R2, R0, R2	// caller expects negative errno
-	MOVW	R2, ret+24(FP)
-	RET
-
-// Linux: kernel/futex/syscalls.c
-// int32 futex(int32 *uaddr, int32 op, int32 val,
-//	struct timespec *timeout, int32 *uaddr2, int32 val2);
-TEXT runtime·futex_time64(SB),NOSPLIT,$20-28
-	MOVW	addr+0(FP), R4
-	MOVW	op+4(FP), R5
-	MOVW	val+8(FP), R6
-	MOVW	ts+12(FP), R7
-
-	MOVW	addr2+16(FP), R8
-	MOVW	val3+20(FP), R9
-
-	MOVW	R8, 16(R29)
-	MOVW	R9, 20(R29)
-
-	MOVW	$SYS_futex_time64, R2
-	SYSCALL
-	BEQ	R7, 2(PC)
-	SUBU	R2, R0, R2	// caller expects negative errno
 	MOVW	R2, ret+24(FP)
 	RET
 
