--- a/mount.nfusr.in
+++ b/mount.nfusr.in
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 NFUSR="@bindir@/nfusr"
 
@@ -12,13 +12,16 @@
 # Parse the 'host' command line argument, return a set of NFS hosts in $HOSTS global.
 function get_hosts () {
     local HOST=$1
-    if [[ "$HOST" =~ ^nfs://.* ]]; then
+    case "$HOST" in
+    nfs://*)
       HOSTS=$(echo "$HOST" | sed 's/,/ /g')
-    else
+      ;;
+    *)
       echo "Invalid host specifier: $HOST."
       usage
       return 1
-    fi
+      ;;
+    esac
 
     return 0
 }
