--- a/Makefile.am
+++ b/Makefile.am
@@ -12,9 +12,6 @@ dist_noinst_DATA =                 \
   resources/js/d3.v3.min.js        \
   resources/js/hogan.min.js
 
-noinst_PROGRAMS = bin2c
-bin2c_SOURCES = src/bin2c.c
-
 BUILT_SOURCES =       \
   src/tpls.h          \
   src/bootstrapcss.h  \
@@ -43,6 +40,9 @@ CLEANFILES =                          \
   resources/js/charts.js.tmp          \
   resources/js/app.js.tmp
 
+bin2c$(EXEEXT):
+	true
+
 # Tpls
 src/tpls.h: bin2c$(EXEEXT) $(srcdir)/resources/tpls.html
 if HAS_SEDTR
