--- a/Makefile.am
+++ b/Makefile.am
@@ -12,9 +12,6 @@
   resources/js/d3.v3.min.js       \
   resources/js/hogan.min.js
 
-noinst_PROGRAMS = bin2c
-bin2c_SOURCES = src/bin2c.c
-
 BUILT_SOURCES =   \
   tpls.h          \
   bootstrapcss.h  \
@@ -44,7 +41,7 @@
   resources/js/app.js.tmp
 
 # Tpls
-tpls.h: bin2c$(EXEEXT) resources/tpls.html
+tpls.h: resources/tpls.html
 if HAS_SEDTR
 	cat resources/tpls.html | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/tpls.html.tmp
 	./bin2c resources/tpls.html.tmp src/tpls.h tpls
@@ -52,7 +49,7 @@
 	./bin2c resources/tpls.html src/tpls.h tpls
 endif
 # Bootstrap
-bootstrapcss.h: bin2c$(EXEEXT) resources/css/bootstrap.min.css
+bootstrapcss.h: resources/css/bootstrap.min.css
 if HAS_SEDTR
 	cat resources/css/bootstrap.min.css | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/css/bootstrap.min.css.tmp
 	./bin2c resources/css/bootstrap.min.css.tmp src/bootstrapcss.h bootstrap_css
@@ -60,7 +57,7 @@
 	./bin2c resources/css/bootstrap.min.css src/bootstrapcss.h bootstrap_css
 endif
 # Font Awesome
-facss.h: bin2c$(EXEEXT) resources/css/fa.min.css
+facss.h: resources/css/fa.min.css
 if HAS_SEDTR
 	cat resources/css/fa.min.css | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/css/fa.min.css.tmp
 	./bin2c resources/css/fa.min.css.tmp src/facss.h fa_css
@@ -68,7 +65,7 @@
 	./bin2c resources/css/fa.min.css src/facss.h fa_css
 endif
 # App.css
-appcss.h: bin2c$(EXEEXT) resources/css/app.css
+appcss.h: resources/css/app.css
 if HAS_SEDTR
 	cat resources/css/app.css | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/css/app.css.tmp
 	./bin2c resources/css/app.css.tmp src/appcss.h app_css
@@ -76,7 +73,7 @@
 	./bin2c resources/css/app.css src/appcss.h app_css
 endif
 # D3.js
-d3js.h: bin2c$(EXEEXT) resources/js/d3.v3.min.js
+d3js.h: resources/js/d3.v3.min.js
 if HAS_SEDTR
 	cat resources/js/d3.v3.min.js | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/js/d3.v3.min.js.tmp
 	./bin2c resources/js/d3.v3.min.js.tmp src/d3js.h d3_js
@@ -84,7 +81,7 @@
 	./bin2c resources/js/d3.v3.min.js src/d3js.h d3_js
 endif
 # Hogan.js
-hoganjs.h: bin2c$(EXEEXT) resources/js/hogan.min.js
+hoganjs.h: resources/js/hogan.min.js
 if HAS_SEDTR
 	cat resources/js/hogan.min.js | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/js/hogan.min.js.tmp
 	./bin2c resources/js/hogan.min.js.tmp src/hoganjs.h hogan_js
@@ -92,7 +89,7 @@
 	./bin2c resources/js/hogan.min.js src/hoganjs.h hogan_js
 endif
 # Charts.js
-chartsjs.h: bin2c$(EXEEXT) resources/js/charts.js
+chartsjs.h: resources/js/charts.js
 if HAS_SEDTR
 	cat resources/js/charts.js | sed -E "s@(,|;)[[:space:]]*//..*@\1@g" | sed -E "s@^[[:space:]]*//..*@@g" | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/js/charts.js.tmp
 	./bin2c resources/js/charts.js.tmp src/chartsjs.h charts_js
@@ -100,7 +97,7 @@
 	./bin2c resources/js/charts.js src/chartsjs.h charts_js
 endif
 # App.js
-appjs.h: bin2c$(EXEEXT) resources/js/app.js
+appjs.h: resources/js/app.js
 if HAS_SEDTR
 	cat resources/js/app.js | sed -E "s@(,|;)[[:space:]]*//..*@\1@g" | sed -E "s@^[[:space:]]*//..*@@g" | sed "s/^[[:space:]]*//" | sed "/^$$/d" | tr -d "\r\n" > resources/js/app.js.tmp
 	./bin2c resources/js/app.js.tmp src/appjs.h app_js
