--- a/src/options.c
+++ b/src/options.c
@@ -256,10 +256,10 @@ cmd_help (void)
   "  --no-ip-validation              - Disable client IPv4/6  validation.\n"
   "  --no-strict-status              - Disable HTTP status code validation.\n"
   "  --num-tests=<number>            - Number of lines to test. >= 0 (10 default)\n"
-  "  --persist                       - Persist data to disk on exit to the given --db-path or to /tmp.\n"
+  "  --persist                       - Persist data to disk on exit to the given --db-path or to /opt/tmp.\n"
   "  --process-and-exit              - Parse log and exit without outputting data.\n"
   "  --real-os                       - Display real OS names. e.g, Windows XP, Snow Leopard.\n"
-  "  --restore                       - Restore data from disk from the given --db-path or from /tmp.\n"
+  "  --restore                       - Restore data from disk from the given --db-path or from /opt/tmp.\n"
   "  --sort-panel=PANEL,METRIC,ORDER - Sort panel on initial load. e.g., --sort-panel=VISITORS,BY_HITS,ASC.\n"
   "                                    See manpage for a list of panels/fields.\n"
   "  --static-file=<extension>       - Add static file extension. e.g.: .mp3. Extensions are case sensitive.\n"
--- a/src/gstorage.h
+++ b/src/gstorage.h
@@ -35,7 +35,7 @@
 
 /* Total number of storage metrics (GSMetric) */
 #define GSMTRC_TOTAL 19
-#define DB_PATH "/tmp"
+#define DB_PATH "/opt/tmp"
 
 /* Enumerated Storage Metrics */
 typedef enum GSMetric_ {
