--- a/sysdeps/linux-gnu/proc.c
+++ b/sysdeps/linux-gnu/proc.c
@@ -242,9 +242,10 @@ process_tasks(pid_t pid, pid_t **ret_tas
 	size_t alloc = 0;
 
 	while (1) {
-		struct dirent entry;
 		struct dirent *result;
-		if (readdir_r(d, &entry, &result) != 0) {
+		errno = 0;
+		result = readdir(d);
+		if (result == NULL && errno != 0) {
 		fail:
 			free(tasks);
 			closedir(d);
--- a/sysdeps/linux-gnu/arm/trace.c
+++ b/sysdeps/linux-gnu/arm/trace.c
@@ -155,6 +155,8 @@ arm_get_next_pcs(struct process *proc,
 	const unsigned cond = BITS(this_instr, 28, 31);
 	const unsigned opcode = BITS(this_instr, 24, 27);
 
+	uint32_t operand1, operand2, result = 0;
+
 	if (cond == COND_NV)
 		switch (opcode) {
 			arch_addr_t addr;
@@ -170,7 +172,6 @@ arm_get_next_pcs(struct process *proc,
 		}
 	else
 		switch (opcode) {
-			uint32_t operand1, operand2, result = 0;
 		case 0x0:
 		case 0x1:			/* data processing */
 		case 0x2:
