--- a/samples/client.conf
+++ b/samples/client.conf
@@ -38,15 +38,15 @@
 # Script to run after VPN is created. All key-value pairs (except password) in
 # this file will be passed to the script as environment variables. Use this
 # script to set up routes, turn on NAT, etc.
-up=/etc/shadowvpn/client_up.sh
+up=/opt/etc/shadowvpn/client_up.sh
 
 # Script to run before stopping VPN. All key-value pairs (except password) in
 # this file will be passed to the script as environment variables. Use this
 # script to restore routes, turn off NAT, etc.
-down=/etc/shadowvpn/client_down.sh
+down=/opt/etc/shadowvpn/client_down.sh
 
 # PID file path
-pidfile=/var/run/shadowvpn.pid
+pidfile=/opt/var/run/shadowvpn.pid
 
 # Log file path
-logfile=/var/log/shadowvpn.log
+logfile=/opt/var/log/shadowvpn.log
--- a/samples/server.conf
+++ b/samples/server.conf
@@ -49,15 +49,15 @@
 # Script to run after VPN is created. All key-value pairs (except password) in
 # this file will be passed to the script as environment variables. Use this
 # script to set up routes, turn on NAT, etc.
-up=/etc/shadowvpn/server_up.sh
+up=/opt/etc/shadowvpn/server_up.sh
 
 # Script to run before stopping VPN. All key-value pairs (except password) in
 # this file will be passed to the script as environment variables. Use this
 # script to restore routes, turn off NAT, etc.
-down=/etc/shadowvpn/server_down.sh
+down=/opt/etc/shadowvpn/server_down.sh
 
 # PID file path
-pidfile=/var/run/shadowvpn.pid
+pidfile=/opt/var/run/shadowvpn.pid
 
 # Log file path
-logfile=/var/log/shadowvpn.log
+logfile=/opt/var/log/shadowvpn.log
