--- a/samples/server.conf
+++ b/samples/server.conf
@@ -30,15 +30,15 @@
 # the script to run after VPN is created
 # use this script to set up routes, NAT, etc
 # configuration in this file will be set as environment variables
-up=/etc/shadowvpn/server_up.sh
+up=/opt/etc/shadowvpn/server_up.sh
 
 # the script to run before stopping VPN
 # use this script to restore routes, NAT, etc
 # configuration in this file will be set as environment variables
-down=/etc/shadowvpn/server_down.sh
+down=/opt/etc/shadowvpn/server_down.sh
 
 # PID file path
-pidfile=/var/run/shadowvpn.pid
+pidfile=/opt/var/run/shadowvpn.pid
 
 # log file path
-logfile=/var/log/shadowvpn.log
+logfile=/opt/var/log/shadowvpn.log
--- a/samples/client.conf
+++ b/samples/client.conf
@@ -31,15 +31,15 @@
 # the script to run after VPN is created
 # use this script to set up routes, NAT, etc
 # configuration in this file will be set as environment variables
-up=/etc/shadowvpn/client_up.sh
+up=/opt/etc/shadowvpn/client_up.sh
 
 # the script to run before stopping VPN
 # use this script to restore routes, NAT, etc
 # configuration in this file will be set as environment variables
-down=/etc/shadowvpn/client_down.sh
+down=/opt/etc/shadowvpn/client_down.sh
 
 # PID file path
-pidfile=/var/run/shadowvpn.pid
+pidfile=/opt/var/run/shadowvpn.pid
 
 # log file path
-logfile=/var/log/shadowvpn.log
+logfile=/opt/var/log/shadowvpn.log
