--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -10,27 +10,27 @@ if DBG_BUILD
 endif
 
 automatic_SOURCES = \
-   $(top_srcdir)/src/automatic.c      \
-   $(top_srcdir)/src/base64.c         \
-   $(top_srcdir)/src/config_parser.c  \
-   $(top_srcdir)/src/downloads.c      \
-   $(top_srcdir)/src/feed_item.c      \
-   $(top_srcdir)/src/file.c           \
-   $(top_srcdir)/src/json.c           \
-   $(top_srcdir)/src/list.c           \
-   $(top_srcdir)/src/output.c         \
-   $(top_srcdir)/src/filters.c        \
-   $(top_srcdir)/src/prowl.c          \
-   $(top_srcdir)/src/pushover.c       \
-   $(top_srcdir)/src/regex.c          \
-   $(top_srcdir)/src/rss_feed.c       \
-   $(top_srcdir)/src/state.c          \
-   $(top_srcdir)/src/torrent.c        \
-   $(top_srcdir)/src/transmission.c   \
-   $(top_srcdir)/src/urlcode.c        \
-   $(top_srcdir)/src/utils.c          \
-   $(top_srcdir)/src/web.c            \
-   $(top_srcdir)/src/xml_parser.c
+   automatic.c      \
+   base64.c         \
+   config_parser.c  \
+   downloads.c      \
+   feed_item.c      \
+   file.c           \
+   json.c           \
+   list.c           \
+   output.c         \
+   filters.c        \
+   prowl.c          \
+   pushover.c       \
+   regex.c          \
+   rss_feed.c       \
+   state.c          \
+   torrent.c        \
+   transmission.c   \
+   urlcode.c        \
+   utils.c          \
+   web.c            \
+   xml_parser.c
 
 noinst_HEADERS =    \
    $(top_srcdir)/include/automatic.h      \
@@ -57,7 +57,7 @@ noinst_HEADERS =    \
 
 automatic_mw_SOURCES =  \
   $(automatic_SOURCES) \
-	$(top_srcdir)/src/memwatch.c
+	memwatch.c
 
 if DBG_BUILD
   automatic_mwdir = .
