From 99ec484ec28675b9928a3740450272e35c20227d Mon Sep 17 00:00:00 2001
From: Xavier G <xavier@kindwolf.org>
Date: Fri, 1 May 2020 13:26:59 +0000
Subject: [PATCH] text header: prevent unwanted color pollution of the next
 line.

---
 src/export/text.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/src/export/text.c
+++ b/src/export/text.c
@@ -132,9 +132,10 @@ text_disp_header(void)
 
 	if (oflag)
 		(void)printf("%-*s", max.mntopts, _("MOUNT OPTIONS"));
-	(void)printf("\n");
-
+	/* reset color before newline to prevent unwanted pollution of the next line */
 	reset_color();
+
+	(void)printf("\n");
 }
 
 /*
