--- a/cmake/compilerFlags.cmake
+++ b/cmake/compilerFlags.cmake
@@ -23,22 +23,6 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Li
 
 
     if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
-        # This fails under Fedora - MinGW - Gcc 8.3
-        if (NOT (MINGW OR CYGWIN OR CMAKE_HOST_SOLARIS))
-            check_cxx_compiler_flag(-fstack-clash-protection HAS_FSTACK_CLASH_PROTECTION)
-            check_cxx_compiler_flag(-fcf-protection HAS_FCF_PROTECTION)
-            check_cxx_compiler_flag(-fstack-protector-strong HAS_FSTACK_PROTECTOR_STRONG)
-            if(HAS_FSTACK_CLASH_PROTECTION)
-                add_compile_options(-fstack-clash-protection)
-            endif()
-            if(HAS_FCF_PROTECTION)
-                add_compile_options(-fcf-protection)
-            endif()
-            if(HAS_FSTACK_PROTECTOR_STRONG)
-                add_compile_options(-fstack-protector-strong)
-            endif()
-        endif()
-
         add_compile_options(-Wp,-D_GLIBCXX_ASSERTIONS)
 
         if (CMAKE_BUILD_TYPE STREQUAL Release AND NOT (APPLE OR MINGW OR MSYS))
