--- a/Makefile.classic
+++ b/Makefile.classic
@@ -27,7 +27,7 @@ OBJDIR = ./bld
 #    See TCC below for the C compiler for building the finished binary.
 #
 BCC = gcc
-BCCFLAGS = $(CFLAGS)
+BCCFLAGS = 
 
 #### The suffix to add to final executable file.  When cross-compiling
 #    to windows, make this ".exe".  Otherwise leave it blank.
@@ -42,7 +42,7 @@ E =
 #
 #TCC = gcc -O6
 #TCC = gcc -g -O0 -Wall -fprofile-arcs -ftest-coverage
-TCC = gcc -g -Os -Wall
+TCC = $(CC)
 
 # To add support for HTTPS
 TCC += -DFOSSIL_ENABLE_SSL
@@ -84,7 +84,7 @@ TCLSH = tclsh
 
 CFLAGS += -fPIE
 CPPFLAGS += -I. -I$(SRCDIR_extsrc) -I$(SRCDIR)
-LIB = -lm -lz -lssl
+#LIB = -lm -lz -lssl
 INSTALLDIR = $(DESTDIR)$(prefix)/bin
 SQLITE3_ORIGINAL = 0
 USE_LINENOISE = 1
