--- a/config.m4
+++ b/config.m4
@@ -5,11 +5,6 @@ if test $PHP_GMAGICK != "no"; then
 
 		AC_MSG_CHECKING(GraphicsMagick configuration program)
 
-		for i in $PHP_GMAGICK /usr/local /usr;
-		do
-			test -r $i/bin/GraphicsMagick-config && WAND_BINARY=$i/bin/GraphicsMagick-config && break
-		done
-
 		if test -z "$WAND_BINARY"; then
 			AC_MSG_ERROR(not found. Please provide a path to GraphicsMagick-config program.)
 		fi
