--- a/config.h
+++ b/config.h
@@ -20,3 +20,3 @@
 #ifdef linux
-# define ENABLE_NLS
+/* # define ENABLE_NLS */
 #endif
--- a/Makefile
+++ b/Makefile
@@ -61,7 +61,7 @@
 CPPFLAGS += $(DEFS) $(INCLUDES)
 
 ##############################################################################
-all: Makefile.depend whois mkpasswd pos
+all: Makefile.depend whois
 
 ##############################################################################
 %.o: %.c
