--- a/make-linux.mk
+++ b/make-linux.mk
@@ -17,15 +17,6 @@
 #   clean: removes all built files, objects, other trash
 #
 
-# Automagically pick clang or gcc, with preference for clang
-# This is only done if we have not overridden these with an environment or CLI variable
-ifeq ($(origin CC),default)
-	CC=$(shell if [ -e /usr/bin/clang ]; then echo clang; else echo gcc; fi)
-endif
-ifeq ($(origin CXX),default)
-	CXX=$(shell if [ -e /usr/bin/clang++ ]; then echo clang++; else echo g++; fi)
-endif
-
 UNAME_M=$(shell $(CC) -dumpmachine | cut -d '-' -f 1)
 
 INCLUDES=
@@ -50,7 +41,7 @@
 # Build with ZT_ENABLE_NETWORK_CONTROLLER=1 to build with the Sqlite network controller
 ifeq ($(ZT_ENABLE_NETWORK_CONTROLLER),1)
         DEFS+=-DZT_ENABLE_NETWORK_CONTROLLER 
-        LDLIBS+=-L/usr/local/lib -lsqlite3
+        LDLIBS+=-lsqlite3
         OBJS+=controller/SqliteNetworkController.o 
 endif
 
@@ -74,8 +65,7 @@
 	CFLAGS+=-Wall -fPIE -fvisibility=hidden -pthread $(INCLUDES) -DNDEBUG $(DEFS)
 	CXXFLAGS?=-O3 -fstack-protector
 	CXXFLAGS+=-Wall -fPIE -fvisibility=hidden -fno-rtti -pthread $(INCLUDES) -DNDEBUG $(DEFS)
-	LDFLAGS=-pie -Wl,-z,relro,-z,now
-	STRIP=strip --strip-all
+	LDFLAGS+=-pie -Wl,-z,relro,-z,now
 endif
 ifeq ($(ZT_TRACE),1)
 	DEFS+=-DZT_TRACE 
