--- a/setup.py
+++ b/setup.py
@@ -208,11 +208,12 @@
 _ext_modules = []
 
 # Check for a system libtorrent and if found, then do not build the libtorrent extension
-build_libtorrent = True
+# Force libtorrent-rasterbar do not build
+build_libtorrent = False
 try:
     from deluge._libtorrent import lt
 except ImportError:
-    build_libtorrent = True
+    build_libtorrent = False
 else:
     build_libtorrent = False
 
