--- a/deluge/core/core.py
+++ b/deluge/core/core.py
@@ -342,7 +342,6 @@
     def _on_alert_session_stats(self, alert):
         """The handler for libtorrent session stats alert"""
         self.session_status.update(alert.values)
-        self._update_session_cache_hit_ratio()
 
     def _update_session_cache_hit_ratio(self):
         """Calculates the cache read/write hit ratios for session_status."""
