--- a/Makefile
+++ b/Makefile
@@ -11,8 +11,8 @@ all: usbrelay libusbrelay.so libusbrelay
 
 libusbrelay.so: libusbrelay.c libusbrelay.h 
 	$(CC) -shared -fPIC -Wl,-soname,$@.$(USBMAJOR) $(CPPFLAGS) $(CFLAGS)  $< $(LDFLAGS) -o $@.$(USBLIBVER) $(LDLIBS)
-	$(LDCONFIG) -n .
-	ln -sr libusbrelay.so.$(USBLIBVER) libusbrelay.so
+	ln -sr libusbrelay.so.$(USBLIBVER) libusbrelay.so.$(USBMAJOR)
+	ln -sr libusbrelay.so.$(USBMAJOR) libusbrelay.so
 
 usbrelay: usbrelay.c libusbrelay.h libusbrelay.so
 	$(CC) $(CPPFLAGS) $(CFLAGS)  $< -l:libusbrelay.so.$(USBLIBVER) -L./ $(LDFLAGS) -o $@ $(LDLIBS)
