--- a/configure.in
+++ b/configure.in
@@ -155,15 +155,6 @@ AC_ARG_WITH([rcdir],
   esac])
 AC_SUBST([rcdir])
 
-# test for -fstack-protector
-saved_CFLAGS="$CFLAGS"
-AC_MSG_CHECKING([whether the C compiler accepts -fstack-protector])
-CFLAGS="$CFLAGS -fstack-protector"
-AC_TRY_COMPILE(,,
-  AC_MSG_RESULT(yes); saved_CFLAGS="$CFLAGS",
-  AC_MSG_RESULT(no))
-CFLAGS="$saved_CFLAGS"
-
 # test for -Wall
 saved_CFLAGS="$CFLAGS"
 AC_MSG_CHECKING([whether the C compiler accepts -Wall])
