--- a/include/libtorrent/ip_filter.hpp
+++ b/include/libtorrent/ip_filter.hpp
@@ -41,10 +41,16 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <vector>
 
 #include <boost/limits.hpp>
-#include <boost/utility.hpp>
 #include <boost/cstdint.hpp>
 #include <boost/tuple/tuple.hpp>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106700
+#include <boost/next_prior.hpp>
+#else
+#include <boost/utility.hpp>
+#endif
+
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
 #include "libtorrent/address.hpp"
--- a/src/kademlia/routing_table.cpp
+++ b/src/kademlia/routing_table.cpp
@@ -54,6 +54,13 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <boost/cstdint.hpp>
 #include <boost/bind.hpp>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 106700
+#include <boost/next_prior.hpp>
+#else
+#include <boost/utility.hpp>
+#endif
+
 #include "libtorrent/aux_/disable_warnings_pop.hpp"
 
 using boost::uint8_t;
