--- a/cmake/Modules/GeneratePkgConfig.cmake	2019-12-09 08:55:16.000000000 +0300
+++ b/cmake/Modules/GeneratePkgConfig.cmake	2020-05-22 08:56:06.289531654 +0300
@@ -30,6 +30,7 @@
 function(_get_target_property_merging_configs _var_name _target_name _propert_name)
 	get_target_property(vals ${_target_name} ${_propert_name})
 	if (NOT vals)
+ 		set(vals "")
 		if (CMAKE_BUILD_TYPE)
 			list(APPEND configs ${CMAKE_BUILD_TYPE})
 		elseif()
--- a/cmake/Modules/GeneratePkgConfig/pkg-config.cmake.in	2019-12-09 08:55:16.000000000 +0300
+++ b/cmake/Modules/GeneratePkgConfig/pkg-config.cmake.in	2020-05-22 08:35:57.715797868 +0300
@@ -1,8 +1,9 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 libdir=${prefix}/@_INSTALL_LIBDIR@
+includedir=${prefix}/@_INSTALL_INCLUDEDIR@
 
 Name: @_PROJECT_NAME@
 Description: @_PROJECT_DESCRIPTION@
 Version: @_PROJECT_VERSION@
 Libs: -L${libdir} -l@_TARGET_OUTPUT_NAME@ @_interface_link_libraries@
-Cflags: @_interface_compile_options@ @_interface_include_dirs@ @_interface_definitions@
+Cflags: @_interface_compile_options@ -I${includedir} @_interface_definitions@
