--- a/lib/replace/wscript
+++ b/lib/replace/wscript
@@ -69,32 +69,6 @@ def configure(conf):
 
     conf.SET_TARGET_TYPE('tirpc', 'EMPTY')
     conf.CHECK_HEADERS('rpc/rpc.h rpc/nettype.h')
-    if not conf.CONFIG_SET('HAVE_RPC_RPC_H'):
-        if conf.CHECK_CFG(package='libtirpc', args='--cflags --libs',
-                       msg='Checking for libtirpc headers',
-                       uselib_store='TIRPC'):
-            conf.CHECK_HEADERS('rpc/rpc.h rpc/nettype.h', lib='tirpc', together=True)
-            conf.SET_TARGET_TYPE('tirpc', 'SYSLIB')
-    if not conf.CONFIG_SET('HAVE_RPC_RPC_H'):
-        if conf.CHECK_CFG(package='libntirpc', args='--cflags',
-                       msg='Checking for libntirpc headers',
-                       uselib_store='TIRPC'):
-            conf.CHECK_HEADERS('rpc/rpc.h rpc/nettype.h', lib='tirpc', together=True)
-            conf.SET_TARGET_TYPE('tirpc', 'SYSLIB')
-    if not conf.CONFIG_SET('HAVE_RPC_RPC_H'):
-            Logs.error('ERROR: No rpc/rpc.h header found, tirpc or libntirpc missing?')
-            sys.exit(1)
-
-    conf.SET_TARGET_TYPE('nsl', 'EMPTY')
-    conf.CHECK_HEADERS('rpc/rpc.h rpcsvc/yp_prot.h', lib='tirpc')
-    if not conf.CONFIG_SET('HAVE_RPCSVC_YP_PROT_H'):
-        if conf.CHECK_CFG(package='libnsl', args='--cflags --libs',
-                          msg='Checking for libnsl',
-                          uselib_store='NSL'):
-            conf.SET_TARGET_TYPE('nsl', 'SYSLIB')
-            conf.CHECK_HEADERS('rpc/rpc.h rpcsvc/yp_prot.h', lib='tirpc nsl')
-        else:
-            conf.SET_TARGET_TYPE('nsl', 'SYSLIB')
     conf.CHECK_HEADERS('rpcsvc/nis.h rpcsvc/ypclnt.h', lib='tirpc nsl')
 
     conf.CHECK_HEADERS('sys/sysctl.h')
