--- a/source3/libads/kerberos_keytab.c
+++ b/source3/libads/kerberos_keytab.c
@@ -126,7 +126,11 @@ int ads_keytab_add_entry(ADS_STRUCT *ads
 		return -1;
 	}
 
+#ifdef HAVE_ADS
 	ret = ads_keytab_open(context, &keytab);
+#else
+	ret = -1;
+#endif /* HAVE_ADS */
 	if (ret != 0) {
 		goto out;
 	}
@@ -313,7 +317,11 @@ int ads_keytab_flush(ADS_STRUCT *ads)
 		return ret;
 	}
 
+#ifdef HAVE_ADS
 	ret = ads_keytab_open(context, &keytab);
+#else
+	ret = -1;
+#endif /* HAVE_ADS */
 	if (ret != 0) {
 		goto out;
 	}
@@ -496,7 +504,11 @@ int ads_keytab_create_default(ADS_STRUCT
 	DEBUG(3, (__location__ ": Searching for keytab entries to preserve "
 		  "and update.\n"));
 
+#ifdef HAVE_ADS
 	ret = ads_keytab_open(context, &keytab);
+#else
+	ret = -1;
+#endif /* HAVE_ADS */
 	if (ret != 0) {
 		goto done;
 	}
@@ -640,7 +652,11 @@ int ads_keytab_list(const char *keytab_n
 	}
 
 	if (keytab_name == NULL) {
+#ifdef HAVE_ADS
 		ret = ads_keytab_open(context, &keytab);
+#else
+		ret = -1;
+#endif /* HAVE_ADS */
 	} else {
 		ret = smb_krb5_kt_open(context, keytab_name, False, &keytab);
 	}
