--- a/Makefile.in	2000-02-06 05:05:57.000000000 +0300
+++ b/Makefile.in	2012-08-14 15:47:44.880812037 +0400
@@ -6,23 +6,25 @@
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 MKINSTALLDIRS= @srcdir@/mkinstalldirs
+DESTDIR=
 
 CC=@CC@
 LIBS=@LIBS@
 CFLAGS=@CFLAGS@
+LDFLAGS=@LDFLAGS@
 
 all: loadwatch lw-ctl
 
 loadwatch: loadwatch.o
-	$(CC) -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 lw-ctl: lw-ctl.o
-	$(CC) -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 install: all
-	$(MKINSTALLDIRS) $(bindir)
-	$(INSTALL_PROGRAM) loadwatch $(bindir)/loadwatch
-	$(INSTALL_PROGRAM) lw-ctl $(bindir)/loadwatch
+	$(MKINSTALLDIRS) $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) loadwatch $(DESTDIR)$(bindir)/loadwatch
+	$(INSTALL_PROGRAM) lw-ctl $(DESTDIR)$(bindir)/lw-ctl
 
 clean:
 	-rm -f *.o loadwatch lw-ctl
