--- loadwatch-1.0+1.1alpha1.orig/Makefile.in
+++ loadwatch-1.0+1.1alpha1/Makefile.in
@@ -6,23 +6,26 @@
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 MKINSTALLDIRS= @srcdir@/mkinstalldirs
+DESTDIR=
 
 CC=@CC@
 LIBS=@LIBS@
+CPPFLAGS=@CPPFLAGS@
 CFLAGS=@CFLAGS@
+LDFLAGS=@LDFLAGS@
 
 all: loadwatch lw-ctl
 
 loadwatch: loadwatch.o
-	$(CC) -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 lw-ctl: lw-ctl.o
-	$(CC) -o $@ $^ $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 install: all
-	$(MKINSTALLDIRS) $(bindir)
-	$(INSTALL_PROGRAM) loadwatch $(bindir)/loadwatch
-	$(INSTALL_PROGRAM) lw-ctl $(bindir)/loadwatch
+	$(MKINSTALLDIRS) $(DESTDIR)$(bindir)
+	$(INSTALL_PROGRAM) loadwatch $(DESTDIR)$(bindir)/loadwatch
+	$(INSTALL_PROGRAM) lw-ctl $(DESTDIR)$(bindir)/lw-ctl
 
 clean:
 	-rm -f *.o loadwatch lw-ctl
