--- a/configure.ac.in
+++ b/configure.ac.in
@@ -286,11 +286,14 @@ if test -d /usr/local/include/ndpi ; the
    echo "WARNING: /usr/local/include/ndpi is present and might lead to compile errors"
 fi
 
-NDPI_INC=`echo $NDPI_CFLAGS | sed -e "s/[ ]*$//"`
-NDPI_LIB=
-NDPI_LIB_DEP=
-
-AC_MSG_CHECKING(for nDPI source)
+PKG_CHECK_MODULES([NDPI], [libndpi >= 2.0], [
+   NDPI_INC=`echo $NDPI_CFLAGS | sed -e "s/[ ]*$//"`
+   # Use static libndpi library as building against the dynamic library fails
+   NDPI_LIB="-lndpi"
+   #NDPI_LIB="$NDPI_LIBS"
+   NDPI_LIB_DEP=
+   ], [
+      AC_MSG_CHECKING(for nDPI source)
       if test -d "./nDPI" ; then :
          NDPI_HOME=./nDPI
       elif test -d "../nDPI" ; then :
@@ -315,7 +318,7 @@ AC_MSG_CHECKING(for nDPI source)
       else
          AC_MSG_RESULT(not found)
       fi
-
+   ])
 
 if test ! -z "$NDPI_CUST_INC" ; then :
   NDPI_INC=$NDPI_CUST_INC
