--- a/chan_quectel.c
+++ b/chan_quectel.c
@@ -255,7 +255,7 @@ static int lock_build(const char * devna
 	#if defined(__FreeBSD__)
 	return snprintf(buf, length, "/var/spool/lock/LCK..%s", basename);
 	#else
-	return snprintf(buf, length, "/var/lock/LCK..%s", basename);
+	return snprintf(buf, length, "/opt/var/lock/LCK..%s", basename);
 	#endif
 }
 
--- a/dc_config.h
+++ b/dc_config.h
@@ -100,7 +100,7 @@ typedef struct dc_gconfig
 	int			discovery_interval;		/*!< The device discovery interval */
 #define DEFAULT_DISCOVERY_INT	60
 	char sms_db[PATHLEN];
-#define DEFAULT_SMS_DB "/var/lib/asterisk/smsdb"
+#define DEFAULT_SMS_DB "/opt/var/lib/asterisk/smsdb"
 	int csms_ttl;
 #define DEFAULT_CSMS_TTL 600
 
--- a/etc/quectel.conf
+++ b/etc/quectel.conf
@@ -1,7 +1,7 @@
 [general]
 
 interval=15			; Number of seconds between trying to connect to devices
-smsdb=/var/lib/asterisk/smsdb
+smsdb=/opt/var/lib/asterisk/smsdb
 csmsttl=600
 
 ;------------------------------ JITTER BUFFER CONFIGURATION --------------------------
--- a/tools/tty.c
+++ b/tools/tty.c
@@ -47,7 +47,7 @@ static int lock_build(const char * devna
 	#if defined(__FreeBSD__)
 	return snprintf(buf, length, "/var/spool/lock/LCK..%s", basename);
 	#else
-	return snprintf(buf, length, "/var/lock/LCK..%s", basename);
+	return snprintf(buf, length, "/opt/var/lock/LCK..%s", basename);
 	#endif
 }
 
