--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION := 0.20.10
 
 # install directory layout
-PREFIX ?= /usr/local
+PREFIX ?= /opt
 INCLUDEDIR ?= $(PREFIX)/include
 LIBDIR ?= $(PREFIX)/lib
 PCLIBDIR ?= $(LIBDIR)/pkgconfig
@@ -41,7 +41,7 @@ ifneq (,$(filter $(shell uname),FreeBSD
 	PCLIBDIR := $(PREFIX)/libdata/pkgconfig
 endif
 
-all: libtree-sitter.a libtree-sitter.$(SOEXTVER)
+all: libtree-sitter.a
 
 libtree-sitter.a: $(OBJ)
 	$(AR) rcs $@ $^
@@ -62,9 +62,6 @@ install: all
 
 	install -d '$(DESTDIR)$(LIBDIR)'
 	install -m644 libtree-sitter.a '$(DESTDIR)$(LIBDIR)'/
-	install -m755 libtree-sitter.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/
-	ln -sf libtree-sitter.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter.$(SOEXTVER_MAJOR)
-	ln -sf libtree-sitter.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter.$(SOEXT)
 
 	install -d '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter
 	install -m644 lib/include/tree_sitter/api.h '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter/
@@ -73,6 +70,6 @@ install: all
 	install -m644 tree-sitter.pc '$(DESTDIR)$(PCLIBDIR)'/
 
 clean:
-	rm -f lib/src/*.o libtree-sitter.a libtree-sitter.$(SOEXT) libtree-sitter.$(SOEXTVER_MAJOR) libtree-sitter.$(SOEXTVER)
+	rm -f lib/src/*.o libtree-sitter.a
 
 .PHONY: all install clean
