--- a/libfuse/Makefile
+++ b/libfuse/Makefile
@@ -56,7 +56,8 @@ CFLAGS := \
 	-std=gnu99 \
 	-Wall \
 	-pipe \
-	-MMD
+	-MMD \
+	$(CXXFLAGS)
 FUSERMOUNT_DIR = $(BINDIR)
 FUSE_FLAGS = \
 	-Iinclude \
@@ -107,8 +108,8 @@ clean:
 distclean: clean
 
 strip:
-	strip --strip-all build/mount.mergerfs
-	strip --strip-all build/mergerfs-fusermount
+	$(STRIP) --strip-all build/mount.mergerfs
+	$(STRIP) --strip-all build/mergerfs-fusermount
 
 install-utils: mergerfs-fusermount mount.mergerfs strip
 	install -D build/mergerfs-fusermount "$(INSTALLBINDIR)/mergerfs-fusermount"
