--- a/Makefile
+++ b/Makefile
@@ -76,7 +75,7 @@
 MFS_FLAGS  = \
 	      -DUSE_XATTR=$(USE_XATTR) \
 	      -DUGID_USE_RWLOCK=$(UGID_USE_RWLOCK)
-LDFLAGS    = \
+LDFLAGS    += \
 	      -pthread \
               -lrt
 
--- a/libfuse/Makefile
+++ b/libfuse/Makefile
@@ -52,7 +52,7 @@
 	-DFUSE_USE_VERSION=29 \
 	-DPACKAGE_VERSION=\"$(VERSION)\" \
 	-DFUSERMOUNT_DIR=\"$(FUSERMOUNT_DIR)\"
-LDFLAGS = \
+LDFLAGS += \
 	-lrt \
 	-pthread
 
