--- a/Makefile
+++ b/Makefile
@@ -50,7 +50,7 @@
 UNIX2MAC        = unix2mac
 
 # Native Language Support (NLS)
-ENABLE_NLS      = 1
+#ENABLE_NLS      = 1
 # Large File Support (LFS)
 LFS             = 1
 # DEBUG=1 adds the -g option to CFLAGS, for adding debug symbols.
@@ -358,7 +358,7 @@
 endif
 
 LDFLAGS_USER    =
-LDFLAGS = $(RPM_OPT_FLAGS) $(LDFLAGS_EXTRA) $(LDFLAGS_USER)
+LDFLAGS += $(RPM_OPT_FLAGS) $(LDFLAGS_EXTRA) $(LDFLAGS_USER)
 LIBS    = $(LIBS_EXTRA)
 
 DEFS_USER       =
@@ -366,8 +366,7 @@
 
 # .......................................................... targets ...
 
-all: $(BIN) $(MAC2UNIX_BIN) $(UNIX2DOS_BIN) $(UNIX2MAC_BIN) doc \
-     mofiles man
+all: $(BIN) $(MAC2UNIX_BIN) $(UNIX2DOS_BIN) $(UNIX2MAC_BIN)
 
 status:
 	@echo "D2U_OS       = $(D2U_OS)"
