--- a/castgetrc.example
+++ b/castgetrc.example
@@ -11,7 +11,7 @@
 # enclosures.
 [*]
 id3contenttype=Podcast
-spool=/home/tom/podcasts
+spool=/opt/tmp/podcasts
 
 #
 # Per-channel settings.
@@ -23,8 +23,8 @@
 # enclosures to /home/tom/podcasts.m3u.
 [dsc]
 url=http://radio.weblogs.com/0001014/categories/dailySourceCode/rss.xml
-spool=/home/tom/podcasts/dsc
-playlist=/home/tom/podcasts.m3u
+spool=/opt/podcasts/dsc
+playlist=/opt/tmp/podcasts/podcasts.m3u
 
 # This will define a channel with the identifier "yeast", set the RSS
 # feed URL for the channel, and set several ID3 tags.
