--- a/module.mk
+++ b/module.mk
@@ -28,13 +28,13 @@ DRV_ONLY		?= 0
 
 
 # Flags
-CXXFLAGS		+= `pkg-config --cflags cups` -Iinclude -Wall -I/opt/local/include
+CXXFLAGS		+= -Iinclude -Wall
 DEBUG_CXXFLAGS		+= -DDEBUG  -DDUMP_CACHE
 OPTIM_CXXFLAGS 		+= -g
-rastertoqpdl_LDFLAGS	:= $(LDFLAGS) -L/opt/local/lib
-rastertoqpdl_LIBS	:= `pkg-config --libs cups` -lcupsimage
+rastertoqpdl_LDFLAGS	:= $(LDFLAGS)
+rastertoqpdl_LIBS	:= -lcupsimage -lcups
 pstoqpdl_LDFLAGS	:= $(LDFLAGS)
-pstoqpdl_LIBS		:= `pkg-config --libs cups` -lcupsimage
+pstoqpdl_LIBS		:= -lcupsimage -lcups
 
 
 # Update compilation flags with defined options
