--- a/accel-pppd/accel-ppp.conf
+++ b/accel-pppd/accel-ppp.conf
@@ -1,5 +1,5 @@
 [modules]
-log_file
+#log_file
 #log_syslog
 #log_tcp
 #log_pgsql
@@ -32,7 +32,7 @@
 #ipv6pool
 
 [core]
-log-error=/var/log/accel-ppp/core.log
+log-error=/opt/var/log/accel-ppp/core.log
 thread-count=4
 
 [common]
@@ -91,7 +91,7 @@
 
 [l2tp]
 verbose=1
-#dictionary=/usr/local/share/accel-ppp/l2tp/dictionary
+#dictionary=/opt/share/accel-ppp/l2tp/dictionary
 #hello-interval=60
 #timeout=60
 #rtimeout=1
@@ -112,13 +112,13 @@
 #cert-hash-sha1=
 #cert-hash-sha256=
 #accept=ssl,proxy
-#ssl-dhparam=/etc/ssl/dhparam.pem
+#ssl-dhparam=/opt/etc/ssl/dhparam.pem
 #ssl-ecdh-curve=prime256v1
 #ssl-ciphers=DEFAULT
 #ssl-prefer-server-ciphers=0
-#ssl-ca-file=/etc/ssl/sstp-ca.crt
-#ssl-pemfile=/etc/ssl/sstp-cert.pem
-#ssl-keyfile=/etc/ssl/sstp-key.pem
+#ssl-ca-file=/opt/etc/ssl/sstp-ca.crt
+#ssl-pemfile=/opt/etc/ssl/sstp-cert.pem
+#ssl-keyfile=/opt/etc/ssl/sstp-key.pem
 #host-name=domain.tld
 #http-error=allow
 #timeout=60
@@ -160,7 +160,7 @@
 #attr-l4-redirect=L4-Redirect
 #attr-l4-redirect-table=4
 #attr-l4-redirect-ipset=l4-redirect
-#lua-file=/etc/accel-ppp.lua
+#lua-file=/opt/etc/accel-ppp.lua
 #offer-delay=0,100:100,200:200,-1:1000
 #vlan-mon=eth0,10-200
 #vlan-timeout=60
@@ -184,7 +184,7 @@
 #wins2=172.16.1.1
 
 [radius]
-#dictionary=/usr/local/share/accel-ppp/radius/dictionary
+#dictionary=/opt/share/accel-ppp/radius/dictionary
 nas-identifier=accel-ppp
 nas-ip-address=127.0.0.1
 gw-ip-address=192.168.100.1
@@ -214,9 +214,9 @@
 192.168.4.0/24
 
 [log]
-log-file=/var/log/accel-ppp/accel-ppp.log
-log-emerg=/var/log/accel-ppp/emerg.log
-log-fail-file=/var/log/accel-ppp/auth-fail.log
+log-file=/opt/var/log/accel-ppp/accel-ppp.log
+log-emerg=/opt/var/log/accel-ppp/emerg.log
+log-fail-file=/opt/var/log/accel-ppp/auth-fail.log
 #log-debug=/dev/stdout
 #syslog=accel-pppd,daemon
 #log-tcp=127.0.0.1:3000
@@ -233,16 +233,16 @@
 
 [pppd-compat]
 verbose=1
-#ip-pre-up=/etc/ppp/ip-pre-up
-ip-up=/etc/ppp/ip-up
-ip-down=/etc/ppp/ip-down
-#ip-change=/etc/ppp/ip-change
-radattr-prefix=/var/run/radattr
+#ip-pre-up=/opt/etc/ppp/ip-pre-up
+ip-up=/opt/etc/ppp/ip-up
+ip-down=/opt/etc/ppp/ip-down
+#ip-change=/opt/etc/ppp/ip-change
+radattr-prefix=/opt/var/run/radattr
 #fork-limit=16
 
 [chap-secrets]
 gw-ip-address=192.168.100.1
-#chap-secrets=/etc/ppp/chap-secrets
+#chap-secrets=/opt/etc/ppp/chap-secrets
 #encrypted=0
 #username-hash=md5
 
