--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ endif ()
 
 project(accel-ppp C)
 
-SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fno-strict-aliasing -D_GNU_SOURCE -fPIC -rdynamic -Wno-unused-result")
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fno-strict-aliasing -D_GNU_SOURCE -rdynamic -Wno-unused-result")
 
 IF (NOT DEFINED LIB_SUFFIX)
 	EXECUTE_PROCESS(
--- a/accel-pppd/triton/CMakeLists.txt
+++ b/accel-pppd/triton/CMakeLists.txt
@@ -33,7 +33,7 @@ INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOUR
 ADD_DEFINITIONS(-DMODULE_PATH="${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/accel-ppp")
 
 ADD_LIBRARY(triton SHARED ${sources_c})
-TARGET_LINK_LIBRARIES(triton dl)
+TARGET_LINK_LIBRARIES(triton dl atomic)
 
 INSTALL(TARGETS triton
 	LIBRARY DESTINATION lib${LIB_SUFFIX}/accel-ppp
