--- a/accel-pppd/ctrl/l2tp/l2tp.c
+++ b/accel-pppd/ctrl/l2tp/l2tp.c
@@ -4912,8 +4912,6 @@
 	fd = socket(AF_PPPOX, SOCK_DGRAM, PX_PROTO_OL2TP);
 	if (fd >= 0)
 		close(fd);
-	else if (system("modprobe -q pppol2tp || modprobe -q l2tp_ppp"))
-		log_warn("unable to load l2tp kernel module\n");
 
 	l2tp_conn = _malloc((UINT16_MAX + 1) * sizeof(struct l2tp_conn_t *));
 	memset(l2tp_conn, 0, (UINT16_MAX + 1) * sizeof(struct l2tp_conn_t *));
--- a/accel-pppd/ctrl/pppoe/pppoe.c
+++ b/accel-pppd/ctrl/pppoe/pppoe.c
@@ -1558,8 +1558,6 @@
 	fd = socket(AF_PPPOX, SOCK_STREAM, PX_PROTO_OE);
 	if (fd >= 0)
 		close(fd);
-	else if (system("modprobe -q pppoe"))
-		log_warn("failed to load pppoe kernel module\n");
 
 	conn_pool = mempool_create(sizeof(struct pppoe_conn_t));
 	pado_pool = mempool_create(sizeof(struct delayed_pado_t));
--- a/accel-pppd/ctrl/pptp/pptp.c
+++ b/accel-pppd/ctrl/pptp/pptp.c
@@ -777,8 +777,6 @@
 	fd = socket(AF_PPPOX, SOCK_STREAM, PX_PROTO_PPTP);
 	if (fd >= 0)
 		close(fd);
-	else if (system("modprobe -q pptp"))
-		log_warn("failed to load pptp kernel module\n");
 
 	serv.hnd.fd = socket(PF_INET, SOCK_STREAM, 0);
 	if (serv.hnd.fd < 0) {
