--- a/accel-pppd/ctrl/l2tp/l2tp.c
+++ b/accel-pppd/ctrl/l2tp/l2tp.c
@@ -5083,8 +5083,6 @@ static void l2tp_init(void)
 	fd = socket(AF_PPPOX, SOCK_DGRAM, PX_PROTO_OL2TP);
 	if (fd >= 0)
 		close(fd);
-	else if (system("modprobe -q pppol2tp || modprobe -q l2tp_ppp"))
-		log_warn("unable to load l2tp kernel module\n");
 
 	l2tp_conn = _malloc((UINT16_MAX + 1) * sizeof(struct l2tp_conn_t *));
 	memset(l2tp_conn, 0, (UINT16_MAX + 1) * sizeof(struct l2tp_conn_t *));
--- a/accel-pppd/ctrl/pppoe/pppoe.c
+++ b/accel-pppd/ctrl/pppoe/pppoe.c
@@ -2137,8 +2137,6 @@ static void pppoe_init(void)
 	fd = socket(AF_PPPOX, SOCK_STREAM, PX_PROTO_OE);
 	if (fd >= 0)
 		close(fd);
-	else if (system("modprobe -q pppoe"))
-		log_warn("failed to load pppoe kernel module\n");
 
 	conn_pool = mempool_create(sizeof(struct pppoe_conn_t));
 	pado_pool = mempool_create(sizeof(struct delayed_pado_t));
--- a/accel-pppd/ctrl/pptp/pptp.c
+++ b/accel-pppd/ctrl/pptp/pptp.c
@@ -839,8 +839,6 @@ static void pptp_init(void)
 	fd = socket(AF_PPPOX, SOCK_STREAM, PX_PROTO_PPTP);
 	if (fd >= 0)
 		close(fd);
-	else if (system("modprobe -q pptp"))
-		log_warn("failed to load pptp kernel module\n");
 
 	serv.hnd.fd = socket(PF_INET, SOCK_STREAM, 0);
 	if (serv.hnd.fd < 0) {
--- a/accel-pppd/ctrl/ipoe/ipoe_netlink.c
+++ b/accel-pppd/ctrl/ipoe/ipoe_netlink.c
@@ -624,7 +624,7 @@ static void init(void)
 {
	int mcg_id;

-	if (access("/sys/module/ipoe", F_OK) && system("modprobe -q ipoe"))
+	if (access("/sys/module/ipoe", F_OK))
		log_warn("failed to load ipoe module\n");

	mcg_id = genl_resolve_mcg(IPOE_GENL_NAME, IPOE_GENL_MCG_PKT, &ipoe_genl_id);
--- a/accel-pppd/shaper/limiter.c
+++ b/accel-pppd/shaper/limiter.c
@@ -636,8 +636,6 @@ static int init_htb(const char *name)
		.qdisc = qdisc_htb_root,
	};

-	if (system("modprobe -q ifb"))
-		log_warn("failed to load ifb kernel module\n");

	memset(&ifr, 0, sizeof(ifr));
	strcpy(ifr.ifr_name, name);
--- a/accel-pppd/vlan-mon/vlan_mon.c
+++ b/accel-pppd/vlan-mon/vlan_mon.c
@@ -522,7 +522,7 @@ static void init(void)
 {
	int mcg_id;

-	if (access("/sys/module/vlan_mon", F_OK) && system("modprobe -q vlan_mon"))
+	if (access("/sys/module/vlan_mon", F_OK))
		log_warn("failed to load vlan_mon module\n");

	mcg_id = genl_resolve_mcg(VLAN_MON_GENL_NAME, VLAN_MON_GENL_MCG, &vlan_mon_genl_id);
