--- ./drivers/pptp/if_pppox.h.orig	2015-11-25 15:15:53.011615165 +0300
+++ ./drivers/pptp/if_pppox.h	2015-11-25 15:16:03.571615341 +0300
@@ -55,7 +55,7 @@
  * Protocols supported by AF_PPPOX
  */
 #define PX_PROTO_OE    0 /* Currently just PPPoE */
-#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,22)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
 #define PX_PROTO_PPTP  1
 #define PX_MAX_PROTO   2
 #else
--- ./accel-pppd/include/if_pppox.h.orig	2015-11-25 15:16:26.223615718 +0300
+++ ./accel-pppd/include/if_pppox.h	2015-11-25 15:16:34.427615854 +0300
@@ -51,7 +51,7 @@
  * Protocols supported by AF_PPPOX
  */
 #define PX_PROTO_OE    0 /* Currently just PPPoE */
-#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,22)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
 #define PX_PROTO_PPTP  1
 #define PX_MAX_PROTO   2
 #else
