--- a/accel-pppd/extra/chap-secrets.c
+++ b/accel-pppd/extra/chap-secrets.c
@@ -21,7 +21,7 @@
 
 #include "memdebug.h"
 
-static char *def_chap_secrets = "/etc/ppp/chap-secrets";
+static char *def_chap_secrets = "/opt/etc/ppp/chap-secrets";
 static char *conf_chap_secrets;
 static int conf_encrypted;
 static in_addr_t conf_gw_ip_address = 0;
--- a/accel-pppd/extra/pppd_compat.c
+++ b/accel-pppd/extra/pppd_compat.c
@@ -32,11 +32,11 @@
 #define ENV_MEM 1024
 #define ENV_MAX 16
 
-static char *conf_ip_up = "/etc/ppp/ip-up";
-static char *conf_ip_pre_up = "/etc/ppp/ip-pre-up";
-static char *conf_ip_down = "/etc/ppp/ip-down";
-static char *conf_ip_change = "/etc/ppp/ip-change";
-static char *conf_radattr_prefix = "/var/run/radattr";
+static char *conf_ip_up = "/opt/etc/ppp/ip-up";
+static char *conf_ip_pre_up = "/opt/etc/ppp/ip-pre-up";
+static char *conf_ip_down = "/opt/etc/ppp/ip-down";
+static char *conf_ip_change = "/opt/etc/ppp/ip-change";
+static char *conf_radattr_prefix = "/opt/var/run/radattr";
 static int conf_verbose = 0;
 
 static void *pd_key;
