--- a/appargs.cpp
+++ b/appargs.cpp
@@ -21,6 +21,7 @@
  *   Christian Ruppert (new include to build with GCC 4.4+)
  * 
  */
+#include <stdio.h>
 
 
 #include <cstdio>
--- a/icd-main.cpp
+++ b/icd-main.cpp
@@ -18,12 +18,14 @@

 #include <map>
 #include <signal.h>
+#include <stdio.h>
 #include <wait.h>
 #include <fcntl.h>
 #include <pwd.h>
 #include <dirent.h>
 #include <syslog.h>
 #include <errno.h>
+#include <unistd.h>
 #include <sys/poll.h>
 #include <sys/stat.h>
 #include <cstdio>
--- a/ict-main.cpp
+++ b/ict-main.cpp
@@ -18,7 +18,8 @@
  */
 
 
-#include <argp.h>
+#include <errno.h>
+#include <unistd.h>
 #include <pwd.h>
 #include <string>
 #include <unistd.h>
@@ -493,7 +494,7 @@
       if (uid != 0) {
         fprintf(stderr, "insufficient privileges to use custom configuration (will use default)\n");
       }
-      else if (euidaccess(cfg.c_str(), R_OK) != 0) {
+      else if (access(cfg.c_str(), R_OK) != 0) {
         perror("cannot read configuration file (will use default)");
       }
     }
--- ./inotify-cxx.cpp
+++ ./inotify-cxx.cpp
@@ -24,6 +24,7 @@
  
 
 #include <errno.h>
+#include <stdio.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <cstdio>
--- a/usertable.cpp
+++ b/usertable.cpp
@@ -19,6 +19,7 @@
  */
 
 
+#include <stdio.h>
 #include <pwd.h>
 #include <syslog.h>
 #include <errno.h>
