--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,8 @@
 
 CHECK_FUNCTION_EXISTS(open_memstream HAVE_OPEN_MEMSTREAM)
 CHECK_FUNCTION_EXISTS(fmemopen HAVE_FMEMOPEN)
-CHECK_FUNCTION_EXISTS(arc4random_uniform HAVE_ARC4RANDOM_UNIFORM)
+SET(HAVE_ARC4RANDOM_UNIFORM 1)
+
 CHECK_FUNCTION_EXISTS(pledge HAVE_PLEDGE)
 CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/sysconfig.h.in
   ${CMAKE_CURRENT_BINARY_DIR}/sysconfig.h)
@@ -74,5 +74,3 @@
     RENAME "rcon")
 ENDIF()
 
-ENABLE_TESTING()
-ADD_SUBDIRECTORY(tests)
--- a/srcrcon.c
+++ b/srcrcon.c
@@ -95,7 +95,7 @@
 
 static void src_rcon_message_random_id(src_rcon_message_t *m)
 {
-    m->id = (int32_t)arc4random_uniform(INT32_MAX-1);
+    m->id = (int32_t)(random() % (INT32_MAX-1));
 }
 
 static rcon_error_t
--- a/main.c
+++ b/main.c
@@ -460,6 +460,7 @@
     }
 #endif
 
+    srandom(time(NULL));
     response = g_byte_array_new();
     r = src_rcon_new();
 
