--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,7 +96,7 @@
 
 # Check for FUSE.
 find_package (FUSE REQUIRED)
-include_directories (SYSTEM ${FUSE_INCLUDE_DIR})
+include_directories (${FUSE_INCLUDE_DIR})
 add_definitions (-D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=29)
 if (CYGWIN)
   # Cygwin build is intended to use WinFsp
@@ -105,7 +105,7 @@
 
 # Check for OpenSSL.
 find_package (OpenSSL REQUIRED)
-include_directories (SYSTEM ${OPENSSL_INCLUDE_DIR})
+include_directories (${OPENSSL_INCLUDE_DIR})
 
 find_program (POD2MAN pod2man)
 
@@ -187,7 +187,7 @@
 if (ENABLE_NLS)
   find_package (Intl)
   if (Intl_FOUND)
-    include_directories (SYSTEM ${Intl_INCLUDE_DIRS})
+    include_directories (${Intl_INCLUDE_DIRS})
   endif()
 
   add_subdirectory(po)
@@ -236,11 +236,11 @@
   endif ()
   set(BUILD_TESTS OFF CACHE BOOL "build tests")
   add_subdirectory(${TINYXML_DIR} EXCLUDE_FROM_ALL)
-  include_directories(SYSTEM ${CMAKE_CURRENT_LIST_DIR}/${TINYXML_DIR})
+  include_directories(${CMAKE_CURRENT_LIST_DIR}/${TINYXML_DIR})
   link_directories(${CMAKE_BINARY_DIR}/${TINYXML_DIR})
 else ()
   find_package (TinyXML REQUIRED)
-  include_directories (SYSTEM ${TINYXML_INCLUDE_DIR})
+  include_directories (${TINYXML_INCLUDE_DIR})
 endif ()
 
 set(SOURCE_FILES
