diff -ur libupnp-1.8-1.8.3.orig/ixml/Makefile.am libupnp-1.8-1.8.3/ixml/Makefile.am
--- libupnp-1.8-1.8.3.orig/ixml/Makefile.am	2017-11-14 14:02:53.000000000 +0300
+++ libupnp-1.8-1.8.3/ixml/Makefile.am	2018-08-29 16:52:03.577037705 +0300
@@ -11,7 +11,7 @@
                           -I$(top_srcdir)/upnp/inc
 AM_CFLAGS		= 
 
-LDADD			= libixml.la
+LDADD			= libixml-1.8.la
 
 if ENABLE_DEBUG
     AM_CPPFLAGS		+= -DDEBUG
@@ -23,12 +23,12 @@
     AM_CPPFLAGS		+= -DIXML_HAVE_SCRIPTSUPPORT
 endif
 
-lib_LTLIBRARIES		= libixml.la
+lib_LTLIBRARIES		= libixml-1.8.la
 
-libixml_la_LDFLAGS	= -version-info $(LT_VERSION_IXML) \
+libixml_1_8_la_LDFLAGS	= -version-info $(LT_VERSION_IXML) \
 			  -export-symbols-regex '^ixml.*'
 
-libixml_la_SOURCES	= \
+libixml_1_8_la_SOURCES	= \
 			src/attr.c \
 			src/document.c \
 			src/element.c \
@@ -42,15 +42,15 @@
 			src/node.c \
 			src/nodeList.c
 
-upnpincludedir		= $(includedir)/upnp
+upnpincludedir		= $(includedir)/upnp-1.8
 upnpinclude_HEADERS	= \
 			inc/ixml.h \
 			inc/ixmldebug.h
 
-check_PROGRAMS          = test_document
+check_PROGRAMS          = test_document-1.8
 TESTS                   = test/test_document.sh
 
-test_document_SOURCES 	= test/test_document.c
+test_document_1_8_SOURCES 	= test/test_document.c
 
 EXTRA_DIST		= test/test_document.sh test/testdata
 
diff -ur libupnp-1.8-1.8.3.orig/libupnp.pc.in libupnp-1.8-1.8.3/libupnp.pc.in
--- libupnp-1.8-1.8.3.orig/libupnp.pc.in	2017-11-14 14:02:53.000000000 +0300
+++ libupnp-1.8-1.8.3/libupnp.pc.in	2018-08-29 16:52:03.577037705 +0300
@@ -3,9 +3,9 @@
 libdir=@libdir@
 includedir=@includedir@
 
-Name: libupnp
+Name: libupnp-1.8
 Description: Linux SDK for UPnP Devices
 Version: @VERSION@
-Libs: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ -L${libdir} -lupnp -lixml
-Cflags: @PTHREAD_CFLAGS@ -I${includedir}/upnp
+Libs: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ -L${libdir} -lupnp-1.8 -lixml-1.8
+Cflags: @PTHREAD_CFLAGS@ -I${includedir}/upnp-1.8
 
diff -ur libupnp-1.8-1.8.3.orig/libupnp.spec libupnp-1.8-1.8.3/libupnp.spec
--- libupnp-1.8-1.8.3.orig/libupnp.spec	2017-11-14 14:02:53.000000000 +0300
+++ libupnp-1.8-1.8.3/libupnp.spec	2018-08-29 16:52:03.577037705 +0300
@@ -48,7 +48,7 @@
 %{__mv} %{buildroot}%{docdir}/html \
 	%{buildroot}%{docdeveldir}/
 
-%{__rm} %{buildroot}%{_libdir}/{libixml.la,libupnp.la}
+%{__rm} %{buildroot}%{_libdir}/{libixml-1.8.la,libupnp-1.8.la}
 
 %post -p /sbin/ldconfig
 
@@ -57,18 +57,18 @@
 %files
 %defattr(-,root,root,-)
 %doc %{docdir}
-%{_libdir}/libixml.so.*
-%{_libdir}/libupnp.so.*
+%{_libdir}/libixml-1.8.so.*
+%{_libdir}/libupnp-1.8.so.*
 
 %files devel
 %defattr(0644,root,root,0755)
 %doc %{docdeveldir}
-%{_includedir}/upnp/
-%{_libdir}/libixml.so
-%{_libdir}/libupnp.so
-%{_libdir}/libixml.a
-%{_libdir}/libupnp.a
-%{_libdir}/pkgconfig/libupnp.pc
+%{_includedir}/upnp-1.8/
+%{_libdir}/libixml-1.8.so
+%{_libdir}/libupnp-1.8.so
+%{_libdir}/libixml-1.8.a
+%{_libdir}/libupnp-1.8.a
+%{_libdir}/pkgconfig/libupnp-1.8.pc
 
 %clean
 rm -rf %{buildroot}
diff -ur libupnp-1.8-1.8.3.orig/upnp/Makefile.am libupnp-1.8-1.8.3/upnp/Makefile.am
--- libupnp-1.8-1.8.3.orig/upnp/Makefile.am	2017-11-14 14:02:53.000000000 +0300
+++ libupnp-1.8-1.8.3/upnp/Makefile.am	2018-08-29 16:52:03.577037705 +0300
@@ -11,11 +11,11 @@
 	-I$(top_srcdir)/ixml/inc
 
 LDADD = \
-	libupnp.la \
-	$(top_builddir)/ixml/libixml.la
+	libupnp-1.8.la \
+	$(top_builddir)/ixml/libixml-1.8.la
 
 
-upnpincludedir = $(includedir)/upnp
+upnpincludedir = $(includedir)/upnp-1.8
 upnpinclude_HEADERS = \
 	inc/ActionComplete.h \
 	inc/ActionRequest.h \
@@ -47,16 +47,16 @@
 endif
 
 
-lib_LTLIBRARIES = libupnp.la
+lib_LTLIBRARIES = libupnp-1.8.la
 
-libupnp_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(srcdir)/src/inc 
+libupnp_1_8_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(srcdir)/src/inc 
 
-libupnp_la_LDFLAGS = \
+libupnp_1_8_la_LDFLAGS = \
 	-version-info $(LT_VERSION_UPNP) \
 	-export-symbols-regex '^Upnp.*' \
-	$(top_builddir)/ixml/libixml.la
+	$(top_builddir)/ixml/libixml-1.8.la
 
-libupnp_la_SOURCES = \
+libupnp_1_8_la_SOURCES = \
 	src/inc/config.h \
 	src/inc/client_table.h \
 	src/inc/ClientSubscription.h \
@@ -95,8 +95,8 @@
 upnpinclude_HEADERS += \
 	inc/ithread.h
 
-libupnp_la_CPPFLAGS += -I$(srcdir)/src/threadutil
-libupnp_la_SOURCES += \
+libupnp_1_8_la_CPPFLAGS += -I$(srcdir)/src/threadutil
+libupnp_1_8_la_SOURCES += \
 	src/threadutil/FreeList.h \
 	src/threadutil/FreeList.c \
 	src/threadutil/LinkedList.h \
@@ -108,7 +108,7 @@
 
 # ssdp
 if ENABLE_SSDP
-libupnp_la_SOURCES += \
+libupnp_1_8_la_SOURCES += \
 	src/ssdp/ssdp_ResultData.c \
 	src/ssdp/ssdp_ResultData.h \
 	src/ssdp/ssdp_device.c \
@@ -118,14 +118,14 @@
 
 # soap
 if ENABLE_SOAP
-libupnp_la_SOURCES += \
+libupnp_1_8_la_SOURCES += \
 	src/soap/soap_device.c \
 	src/soap/soap_ctrlpt.c \
 	src/soap/soap_common.c
 endif
 
 # genlib
-libupnp_la_SOURCES += \
+libupnp_1_8_la_SOURCES += \
 	src/genlib/miniserver/miniserver.c \
 	src/genlib/client_table/client_table.c \
 	src/genlib/client_table/ClientSubscription.c \
@@ -144,14 +144,14 @@
 
 # gena
 if ENABLE_GENA
-libupnp_la_SOURCES += \
+libupnp_1_8_la_SOURCES += \
 	src/gena/gena_device.c \
 	src/gena/gena_ctrlpt.c \
 	src/gena/gena_callback2.c
 endif
 
 # api
-libupnp_la_SOURCES += \
+libupnp_1_8_la_SOURCES += \
 	src/api/ActionComplete.c \
 	src/api/ActionRequest.c \
 	src/api/Discovery.c \
@@ -165,11 +165,11 @@
 	src/api/upnpapi.c
 
 if ENABLE_TOOLS
-libupnp_la_SOURCES += src/api/upnptools.c
+libupnp_1_8_la_SOURCES += src/api/upnptools.c
 endif
 
 if ENABLE_DEBUG
-libupnp_la_SOURCES += src/api/upnpdebug.c 
+libupnp_1_8_la_SOURCES += src/api/upnpdebug.c 
 AM_CPPFLAGS         += -DDEBUG -DSTATS
 else
 AM_CPPFLAGS         += -DNDEBUG
@@ -178,7 +178,7 @@
 
 # uuid
 if ENABLE_UUID
-libupnp_la_SOURCES += \
+libupnp_1_8_la_SOURCES += \
 	src/uuid/md5.c \
 	src/uuid/sysdep.c \
 	src/uuid/uuid.c
@@ -186,20 +186,20 @@
 
 
 # urlconfig
-libupnp_la_SOURCES += src/urlconfig/urlconfig.c
+libupnp_1_8_la_SOURCES += src/urlconfig/urlconfig.c
 
 
 # inet_pton (needed on Win32, compiles to nothing elsewhere)
-libupnp_la_SOURCES += \
+libupnp_1_8_la_SOURCES += \
 	src/inet_pton.c \
 	src/inc/inet_pton.h
 
 
 # check / distcheck tests
-check_PROGRAMS = test_init test_url
-TESTS = test_init test_url
-test_init_SOURCES = test/test_init.c
-test_url_SOURCES = test/test_url.c
+check_PROGRAMS = test_init-1.8 test_url-1.8
+TESTS = test_init-1.8 test_url-1.8
+test_init_1_8_SOURCES = test/test_init.c
+test_url_1_8_SOURCES = test/test_url.c
 
 
 EXTRA_DIST = \
diff -ur libupnp-1.8-1.8.3.orig/upnp/sample/Makefile.am libupnp-1.8-1.8.3/upnp/sample/Makefile.am
--- libupnp-1.8-1.8.3.orig/upnp/sample/Makefile.am	2017-11-14 14:02:53.000000000 +0300
+++ libupnp-1.8-1.8.3/upnp/sample/Makefile.am	2018-08-29 16:52:03.577037705 +0300
@@ -9,49 +9,49 @@
 	-I$(top_srcdir)/ixml/inc
 
 LDADD = \
-	$(top_builddir)/upnp/libupnp.la \
-	$(top_builddir)/ixml/libixml.la
+	$(top_builddir)/upnp/libupnp-1.8.la \
+	$(top_builddir)/ixml/libixml-1.8.la
 
 # samples
 noinst_PROGRAMS =
 if ENABLE_SAMPLES
 if ENABLE_CLIENT
-noinst_PROGRAMS += tv_ctrlpt
-tv_ctrlpt_CPPFLAGS = \
+noinst_PROGRAMS += tv_ctrlpt-1.8
+tv_ctrlpt_1_8_CPPFLAGS = \
 	$(AM_CPPFLAGS) \
 	-I$(srcdir)/common/ \
 	-I$(srcdir)/tvctrlpt
 if ENABLE_DEVICE
-noinst_PROGRAMS += tv_combo
-tv_combo_CPPFLAGS = $(AM_CPPFLAGS) \
+noinst_PROGRAMS += tv_combo-1.8
+tv_combo_1_8_CPPFLAGS = $(AM_CPPFLAGS) \
 	-I$(srcdir)/common/ \
 	-I$(srcdir)/tvcombo
 endif
 endif
 if ENABLE_DEVICE
-noinst_PROGRAMS += tv_device
-tv_device_CPPFLAGS = \
+noinst_PROGRAMS += tv_device-1.8
+tv_device_1_8_CPPFLAGS = \
 	$(AM_CPPFLAGS) \
 	-I$(srcdir)/common/ \
 	-I$(srcdir)/tvdevice
 endif
 endif
 
-tv_device_SOURCES = \
+tv_device_1_8_SOURCES = \
 	common/sample_util.c \
 	common/sample_util.h \
 	common/tv_device.c \
 	common/tv_device.h \
 	linux/tv_device_main.c
 
-tv_ctrlpt_SOURCES = \
+tv_ctrlpt_1_8_SOURCES = \
 	common/sample_util.c \
 	common/sample_util.h \
 	common/tv_ctrlpt.c \
 	common/tv_ctrlpt.h \
 	linux/tv_ctrlpt_main.c
 
-tv_combo_SOURCES = \
+tv_combo_1_8_SOURCES = \
 	common/sample_util.c \
 	common/sample_util.h \
 	common/tv_ctrlpt.c \
@@ -63,8 +63,8 @@
 if WITH_DOCUMENTATION
 examplesdir = $(docdir)/examples
 examples_DATA = \
-	$(tv_ctrlpt_SOURCES) \
-	$(tv_device_SOURCES)
+	$(tv_ctrlpt_1_8_SOURCES) \
+	$(tv_device_1_8_SOURCES)
 endif
 
 EXTRA_DIST = \
diff -ur libupnp-1.8-1.8.3.orig/upnp/unittest/Makefile.am libupnp-1.8-1.8.3/upnp/unittest/Makefile.am
--- libupnp-1.8-1.8.3.orig/upnp/unittest/Makefile.am	2017-11-14 14:02:53.000000000 +0300
+++ libupnp-1.8-1.8.3/upnp/unittest/Makefile.am	2018-08-29 16:52:03.577037705 +0300
@@ -10,16 +10,16 @@
 	-I$(top_srcdir)/upnp/src/inc
 
 LDADD = \
-	$(top_builddir)/upnp/libupnp.la \
-	$(top_builddir)/ixml/libixml.la
+	$(top_builddir)/upnp/libupnp-1.8.la \
+	$(top_builddir)/ixml/libixml-1.8.la
 
 #unittest
-noinst_PROGRAMS = unittest
+noinst_PROGRAMS = unittest-1.8
 
-unittest_CPPFLAGS = \
+unittest_1_8_CPPFLAGS = \
 	$(AM_CPPFLAGS)
 
-unittest_SOURCES = \
+unittest_1_8_SOURCES = \
 	main.c \
 	templates/FirstObject.c \
 	templates/FirstObject.h \
