--- a/Makefile.cmdline
+++ b/Makefile.cmdline
@@ -8,17 +8,17 @@
 DUKTAPE_CMDLINE_SOURCES = \
 	examples/cmdline/duk_cmdline.c
 
-CC	= gcc
-CCOPTS	= -Os -pedantic -std=c99 -Wall -fstrict-aliasing -fomit-frame-pointer
+CC	?= gcc
+CCOPTS	= $(CFLAGS) -pedantic -std=c99 -Wall -fstrict-aliasing -fomit-frame-pointer
 CCOPTS += -I./src   # duktape.h and duk_config.h must be in include path
-CCLIBS	= -lm
+CCLIBS	= $(LDFLAGS) -lm
 
 # If you want linenoise, you can enable these.  At the moment linenoise
 # will cause some harmless compilation warnings.
-#CCOPTS += -DDUK_CMDLINE_FANCY
-#DUKTAPE_CMDLINE_SOURCES += linenoise/linenoise.c
-#CCOPTS += -I./linenoise
-#duk:	linenoise
+CCOPTS += -DDUK_CMDLINE_FANCY
+DUKTAPE_CMDLINE_SOURCES += linenoise/linenoise.c
+CCOPTS += -I./linenoise
+duk:	linenoise
 
 # Optional feature defines, see: http://duktape.org/guide.html#compiling
 CCOPTS += -DDUK_OPT_SELF_TESTS
