--- a/Makefile.cmdline
+++ b/Makefile.cmdline
@@ -8,10 +8,10 @@
 CMDLINE_SOURCES = \
 	examples/cmdline/duk_cmdline.c
 
-CC	= gcc
-CCOPTS	= -Os -pedantic -std=c99 -Wall -fstrict-aliasing -fomit-frame-pointer
+CC	?= gcc
+CCOPTS	= $(CFLAGS) -pedantic -std=c99 -Wall -fstrict-aliasing -fomit-frame-pointer
 CCOPTS += -I./src   # duktape.h and duk_config.h must be in include path
-CCLIBS	= -lm
+CCLIBS	= $(LDFLAGS) -lm
 
 # Enable print() and alert() for command line using an optional extra module.
 CCOPTS += -DDUK_CMDLINE_PRINTALERT_SUPPORT -I./extras/print-alert
@@ -31,9 +31,9 @@
 
 # If you want linenoise, you can enable these.  At the moment linenoise
 # will cause some harmless compilation warnings.
-#CCOPTS += -DDUK_CMDLINE_FANCY -I./linenoise
-#CMDLINE_SOURCES += linenoise/linenoise.c
-#duk:	linenoise
+CCOPTS += -DDUK_CMDLINE_FANCY -I./linenoise
+CMDLINE_SOURCES += linenoise/linenoise.c
+duk:	linenoise
 
 # Use the tools/configure.py utility to modify Duktape default configuration:
 # http://duktape.org/guide.html#compiling
