--- a/Makefile.sharedlibrary
+++ b/Makefile.sharedlibrary
@@ -29,7 +29,7 @@
 REAL_VERSION=$(SONAME_VERSION).$(DUK_VERSION)
 
 # Change to actual path for actual distribution packaging.
-INSTALL_PREFIX=/usr/local
+INSTALL_PREFIX?=$(DESTDIR)/opt
 
 # The 'noline' variant may be more appropriate for some distributions; it
 # doesn't have #line directives in the combined source.
@@ -37,23 +37,28 @@
 #DUKTAPE_SRCDIR=./src-noline
 
 .PHONY: all
-all: libduktape.so.$(REAL_VERSION) libduktaped.so.$(REAL_VERSION)
+all: libduktape.so.$(REAL_VERSION) libduktaped.so.$(REAL_VERSION) duk
 
 # If the default duk_config.h is not suitable for the distribution, modify it
 # before compiling the shared library and copy the same, edited duk_config.h
 # to $INSTALL_PREFIX/include on installation.
 
 libduktape.so.$(REAL_VERSION):
-	gcc -shared -fPIC -Wall -Wextra -Os -Wl,-soname,libduktape.so.$(SONAME_VERSION) \
+	$(GCC) $(CFLAGS) -shared -fPIC -Wall -Wextra -Os -Wl,-soname,libduktape.so.$(SONAME_VERSION) \
 		-o $@ $(DUKTAPE_SRCDIR)/duktape.c
+	ln -s libduktape.so.$(REAL_VERSION) libduktape.so
+	ln -s libduktape.so.$(REAL_VERSION) libduktape.so.$(SONAME_VERSION)
 
 libduktaped.so.$(REAL_VERSION):
-	gcc -shared -fPIC -g -Wall -Wextra -Os -Wl,-soname,libduktaped.so.$(SONAME_VERSION) \
+	$(GCC) $(CFLAGS) -shared -fPIC -g -Wall -Wextra -Os -Wl,-soname,libduktaped.so.$(SONAME_VERSION) \
 		-o $@ $(DUKTAPE_SRCDIR)/duktape.c
+	ln -s libduktaped.so.$(REAL_VERSION) libduktaped.so
+	ln -s libduktaped.so.$(REAL_VERSION) libduktaped.so.$(SONAME_VERSION)
 
 # Symlinks depend on platform conventions.
 .PHONY: install
 install: libduktape.so.$(REAL_VERSION) libduktaped.so.$(REAL_VERSION)
+	mkdir -p $(INSTALL_PREFIX)/lib
 	cp $+ $(INSTALL_PREFIX)/lib/
 	rm -f $(INSTALL_PREFIX)/lib/libduktape.so $(INSTALL_PREFIX)/lib/libduktape.so.$(SONAME_VERSION)
 	ln -s libduktape.so.$(REAL_VERSION) $(INSTALL_PREFIX)/lib/libduktape.so
@@ -61,6 +66,7 @@
 	rm -f $(INSTALL_PREFIX)/lib/libduktaped.so $(INSTALL_PREFIX)/lib/libduktaped.so.$(SONAME_VERSION)
 	ln -s libduktaped.so.$(REAL_VERSION) $(INSTALL_PREFIX)/lib/libduktaped.so
 	ln -s libduktaped.so.$(REAL_VERSION) $(INSTALL_PREFIX)/lib/libduktaped.so.$(SONAME_VERSION)
+	mkdir -p $(INSTALL_PREFIX)/include
 	cp $(DUKTAPE_SRCDIR)/duktape.h $(DUKTAPE_SRCDIR)/duk_config.h $(INSTALL_PREFIX)/include/
 
 # Note: assumes /usr/local/include/ and /usr/local/lib/ are in include/link
@@ -68,4 +74,4 @@
 #CCOPTS=-I/usr/local/include -L/usr/local/lib
 CCOPTS=-I./examples/cmdline
 duk:
-	gcc $(CCOPTS) -Wall -Wextra -Os -o $@ ./examples/cmdline/duk_cmdline.c -lduktape -lm
+	$(GCC) $(CFLAGS) -Isrc $(LDFLAGS) -DDUK_CMDLINE_PRINTALERT_SUPPORT -I./extras/print-alert -L. -Wall -Wextra -o $@ ./examples/cmdline/duk_cmdline.c ./extras/print-alert/duk_print_alert.c -lduktape -lm
