From: Stefan Fritsch <sf@sfritsch.de>
Subject: don't return 0 on error
Date: 2009-07-24 15:23:32 +0200

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=433757

--- a/mp3gain.c
+++ b/mp3gain.c
@@ -1992,6 +1992,7 @@
 					} else {
 						fprintf(stderr,"No undo information in %s\n",argv[mainloop]);
 					}
+					gSuccess = 0;
 				}
 		  }
 	  }
@@ -2062,8 +2063,9 @@
 #else
 		  if ((inf == NULL)&&(tagInfo[mainloop].recalc > 0)) {
 #endif
-			  fprintf(stdout, "Can't open %s for reading\n",argv[mainloop]);
-              fflush(stdout);
+			  fprintf(stderr, "Can't open %s for reading\n",argv[mainloop]);
+			  fflush(stderr);
+			  gSuccess = 0;
 		  }
 		  else {
 #ifdef AACGAIN
@@ -2148,8 +2150,9 @@
 				if (!ok) {
 #endif
                     if (!BadLayer) {
-						fprintf(stdout,"Can't find any valid MP3 frames in file %s\n",argv[mainloop]);
-                        fflush(stdout);
+						fprintf(stderr,"Can't find any valid MP3 frames in file %s\n",argv[mainloop]);
+						fflush(stderr);
+						gSuccess = 0;
                     }
 				}
 				else {
@@ -2184,8 +2187,8 @@
 								(Xingcheck[0] == 'I' && Xingcheck[1] == 'n' && Xingcheck[2] == 'f' && Xingcheck[3] == 'o')) {
 							bitridx = (curframe[2] >> 4) & 0x0F;
 							if (bitridx == 0) {
-								fprintf(stdout, "%s is free format (not currently supported)\n",curfilename);
-								fflush(stdout);
+								fprintf(stderr, "%s is free format (not currently supported)\n",curfilename);
+								fflush(stderr);
 								ok = 0;
 							}
 							else {
@@ -2228,8 +2231,8 @@
 						while (ok) {
 							bitridx = (curframe[2] >> 4) & 0x0F;
 							if (bitridx == 0) {
-								fprintf(stdout,"%s is free format (not currently supported)\n",curfilename);
-								fflush(stdout);
+								fprintf(stderr,"%s is free format (not currently supported)\n",curfilename);
+								fflush(stderr);
 								ok = 0;
 							}
 							else {
@@ -2315,8 +2318,9 @@
 					}
 
 					if (dBchange == GAIN_NOT_ENOUGH_SAMPLES) {
-						fprintf(stdout,"Not enough samples in %s to do analysis\n",argv[mainloop]);
-                        fflush(stdout);
+						fprintf(stderr,"Not enough samples in %s to do analysis\n",argv[mainloop]);
+                        			fflush(stderr);
+						gSuccess = 0;
 						numFiles--;
 					}
 					else {
@@ -2456,8 +2460,8 @@
 		}
 
 		if (dBchange == GAIN_NOT_ENOUGH_SAMPLES) {
-			fprintf(stdout,"Not enough samples in mp3 files to do analysis\n");
-            fflush(stdout);
+			fprintf(stderr,"Not enough samples in mp3 files to do analysis\n");
+			fflush(stderr);
 		}
 		else {
 			Float_t maxmaxsample;
