From: Stefan Fritsch <sf@debian.org>
Subject: fix overwrite query loop
Date: 2007-06-17 17:02:34 +0200

Assume "n" if we don't have stdin.

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=428301

--- a/mp3gain.c
+++ b/mp3gain.c
@@ -1213,7 +1213,7 @@
 static 
 int queryUserForClipping(char * argv_mainloop,int intGainChange)
 {
-	char ch;
+	int ch;
 
 	fprintf(stderr,"\nWARNING: %s may clip with mp3 gain change %d\n",argv_mainloop,intGainChange);
 	ch = 0;
@@ -1223,6 +1223,9 @@
 		fprintf(stderr,"Make change? [y/n]:");
 		fflush(stderr);
 		ch = getchar();
+		if (ch == EOF) {
+			ch='N';
+		}
 		ch = toupper(ch);
 	}
 	if (ch == 'N')
