From: Stefan Fritsch <sf@debian.org>
Subject: fix some issues found by cppcheck
Date: 2010-04-21 00:36:31 +0200

--- a/id3tag.c
+++ b/id3tag.c
@@ -303,6 +303,8 @@
 				break;
 			default:
 				va_end(ap);
+				free(frame->data);
+				free(frame);
 				return NULL;
 		}
 	}
--- a/mp3gain.c
+++ b/mp3gain.c
@@ -762,6 +762,7 @@
             passError(MP3GAIN_UNSPECIFED_ERROR, 3,
                 "\nCan't open ", outfilename, " for temp writing\n");
 			NowWriting = 0;
+			free(outfilename);
 			return M3G_ERR_CANT_MAKE_TMP;
 		} 
  
@@ -779,6 +780,7 @@
 	  passError( MP3GAIN_UNSPECIFED_ERROR, 3,
           "\nCan't open ", filename, " for modifying\n");
 	  NowWriting = 0;
+	  free(outfilename);
 	  return M3G_ERR_CANT_MODIFY_FILE;
   }
   else {
