Description: add option to force tag update
Author: Simon McVittie <smcv@debian.org>
Forwarded: https://sourceforge.net/tracker/?func=detail&aid=3022529&group_id=49979&atid=458161

--- a/mp3gain.c
+++ b/mp3gain.c
@@ -143,6 +143,7 @@
 int skipTag = 0;
 int deleteTag = 0;
 int forceRecalculateTag = 0;
+int forceUpdateTag = 0;
 int checkTagOnly = 0;
 static int useId3 = 0;
 
@@ -1672,6 +1673,10 @@
                         case 'S':
                             skipTag = !0;
                             break;
+			case 'u':
+			case 'U':
+			    forceUpdateTag = !0;
+			    break;
                         case 'r':
                         case 'R':
                             forceRecalculateTag = !0;
@@ -1761,7 +1766,7 @@
       fileTags[mainloop].apeTag = NULL;
 	  fileTags[mainloop].lyrics3tag = NULL;
 	  fileTags[mainloop].id31tag = NULL;
-	  tagInfo[mainloop].dirty = 0;
+	  tagInfo[mainloop].dirty = forceUpdateTag;
 	  tagInfo[mainloop].haveAlbumGain = 0;
 	  tagInfo[mainloop].haveAlbumPeak = 0;
 	  tagInfo[mainloop].haveTrackGain = 0;
