From: Simon McVittie <smcv@debian.org>
Date: Sun, 16 Mar 2014 22:21:26 +0000
Subject: Increase MAXFRAMESIZE to 3456 bytes, which is much closer to reality

Author: Daniel Kobras
Origin: vendor, Debian (mpg123/0.59r-14)
---
 mpglibDBL/mpg123.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mpglibDBL/mpg123.h b/mpglibDBL/mpg123.h
index 691eb9c..08bd217 100644
--- a/mpglibDBL/mpg123.h
+++ b/mpglibDBL/mpg123.h
@@ -61,7 +61,7 @@ char *strchr (), *strrchr ();
 #define         MPG_MD_DUAL_CHANNEL     2
 #define         MPG_MD_MONO             3
 
-#define MAXFRAMESIZE 1792
+#define MAXFRAMESIZE 3456
 
 /* AF: ADDED FOR LAYER1/LAYER2 */
 #define         SCALE_BLOCK             12
