--- a/Makefile
+++ b/Makefile
@@ -1,14 +1,14 @@
-CC     = gcc
-CFLAGS = -W -Wall -pedantic -ansi -std=c99 -g
+CC     ?= gcc
+CFLAGS += -W -Wall -pedantic -ansi -std=c99 -g
 
-LDFLAGS = -lncursesw
+LDFLAGS += -lncursesw
 
 # OS X installs ncurses with wide character support, but not as "libncurses"
 ifeq ($(shell uname -s),Darwin)
 	LDFLAGS = -lncurses
 endif
 
-PREFIX  = /usr/local
+PREFIX  ?= /usr/local
 
 EXECUTABLE = ttysolitaire
 SRC_DIR    = src
