--- a/src/ttysolitaire.c
+++ b/src/ttysolitaire.c
@@ -122,14 +122,5 @@
 }
 
 void version() {
-  FILE *version_file;
-  char version_string[6];
-
-  if (!(version_file = fopen("VERSION", "rb"))) {
-    tty_solitaire_generic_error(errno, __FILE__, __LINE__);
-  }
-  fread(version_string, 1, 5, version_file);
-  version_string[5] = '\0';
-  printf("%s\n", version_string);
-  fclose(version_file);
+  printf("0.0.0\n");
 }
