--- a/git/configure.ac
+++ b/git/configure.ac
@@ -852,7 +852,8 @@ AC_RUN_IFELSE(
 	return v != 0xfe + 0xff;
 	]])],
 	[ac_cv_iconv_omits_bom=no],
-	[ac_cv_iconv_omits_bom=yes])
+	[ac_cv_iconv_omits_bom=yes],
+	[ac_cv_iconv_omits_bom=no])
 
 LIBS="$old_LIBS"
 ])
@@ -966,7 +967,8 @@ AC_RUN_IFELSE(
 		FILE *f = fopen(".", "r");
 		return f != NULL;]])],
 	[ac_cv_fread_reads_directories=no],
-	[ac_cv_fread_reads_directories=yes])
+	[ac_cv_fread_reads_directories=yes],
+	[ac_cv_fread_reads_directories=no])
 ])
 if test $ac_cv_fread_reads_directories = yes; then
 	FREAD_READS_DIRECTORIES=UnfortunatelyYes
@@ -1000,7 +1002,8 @@ AC_RUN_IFELSE(
 		  if (snprintf(buf, 3, "%s", "12345") != 5
 		      || strcmp(buf, "12")) return 1]])],
 	[ac_cv_snprintf_returns_bogus=no],
-	[ac_cv_snprintf_returns_bogus=yes])
+	[ac_cv_snprintf_returns_bogus=yes],
+	[ac_cv_snprintf_returns_bogus=no])
 ])
 if test $ac_cv_snprintf_returns_bogus = yes; then
 	SNPRINTF_RETURNS_BOGUS=UnfortunatelyYes
@@ -1021,9 +1024,10 @@ AC_EGREP_CPP(yippeeyeswehaveit,
 	S_IFIFO == 0010000 && S_IFSOCK == 0140000
 yippeeyeswehaveit
 #endif
-]),
+])
 	[ac_cv_sane_mode_bits=yes],
-	[ac_cv_sane_mode_bits=no])
+	[ac_cv_sane_mode_bits=no],
+	[ac_cv_sane_mode_bits=yes])
 ])
 if test $ac_cv_sane_mode_bits = yes; then
 	NEEDS_MODE_TRANSLATION=
