--- a/Makefile
+++ b/Makefile
@@ -2,11 +2,11 @@ all::
 
 CGIT_VERSION = v1.2.3
 CGIT_SCRIPT_NAME = cgit.cgi
-CGIT_SCRIPT_PATH = /var/www/htdocs/cgit
+CGIT_SCRIPT_PATH = /opt/var/www/htdocs/cgit
 CGIT_DATA_PATH = $(CGIT_SCRIPT_PATH)
-CGIT_CONFIG = /etc/cgitrc
-CACHE_ROOT = /var/cache/cgit
-prefix = /usr/local
+CGIT_CONFIG = /opt/etc/cgitrc
+CACHE_ROOT = /opt/var/cache/cgit
+prefix = /opt
 libdir = $(prefix)/lib
 filterdir = $(libdir)/cgit/filters
 docdir = $(prefix)/share/doc/cgit
