--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,9 +15,7 @@
 SET(CMAKE_C_FLAGS_RELEASE "-O2")
 SET(CMAKE_C_FLAGS_DEBUG "-O0 -g -DDEBUG")
 
-if (UNIX AND APPLE)
-  include_directories("/usr/local/include/osxfuse/fuse")
-endif()
+include_directories(${OPTIONAL_INCLUDE_DIR})
 
 add_definitions(-D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=26)
 
@@ -42,4 +40,4 @@
 ENDIF (WITH_XATTR)
 
 add_subdirectory(src)
-add_subdirectory(man)
+#add_subdirectory(man)
